# Association tests - Part I

These exercises introduce genetic association testing: how to identify which genetic variants are associated with a phenotype. Here, we introduce the "null model" and single-variant association testing, as is commonly performed in GWAS (Genome Wide Association Studies).

## Null model

The first step in our association testing procedure is to fit the null model -- i.e., a model fit under the null hypothesis of no individual variant association. Operationally, this is fitting a regression model with the desired outcome phenotype, fixed effect covariates, and random effects.   

### Prepare the data

To fit the null model, we will need to create an `AnnotatedDataFrame` with sample information and phenotype data; this class is defined in the Biobase package. We will merge our sample annotation file, which is indexed by a `sample.id` column matched to the GDS file, with our phenotype file, which is indexed by a `subject_id` column. 

NOTE: In this example, we use the 1000 Genomes IDs for both sample and subject IDs, though we would generally advise using separate IDs for samples (sequencing instances) and subjects (individuals).

```{r null_model}
# sample annotation
repo_path <- "https://github.com/UW-GAC/SISG_2021/raw/master"
if (!dir.exists("data")) dir.create("data")
sampfile <- "data/sample_annotation.RData"
if (!file.exists(sampfile)) download.file(file.path(repo_path, sampfile), sampfile)
samp <- get(load(sampfile))

library(Biobase)
# access the data with the pData() function
head(pData(samp))

# access the metadata with the varMetadata() function
varMetadata(samp)

# phenotype data
phenfile <- "data/phenotype_annotation.RData"
if (!file.exists(phenfile)) download.file(file.path(repo_path, phenfile), phenfile)
phen <- get(load(phenfile))

# access the data with the pData() function
head(pData(phen))

# access the metadata with the varMetadata() function
varMetadata(phen)

# merge sample annotation with phenotype data
library(dplyr)
dat <- pData(samp) %>%
    left_join(pData(phen), by=c("subject.id"="subject_id", "sex"="sex"))
head(dat)

# merge the metadata
meta <- bind_rows(varMetadata(samp), varMetadata(phen)[3:5,,drop=FALSE])

# make an AnnotatedDataFrame
annot <- AnnotatedDataFrame(dat, meta)
save(annot, file="data/sample_phenotype_annotation.RData")
```

### Fit the null model

We will test for an association between variant genotypes and height, adjusting for sex, age, and study as fixed effect covariates. If the sample set involves multiple distinct groups with different variances for the phenotype, we recommend allowing for heterogeneous residual variance among groups with the `group.var` parameter. We saw in a previous exercise that the variance of height differs by study.

```{r null_model_fit}
library(GENESIS)

# fit the null model
nullmod <- fitNullModel(annot, 
                        outcome="height", 
                        covars=c("sex", "age", "study"), 
                        group.var="study", 
                        verbose=FALSE)
save(nullmod, file="data/null_model.RData")
```

The `fitNullModel` function returns a lot of information about the model that was fit. We examine some of that information below; to see all of the components, try `names(nullmod)`.

```{r null_model_results}
# description of the model we fit
nullmod$model

# fixed effect regression estimates
nullmod$fixef

# variance component estimates by group.var
nullmod$varComp

# model fit: fitted values, residuals
head(nullmod$fit)

library(ggplot2)
ggplot(nullmod$fit, aes(x = fitted.values, y = resid.marginal)) + 
    geom_point(alpha = 0.5) + 
    geom_hline(yintercept = 0) + 
    geom_smooth(method = 'lm')
```

## Exercise

1. Inverse Normal transform: as discussed in the lecture, we recommend a fully adjusted two-stage inverse Normalization procedure for fitting the null model when phenotypes have non-Normal distributions. Using the `two.stage` option in `fitNullModel`, fit a two-stage null model. Compare these residuals with the residuals from the original null model.  

## Solution

To run the fully adjusted two.stage null model, we simply set the `two.stage` option to `TRUE`. The `norm.option` parameter determines if the inverse Normalization should be done with all samples together (`"all"`) or within each `group.var` group separately (`"by.group"`).

```{r null_model_two_stage}
nullmod.twostage <- fitNullModel(annot, 
                                 outcome="height", 
                                 covars=c("sex", "age", "study"), 
                                 group.var="study", 
                                 two.stage = TRUE,
                                 norm.option = "all",
                                 verbose=FALSE)
save(nullmod.twostage, file="data/null_model_two_stage.RData")

# description of the model we fit
nullmod.twostage$model

# compare the marginal residuals

# merge the data for plotting
pdat <- merge(nullmod$fit, nullmod.twostage$fit, 
              by = 'sample.id', suffixes = c('.orig', '.twostage'))
pdat <- merge(pdat, pData(annot), by = 'sample.id')

# distribution of residuals - original null model
ggplot(pdat, aes(x = resid.marginal.orig)) + 
    geom_density(aes(color = study)) + 
    geom_density(size = 2)

# distribution of residuals - two stage null model
ggplot(pdat, aes(x = resid.marginal.twostage)) + 
    geom_density(aes(color = study)) + 
    geom_density(size = 2)

# compare residuals
ggplot(pdat, aes(x = resid.marginal.orig, y = resid.marginal.twostage, color = study)) + 
    geom_point() + 
    geom_abline(intercept = 0, slope = 1)

```

See [Sofer et al.](https://onlinelibrary.wiley.com/doi/10.1002/gepi.22188) for more information on the fully adjusted two-stage model.


## Single-variant association tests

After fitting our null model, we can use score tests to test each variant across the genome individually for association with the outcome phenotype (i.e. height in our example). Performing these single-variant tests genome-wide is commonly referred to as a GWAS (Genome-Wide Association Study). 

We use the `assocTestSingle` function in GENESIS. First, we have to create a `SeqVarData` object including both the GDS file and the sample annotation containing phenotype data. We then create a `SeqVarBlockIterator` object, which breaks the set of all variants in the `SeqVarData` object into blocks, allowing us to analyze genome-wide in manageable pieces. The `assocTestSingle` function iterates over all blocks of variants in the `SeqVarBlockIterator` object and then concatenates and returns the results.

```{r assoc_single}
library(SeqVarTools)
gdsfile <- "data/1KG_phase3_subset_chr1.gds"
if (!file.exists(gdsfile)) download.file(file.path(repo_path, gdsfile), gdsfile)
gdsfmt::showfile.gds(closeall=TRUE) # make sure file is not already open
gds <- seqOpen(gdsfile)

# make the seqVarData object
seqData <- SeqVarData(gds, sampleData=annot)

# make the iterator object
iterator <- SeqVarBlockIterator(seqData, verbose=FALSE)

# run the single-variant association test
assoc <- assocTestSingle(iterator, nullmod)
dim(assoc)
head(assoc)
```

We make a QQ plot to examine the results.

```{r assoc_single_qq}
library(ggplot2)
qqPlot <- function(pval) {
    pval <- pval[!is.na(pval)]
    n <- length(pval)
    x <- 1:n
    dat <- data.frame(obs=sort(pval),
                      exp=x/n,
                      upper=qbeta(0.025, x, rev(x)),
                      lower=qbeta(0.975, x, rev(x)))
    
    ggplot(dat, aes(-log10(exp), -log10(obs))) +
        geom_line(aes(-log10(exp), -log10(upper)), color="gray") +
        geom_line(aes(-log10(exp), -log10(lower)), color="gray") +
        geom_point() +
        geom_abline(intercept=0, slope=1, color="red") +
        xlab(expression(paste(-log[10], "(expected P)"))) +
        ylab(expression(paste(-log[10], "(observed P)"))) +
        theme_bw()
}    

qqPlot(assoc$Score.pval)
```

A lot of the variants we tested are very rare -- the alternate allele is not observed for many samples. Single-variant tests do not perform well for very rare variants (we will discuss testing rare variants in more detail in the next session). We can use the minor allele count (MAC) observed in the sample to filter rare variants that we may expect to have unreliable test results.

```{r mac}
summary(assoc$MAC)
sum(assoc$MAC < 5)

qqPlot(assoc$Score.pval[assoc$MAC >= 5])
```

We should expect the majority of variants to fall near the red `y=x` line in the QQ plot. The deviation above the line, commonly referred to as "inflation" is indicative of some model issue. In this example, the issue is likely driven by the fact that we've ignored genetic ancestry and relatedness among these subjects -- more to come later when we discuss mixed models. 

## Exercise

1. Logistic regression: GENESIS also supports testing binary (e.g. case/control) outcomes. We can fit a null model using logistic regression by specifying the argument `family=binomial` in the `fitNullModel` function. Use the `status` column in the sample annotation to fit a null model for simulated case/control status, with `sex` and `Population` as covariates. Run single-variant association tests using this model and make a QQ plot of all variants with MAC >= 5.

## Solution

When testing binary outcomes, we should fit our null model using logistic regression. To do so, we simply set the argument `family=binomial` in `fitNullModel`. Note that the parameter `group.var` is no longer relevant here, as the logistic model specifies the mean-variance relationship. 

```{r exercise_logistic}
# fit the null model with logistic regression
nullmod.status <- fitNullModel(annot, 
                               outcome="status", 
                               covars=c("sex", "Population"), 
                               family=binomial, 
                               verbose=FALSE)

resetIterator(iterator, verbose=FALSE)
# run the single-variant association test
assoc.status <- assocTestSingle(iterator, nullmod.status, test="Score")
dim(assoc.status)
head(assoc.status)

# make a QQ plot
qqPlot(assoc.status$Score.pval[assoc.status$MAC >= 5])
```

Extra: in samples with highly imbalanced case:control ratios, the Score test can perform poorly for low frequency variants. Saddlepoint approximation (SPA) can be used to improve p-value calculations, and is available in GENESIS by setting the argument `test=Score.SPA` in `assocTestSingle`. See [Dey et al.](https://www.cell.com/ajhg/fulltext/S0002-9297(17)30201-X) and [Zhou et al.](https://www.nature.com/articles/s41588-018-0184-y) for details on using SPA in GWAS.

```{r close}
seqClose(gds)
```