# GDS format

GDS is Genomic Data Structure, a storage format that can efficiently store genomic data and provide fast random access to subsets of the data. For more information on GDS for sequence data, read the [SeqArray package vignette](https://github.com/zhengxwen/SeqArray/blob/master/vignettes/SeqArrayTutorial.Rmd).

## Convert a VCF to GDS

To use the R packages developed at the University of Washington Genetic Analysis Center for sequence data, we first need to convert a VCF file to GDS. (If the file is BCF, use [https://samtools.github.io/bcftools/bcftools.html](bcftools) to convert to VCF.)

```{r vcf2gds, message = FALSE}
library(SeqArray)
repo_path <- "https://github.com/UW-GAC/SISG_2021/raw/master"
if (!dir.exists("data")) dir.create("data")
vcffile <- "data/1KG_phase3_subset_chr1.vcf.gz"
if (!file.exists(vcffile)) download.file(file.path(repo_path, vcffile), vcffile)
gdsfile <- "data/1KG_phase3_subset_chr1.gds"
# convert the VCF to GDS
seqVCF2GDS(vcffile, gdsfile, fmt.import="GT", storage.option="LZMA_RA")
```

## Exploring a GDS file

We can interact with the GDS file using the SeqArray package.

```{r seqarray}
# open a connection to the GDS file
gds <- seqOpen(gdsfile)
gds
```

The `seqGetData` function is the basic function for reading in data from a GDS file
```{r seqGetData}
# the unique sample identifier comes from the VCF header
sample.id <- seqGetData(gds, "sample.id")
length(sample.id)
head(sample.id)

# a unique integer ID is assigned to each variant
variant.id <- seqGetData(gds, "variant.id")
length(variant.id)
head(variant.id)

chr <- seqGetData(gds, "chromosome")
head(chr)

pos <- seqGetData(gds, "position")
head(pos)

id <- seqGetData(gds, "annotation/id")
head(id)
```

There are additional useful functions for summary level data.

```{r ref_freq}
# reference allele frequency of each variant
afreq <- seqAlleleFreq(gds)
head(afreq)
summary(afreq)
hist(afreq, breaks=50)
```

We can define a filter on the `gds` object. After using the `seqSetFilter` command, all subsequent reads from the `gds` object are restricted to the selected subset of data, until a new filter is defined or `seqResetFilter` is called.

```{r filter}
seqSetFilter(gds, variant.id=91:100, sample.id=sample.id[1:5])
```

Genotype data is stored in a 3-dimensional array, where the first dimension is always length 2 for diploid genotypes. The second and third dimensions are samples and variants, respectively. The values of the array denote alleles: `0` is the reference allele and `1` is the alternate allele. For multiallelic variants, other alternate alleles are represented as integers `> 1`.

```{r genotypes}
geno <- seqGetData(gds, "genotype")
dim(geno)
geno[,,1:2]
```

The [SeqVarTools package](http://bioconductor.org/packages/SeqVarTools) has some additional functions for interacting with SeqArray-format GDS files. There are functions providing more intuitive ways to read in genotypes. 

```{r seqvartools_geno}
library(SeqVarTools)

# return genotypes in matrix format
getGenotype(gds)
getGenotypeAlleles(gds)
refDosage(gds)
altDosage(gds)
```

There are functions to extract variant information. 

```{r seqvartools_varinfo}
# look at reference and alternate alleles
refChar(gds)
altChar(gds)

# data.frame of variant information
variantInfo(gds)

# reset the filter to all variants and samples
seqResetFilter(gds)

# how many alleles for each variant?
n <- seqNumAllele(gds)
table(n)

# some variants have more than one alternate allele
multi.allelic <- which(n > 2)
altChar(gds)[multi.allelic]

# extract a particular alternate allele
# first alternate
altChar(gds, n=1)[multi.allelic]
# second alternate
altChar(gds, n=2)[multi.allelic]

# how many variants are biallelic SNVs?
table(isSNV(gds, biallelic=TRUE))
# how many variants are SNVs vs INDELs?
table(isSNV(gds, biallelic=FALSE))
# 11 SNVs are multi-allelic
```

We can also return variant information as a `GRanges` object from the [GenomicRanges package](https://bioconductor.org/packages/release/bioc/manuals/GenomicRanges/man/GenomicRanges.pdf). This format for representing sequence data is common across many Bioconductor packages. Chromosome is stored in the `seqnames` column. The `ranges` column has variant position, which can be a single base pair or a range.

```{r granges}
gr <- granges(gds)
gr
```

Always use the `seqClose` command to close your connection to a GDS file when you are done working with it. Trying to open an already opened GDS will result in an error. 

```{r intro_close}
seqClose(gds)
```


## Exercises 

1. Set a filter selecting only multi-allelic variants. Inspect their genotypes using the different methods you learned above. Use the `alleleDosage` method to find dosage for the second (and third, etc.) alternate allele.

2. Use the `hwe` function in SeqVarTools to run a Hardy-Weinberg Equilibrium test on each variant. Identify a variant with low p-value and inspect its genotypes. (Note that the HWE test is only valid for biallelic variants, and will return `NA` for multiallelic variants.)


## Solutions

1. Set a filter selecting only multi-allelic variants. Inspect their genotypes using the different methods you learned above. Use the `alleleDosage` method to find dosage for the second (and third, etc.) alternate allele.

```{r exercise_gds}
# open a connection to the GDS file again
gds <- seqOpen(gdsfile)

# set your filter
seqSetFilter(gds, variant.sel=multi.allelic)

geno <- seqGetData(gds, "genotype")
dim(geno)
geno[,1:5,1:5]

geno <- getGenotype(gds)
dim(geno)
head(geno)

geno <- getGenotypeAlleles(gds)
head(geno)

# count of reference alleles
dos <- refDosage(gds)
head(dos)

# count of *any* alternate alleles
dos <- altDosage(gds)
head(dos)

# count of the first alternate allele 
dos <- alleleDosage(gds, n=1)
head(dos)

# count of the third alternate allele
dos <- alleleDosage(gds, n=3)
head(dos)

# count of *each* of the alternate alleles
# returns multiple columns per variant
dos <- expandedAltDosage(gds)
head(dos)
```

2. Use the `hwe` function in SeqVarTools to run a Hardy-Weinberg Equilibrium test on each variant. Identify a variant with low p-value and inspect its genotypes. (Note that the HWE test is only valid for biallelic variants, and will return `NA` for multiallelic variants.)

```{r exercise_hwe}
# reset the filter to all variants
seqResetFilter(gds)

# run HWE test
hwe.res <- hwe(gds)

# identify variants with small p-values
lowp <- !is.na(hwe.res$p) & hwe.res$p < 1e-4
head(hwe.res[lowp,])

# look at the genotypes of the most significant variant
minp <- which.min(hwe.res$p)
hwe.res[minp,]

seqSetFilter(gds, variant.id=minp)
table(getGenotype(gds))
table(altDosage(gds))
```

Don't forget to close your connection to the GDS file when you're done!

```{r exercise_close}
seqClose(gds)
```
