<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>6 Ancestry and Relatedness Inference | SISG 2021 Module 16: Computational Pipeline for WGS Data</title>
  <meta name="description" content="Course materials for SISG 2021 Module 16" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="6 Ancestry and Relatedness Inference | SISG 2021 Module 16: Computational Pipeline for WGS Data" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Course materials for SISG 2021 Module 16" />
  <meta name="github-repo" content="UW-GAC/SISG_2021" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="6 Ancestry and Relatedness Inference | SISG 2021 Module 16: Computational Pipeline for WGS Data" />
  
  <meta name="twitter:description" content="Course materials for SISG 2021 Module 16" />
  



<meta name="date" content="2021-07-20" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="association-tests-part-ii.html"/>
<link rel="next" href="mixed-models.html"/>
<script src="libs/header-attrs-2.9/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#schedule"><i class="fa fa-check"></i><b>1.1</b> Schedule</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#r-packages-used"><i class="fa fa-check"></i><b>1.2</b> R packages used</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#resources"><i class="fa fa-check"></i><b>1.3</b> Resources</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="gds-format.html"><a href="gds-format.html"><i class="fa fa-check"></i><b>2</b> GDS format</a>
<ul>
<li class="chapter" data-level="2.1" data-path="gds-format.html"><a href="gds-format.html#convert-a-vcf-to-gds"><i class="fa fa-check"></i><b>2.1</b> Convert a VCF to GDS</a></li>
<li class="chapter" data-level="2.2" data-path="gds-format.html"><a href="gds-format.html#exploring-a-gds-file"><i class="fa fa-check"></i><b>2.2</b> Exploring a GDS file</a></li>
<li class="chapter" data-level="2.3" data-path="gds-format.html"><a href="gds-format.html#exercises"><i class="fa fa-check"></i><b>2.3</b> Exercises</a></li>
<li class="chapter" data-level="2.4" data-path="gds-format.html"><a href="gds-format.html#solutions"><i class="fa fa-check"></i><b>2.4</b> Solutions</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="phenotype-harmonization.html"><a href="phenotype-harmonization.html"><i class="fa fa-check"></i><b>3</b> Phenotype Harmonization</a>
<ul>
<li class="chapter" data-level="3.1" data-path="phenotype-harmonization.html"><a href="phenotype-harmonization.html#inspect-individual-study-data-in-r"><i class="fa fa-check"></i><b>3.1</b> Inspect individual study data in R</a></li>
<li class="chapter" data-level="3.2" data-path="phenotype-harmonization.html"><a href="phenotype-harmonization.html#compare-study-values"><i class="fa fa-check"></i><b>3.2</b> Compare study values</a></li>
<li class="chapter" data-level="3.3" data-path="phenotype-harmonization.html"><a href="phenotype-harmonization.html#using-regression-models-to-compare-studies"><i class="fa fa-check"></i><b>3.3</b> Using regression models to compare studies</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="phenotype-harmonization.html"><a href="phenotype-harmonization.html#create-an-annotated-data-frame"><i class="fa fa-check"></i><b>3.3.1</b> Create an Annotated Data Frame</a></li>
<li class="chapter" data-level="3.3.2" data-path="phenotype-harmonization.html"><a href="phenotype-harmonization.html#fit-a-regression-model-without-study"><i class="fa fa-check"></i><b>3.3.2</b> Fit a regression model without study</a></li>
<li class="chapter" data-level="3.3.3" data-path="phenotype-harmonization.html"><a href="phenotype-harmonization.html#fit-a-model-with-study"><i class="fa fa-check"></i><b>3.3.3</b> Fit a model with study</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="phenotype-harmonization.html"><a href="phenotype-harmonization.html#final-considerations"><i class="fa fa-check"></i><b>3.4</b> Final considerations</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="association-tests-part-i.html"><a href="association-tests-part-i.html"><i class="fa fa-check"></i><b>4</b> Association tests - Part I</a>
<ul>
<li class="chapter" data-level="4.1" data-path="association-tests-part-i.html"><a href="association-tests-part-i.html#null-model"><i class="fa fa-check"></i><b>4.1</b> Null model</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="association-tests-part-i.html"><a href="association-tests-part-i.html#prepare-the-data"><i class="fa fa-check"></i><b>4.1.1</b> Prepare the data</a></li>
<li class="chapter" data-level="4.1.2" data-path="association-tests-part-i.html"><a href="association-tests-part-i.html#fit-the-null-model"><i class="fa fa-check"></i><b>4.1.2</b> Fit the null model</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="association-tests-part-i.html"><a href="association-tests-part-i.html#exercise"><i class="fa fa-check"></i><b>4.2</b> Exercise</a></li>
<li class="chapter" data-level="4.3" data-path="association-tests-part-i.html"><a href="association-tests-part-i.html#solution"><i class="fa fa-check"></i><b>4.3</b> Solution</a></li>
<li class="chapter" data-level="4.4" data-path="association-tests-part-i.html"><a href="association-tests-part-i.html#single-variant-association-tests"><i class="fa fa-check"></i><b>4.4</b> Single-variant association tests</a></li>
<li class="chapter" data-level="4.5" data-path="association-tests-part-i.html"><a href="association-tests-part-i.html#exercise-1"><i class="fa fa-check"></i><b>4.5</b> Exercise</a></li>
<li class="chapter" data-level="4.6" data-path="association-tests-part-i.html"><a href="association-tests-part-i.html#solution-1"><i class="fa fa-check"></i><b>4.6</b> Solution</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="association-tests-part-ii.html"><a href="association-tests-part-ii.html"><i class="fa fa-check"></i><b>5</b> Association tests - Part II</a>
<ul>
<li class="chapter" data-level="5.1" data-path="association-tests-part-ii.html"><a href="association-tests-part-ii.html#sliding-window-tests"><i class="fa fa-check"></i><b>5.1</b> Sliding window tests</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="association-tests-part-ii.html"><a href="association-tests-part-ii.html#burden-test"><i class="fa fa-check"></i><b>5.1.1</b> Burden test</a></li>
<li class="chapter" data-level="5.1.2" data-path="association-tests-part-ii.html"><a href="association-tests-part-ii.html#skat-test"><i class="fa fa-check"></i><b>5.1.2</b> SKAT test</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="association-tests-part-ii.html"><a href="association-tests-part-ii.html#exercise-2"><i class="fa fa-check"></i><b>5.2</b> Exercise</a></li>
<li class="chapter" data-level="5.3" data-path="association-tests-part-ii.html"><a href="association-tests-part-ii.html#solution-2"><i class="fa fa-check"></i><b>5.3</b> Solution</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="ancestry-and-relatedness-inference.html"><a href="ancestry-and-relatedness-inference.html"><i class="fa fa-check"></i><b>6</b> Ancestry and Relatedness Inference</a>
<ul>
<li class="chapter" data-level="6.1" data-path="ancestry-and-relatedness-inference.html"><a href="ancestry-and-relatedness-inference.html#ld-pruning"><i class="fa fa-check"></i><b>6.1</b> LD-pruning</a></li>
<li class="chapter" data-level="6.2" data-path="ancestry-and-relatedness-inference.html"><a href="ancestry-and-relatedness-inference.html#computing-a-grm"><i class="fa fa-check"></i><b>6.2</b> Computing a GRM</a></li>
<li class="chapter" data-level="6.3" data-path="ancestry-and-relatedness-inference.html"><a href="ancestry-and-relatedness-inference.html#de-convoluting-ancestry-and-relatedness"><i class="fa fa-check"></i><b>6.3</b> De-convoluting ancestry and relatedness</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="ancestry-and-relatedness-inference.html"><a href="ancestry-and-relatedness-inference.html#king"><i class="fa fa-check"></i><b>6.3.1</b> KING</a></li>
<li class="chapter" data-level="6.3.2" data-path="ancestry-and-relatedness-inference.html"><a href="ancestry-and-relatedness-inference.html#pc-air"><i class="fa fa-check"></i><b>6.3.2</b> PC-AiR</a></li>
<li class="chapter" data-level="6.3.3" data-path="ancestry-and-relatedness-inference.html"><a href="ancestry-and-relatedness-inference.html#pc-relate"><i class="fa fa-check"></i><b>6.3.3</b> PC-Relate</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="ancestry-and-relatedness-inference.html"><a href="ancestry-and-relatedness-inference.html#exercises-1"><i class="fa fa-check"></i><b>6.4</b> Exercises</a></li>
<li class="chapter" data-level="6.5" data-path="ancestry-and-relatedness-inference.html"><a href="ancestry-and-relatedness-inference.html#solutions-1"><i class="fa fa-check"></i><b>6.5</b> Solutions</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="mixed-models.html"><a href="mixed-models.html"><i class="fa fa-check"></i><b>7</b> Mixed models</a>
<ul>
<li class="chapter" data-level="7.1" data-path="mixed-models.html"><a href="mixed-models.html#null-model-1"><i class="fa fa-check"></i><b>7.1</b> Null model</a></li>
<li class="chapter" data-level="7.2" data-path="mixed-models.html"><a href="mixed-models.html#single-variant-tests"><i class="fa fa-check"></i><b>7.2</b> Single-variant tests</a></li>
<li class="chapter" data-level="7.3" data-path="mixed-models.html"><a href="mixed-models.html#exercise-3"><i class="fa fa-check"></i><b>7.3</b> Exercise</a></li>
<li class="chapter" data-level="7.4" data-path="mixed-models.html"><a href="mixed-models.html#solution-3"><i class="fa fa-check"></i><b>7.4</b> Solution</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="variant-annotation.html"><a href="variant-annotation.html"><i class="fa fa-check"></i><b>8</b> Variant annotation</a>
<ul>
<li class="chapter" data-level="8.1" data-path="variant-annotation.html"><a href="variant-annotation.html#use-case-1"><i class="fa fa-check"></i><b>8.1</b> Use case 1</a></li>
<li class="chapter" data-level="8.2" data-path="variant-annotation.html"><a href="variant-annotation.html#use-case-2"><i class="fa fa-check"></i><b>8.2</b> Use case 2</a></li>
<li class="chapter" data-level="8.3" data-path="variant-annotation.html"><a href="variant-annotation.html#use-case-3"><i class="fa fa-check"></i><b>8.3</b> Use case 3</a></li>
<li class="chapter" data-level="8.4" data-path="variant-annotation.html"><a href="variant-annotation.html#exercise-4"><i class="fa fa-check"></i><b>8.4</b> Exercise</a></li>
<li class="chapter" data-level="8.5" data-path="variant-annotation.html"><a href="variant-annotation.html#solution-4"><i class="fa fa-check"></i><b>8.5</b> Solution</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="annotation-informed-aggregate-association-tests.html"><a href="annotation-informed-aggregate-association-tests.html"><i class="fa fa-check"></i><b>9</b> Annotation informed aggregate association tests</a>
<ul>
<li class="chapter" data-level="9.1" data-path="annotation-informed-aggregate-association-tests.html"><a href="annotation-informed-aggregate-association-tests.html#aggregate-unit-for-association-testing-exercise"><i class="fa fa-check"></i><b>9.1</b> Aggregate unit for association testing exercise</a></li>
<li class="chapter" data-level="9.2" data-path="annotation-informed-aggregate-association-tests.html"><a href="annotation-informed-aggregate-association-tests.html#association-testing-with-aggregate-units"><i class="fa fa-check"></i><b>9.2</b> Association testing with aggregate units</a></li>
<li class="chapter" data-level="9.3" data-path="annotation-informed-aggregate-association-tests.html"><a href="annotation-informed-aggregate-association-tests.html#exercise-5"><i class="fa fa-check"></i><b>9.3</b> Exercise</a></li>
<li class="chapter" data-level="9.4" data-path="annotation-informed-aggregate-association-tests.html"><a href="annotation-informed-aggregate-association-tests.html#solution-5"><i class="fa fa-check"></i><b>9.4</b> Solution</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="running-a-gwas-workflow-on-biodata-catalyst.html"><a href="running-a-gwas-workflow-on-biodata-catalyst.html"><i class="fa fa-check"></i><b>10</b> Running a GWAS workflow on BioData Catalyst</a>
<ul>
<li class="chapter" data-level="10.1" data-path="running-a-gwas-workflow-on-biodata-catalyst.html"><a href="running-a-gwas-workflow-on-biodata-catalyst.html#preparing-the-genotype-data"><i class="fa fa-check"></i><b>10.1</b> Preparing the genotype data</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="running-a-gwas-workflow-on-biodata-catalyst.html"><a href="running-a-gwas-workflow-on-biodata-catalyst.html#bcftools-merge-and-filter"><i class="fa fa-check"></i><b>10.1.1</b> Bcftools Merge and Filter</a></li>
<li class="chapter" data-level="10.1.2" data-path="running-a-gwas-workflow-on-biodata-catalyst.html"><a href="running-a-gwas-workflow-on-biodata-catalyst.html#vcf-to-gds-converter"><i class="fa fa-check"></i><b>10.1.2</b> VCF to GDS Converter</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="running-a-gwas-workflow-on-biodata-catalyst.html"><a href="running-a-gwas-workflow-on-biodata-catalyst.html#ancestry-and-relatedness"><i class="fa fa-check"></i><b>10.2</b> Ancestry and Relatedness</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="running-a-gwas-workflow-on-biodata-catalyst.html"><a href="running-a-gwas-workflow-on-biodata-catalyst.html#king-robust"><i class="fa fa-check"></i><b>10.2.1</b> KING robust</a></li>
<li class="chapter" data-level="10.2.2" data-path="running-a-gwas-workflow-on-biodata-catalyst.html"><a href="running-a-gwas-workflow-on-biodata-catalyst.html#pc-air-1"><i class="fa fa-check"></i><b>10.2.2</b> PC-AiR</a></li>
<li class="chapter" data-level="10.2.3" data-path="running-a-gwas-workflow-on-biodata-catalyst.html"><a href="running-a-gwas-workflow-on-biodata-catalyst.html#pc-relate-1"><i class="fa fa-check"></i><b>10.2.3</b> PC-Relate</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="running-a-gwas-workflow-on-biodata-catalyst.html"><a href="running-a-gwas-workflow-on-biodata-catalyst.html#association-testing"><i class="fa fa-check"></i><b>10.3</b> Association Testing</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="running-a-gwas-workflow-on-biodata-catalyst.html"><a href="running-a-gwas-workflow-on-biodata-catalyst.html#null-model-2"><i class="fa fa-check"></i><b>10.3.1</b> Null Model</a></li>
<li class="chapter" data-level="10.3.2" data-path="running-a-gwas-workflow-on-biodata-catalyst.html"><a href="running-a-gwas-workflow-on-biodata-catalyst.html#single-variant-association-test"><i class="fa fa-check"></i><b>10.3.2</b> Single variant association test</a></li>
<li class="chapter" data-level="10.3.3" data-path="running-a-gwas-workflow-on-biodata-catalyst.html"><a href="running-a-gwas-workflow-on-biodata-catalyst.html#aggregate-variant-test"><i class="fa fa-check"></i><b>10.3.3</b> Aggregate variant test</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="running-a-gwas-workflow-on-biodata-catalyst.html"><a href="running-a-gwas-workflow-on-biodata-catalyst.html#analysis-follow-up"><i class="fa fa-check"></i><b>10.4</b> Analysis follow-up</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">SISG 2021 Module 16: Computational Pipeline for WGS Data</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="ancestry-and-relatedness-inference" class="section level1" number="6">
<h1><span class="header-section-number">6</span> Ancestry and Relatedness Inference</h1>
<div id="ld-pruning" class="section level2" number="6.1">
<h2><span class="header-section-number">6.1</span> LD-pruning</h2>
<p>We generally advise that population structure and relatedness inference be performed using a set of (nearly) independent genetic variants. To find this set of variants, we perform linkage-disequilibrium (LD) pruning on the study sample set. We typically use an LD threshold of <code>r^2 &lt; 0.1</code> to select variants.</p>
<div class="sourceCode" id="cb228"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb228-1"><a href="ancestry-and-relatedness-inference.html#cb228-1"></a><span class="kw">library</span>(SeqArray)</span>
<span id="cb228-2"><a href="ancestry-and-relatedness-inference.html#cb228-2"></a>repo_path &lt;-<span class="st"> &quot;https://github.com/UW-GAC/SISG_2021/raw/master&quot;</span></span>
<span id="cb228-3"><a href="ancestry-and-relatedness-inference.html#cb228-3"></a></span>
<span id="cb228-4"><a href="ancestry-and-relatedness-inference.html#cb228-4"></a><span class="co"># use a GDS file with all chromosomes</span></span>
<span id="cb228-5"><a href="ancestry-and-relatedness-inference.html#cb228-5"></a><span class="cf">if</span> (<span class="op">!</span><span class="kw">dir.exists</span>(<span class="st">&quot;data&quot;</span>)) <span class="kw">dir.create</span>(<span class="st">&quot;data&quot;</span>)</span>
<span id="cb228-6"><a href="ancestry-and-relatedness-inference.html#cb228-6"></a>gdsfile &lt;-<span class="st"> &quot;data/1KG_phase3_subset.gds&quot;</span></span>
<span id="cb228-7"><a href="ancestry-and-relatedness-inference.html#cb228-7"></a><span class="cf">if</span> (<span class="op">!</span><span class="kw">file.exists</span>(gdsfile)) <span class="kw">download.file</span>(<span class="kw">file.path</span>(repo_path, gdsfile), gdsfile)</span>
<span id="cb228-8"><a href="ancestry-and-relatedness-inference.html#cb228-8"></a>gdsfmt<span class="op">::</span><span class="kw">showfile.gds</span>(<span class="dt">closeall=</span><span class="ot">TRUE</span>) <span class="co"># make sure file is not already open</span></span>
<span id="cb228-9"><a href="ancestry-and-relatedness-inference.html#cb228-9"></a>gds &lt;-<span class="st"> </span><span class="kw">seqOpen</span>(gdsfile)</span>
<span id="cb228-10"><a href="ancestry-and-relatedness-inference.html#cb228-10"></a></span>
<span id="cb228-11"><a href="ancestry-and-relatedness-inference.html#cb228-11"></a><span class="co"># run LD pruning</span></span>
<span id="cb228-12"><a href="ancestry-and-relatedness-inference.html#cb228-12"></a><span class="kw">library</span>(SNPRelate)</span>
<span id="cb228-13"><a href="ancestry-and-relatedness-inference.html#cb228-13"></a><span class="kw">set.seed</span>(<span class="dv">100</span>) <span class="co"># LD pruning has a random element; so make this reproducible</span></span>
<span id="cb228-14"><a href="ancestry-and-relatedness-inference.html#cb228-14"></a>snpset &lt;-<span class="st"> </span><span class="kw">snpgdsLDpruning</span>(gds, </span>
<span id="cb228-15"><a href="ancestry-and-relatedness-inference.html#cb228-15"></a>                          <span class="dt">method=</span><span class="st">&quot;corr&quot;</span>, </span>
<span id="cb228-16"><a href="ancestry-and-relatedness-inference.html#cb228-16"></a>                          <span class="dt">slide.max.bp=</span><span class="fl">10e6</span>, </span>
<span id="cb228-17"><a href="ancestry-and-relatedness-inference.html#cb228-17"></a>                          <span class="dt">ld.threshold=</span><span class="kw">sqrt</span>(<span class="fl">0.1</span>))</span></code></pre></div>
<pre><code>## SNV pruning based on LD:
## Excluding 1,120 SNVs on non-autosomes
## Calculating allele counts/frequencies ...
## 
[..................................................]  0%, ETC: ---    
[==================================================] 100%, completed, 1s
## # of selected variants: 24,387
## Excluding 253 SNVs (monomorphic: TRUE, MAF: NaN, missing rate: NaN)
##     # of samples: 1,126
##     # of SNVs: 24,387
##     using 1 thread
##     sliding window: 10,000,000 basepairs, Inf SNPs
##     |LD| threshold: 0.316228
##     method: correlation
## Chromosome 1: 92.23%, 1,033/1,120
## Chromosome 2: 91.16%, 1,021/1,120
## Chromosome 3: 91.16%, 1,021/1,120
## Chromosome 4: 91.16%, 1,021/1,120
## Chromosome 5: 91.88%, 1,029/1,120
## Chromosome 6: 89.02%, 997/1,120
## Chromosome 7: 90.54%, 1,014/1,120
## Chromosome 8: 89.02%, 997/1,120
## Chromosome 9: 88.66%, 993/1,120
## Chromosome 10: 86.88%, 973/1,120
## Chromosome 11: 86.79%, 972/1,120
## Chromosome 12: 88.30%, 989/1,120
## Chromosome 13: 86.34%, 967/1,120
## Chromosome 14: 84.46%, 946/1,120
## Chromosome 15: 84.11%, 942/1,120
## Chromosome 16: 83.93%, 940/1,120
## Chromosome 17: 84.91%, 951/1,120
## Chromosome 18: 84.46%, 946/1,120
## Chromosome 19: 81.96%, 918/1,120
## Chromosome 20: 82.05%, 919/1,120
## Chromosome 21: 75.09%, 841/1,120
## Chromosome 22: 76.52%, 857/1,120
## 21,287 markers are selected in total.</code></pre>
<div class="sourceCode" id="cb230"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb230-1"><a href="ancestry-and-relatedness-inference.html#cb230-1"></a><span class="co"># how many variants on each chr?</span></span>
<span id="cb230-2"><a href="ancestry-and-relatedness-inference.html#cb230-2"></a><span class="kw">sapply</span>(snpset, length)</span></code></pre></div>
<pre><code>##  chr1  chr2  chr3  chr4  chr5  chr6  chr7  chr8  chr9 chr10 chr11 chr12 chr13 
##  1033  1021  1021  1021  1029   997  1014   997   993   973   972   989   967 
## chr14 chr15 chr16 chr17 chr18 chr19 chr20 chr21 chr22 
##   946   942   940   951   946   918   919   841   857</code></pre>
<div class="sourceCode" id="cb232"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb232-1"><a href="ancestry-and-relatedness-inference.html#cb232-1"></a><span class="co"># get the full list of LD-pruned variants </span></span>
<span id="cb232-2"><a href="ancestry-and-relatedness-inference.html#cb232-2"></a>pruned &lt;-<span class="st"> </span><span class="kw">unlist</span>(snpset, <span class="dt">use.names=</span><span class="ot">FALSE</span>)</span>
<span id="cb232-3"><a href="ancestry-and-relatedness-inference.html#cb232-3"></a><span class="kw">length</span>(pruned)</span></code></pre></div>
<pre><code>## [1] 21287</code></pre>
</div>
<div id="computing-a-grm" class="section level2" number="6.2">
<h2><span class="header-section-number">6.2</span> Computing a GRM</h2>
<p>We can use the <a href="https://github.com/zhengxwen/SNPRelate">SNPRelate package</a> to compute a Genetic Relationship matrix (GRM). A GRM captures genetic relatedness due to both distant ancestry (i.e. population structure) and recent kinship (i.e. family structure) in a single matrix.</p>
<p>SNPRelate offers several algorithms for computing a GRM, including the commonly-used GCTA <a href="https://www.ncbi.nlm.nih.gov/pubmed/21167468">Yang et al 2011</a>. The most recent algorithm added to the package is “IndivBeta” <a href="https://www.ncbi.nlm.nih.gov/pubmed/28550018">Weir and Goudet 2017</a>.</p>
<div class="sourceCode" id="cb234"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb234-1"><a href="ancestry-and-relatedness-inference.html#cb234-1"></a><span class="co"># compute the GRM</span></span>
<span id="cb234-2"><a href="ancestry-and-relatedness-inference.html#cb234-2"></a><span class="kw">library</span>(SNPRelate)</span>
<span id="cb234-3"><a href="ancestry-and-relatedness-inference.html#cb234-3"></a>grm &lt;-<span class="st"> </span><span class="kw">snpgdsGRM</span>(gds, <span class="dt">method=</span><span class="st">&quot;GCTA&quot;</span>, <span class="dt">snp.id =</span> pruned)</span></code></pre></div>
<pre><code>## Genetic Relationship Matrix (GRM, GCTA):
## Calculating allele counts/frequencies ...
## 
[..................................................]  0%, ETC: ---    
[==================================================] 100%, completed, 0s
## # of selected variants: 21,287
##     # of samples: 1,126
##     # of SNVs: 21,287
##     using 1 thread
## CPU capabilities: Double-Precision SSE2
## Tue Jul 20 14:19:28 2021    (internal increment: 316)
## 
[..................................................]  0%, ETC: ---        
[==================================================] 100%, completed, 11s
## Tue Jul 20 14:19:39 2021    Done.</code></pre>
<div class="sourceCode" id="cb236"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb236-1"><a href="ancestry-and-relatedness-inference.html#cb236-1"></a><span class="kw">names</span>(grm)</span></code></pre></div>
<pre><code>## [1] &quot;sample.id&quot; &quot;snp.id&quot;    &quot;method&quot;    &quot;grm&quot;</code></pre>
<div class="sourceCode" id="cb238"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb238-1"><a href="ancestry-and-relatedness-inference.html#cb238-1"></a><span class="kw">dim</span>(grm<span class="op">$</span>grm)</span></code></pre></div>
<pre><code>## [1] 1126 1126</code></pre>
<div class="sourceCode" id="cb240"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb240-1"><a href="ancestry-and-relatedness-inference.html#cb240-1"></a><span class="co"># look at the top corner of the matrix</span></span>
<span id="cb240-2"><a href="ancestry-and-relatedness-inference.html#cb240-2"></a>grm<span class="op">$</span>grm[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>##            [,1]       [,2]       [,3]       [,4]       [,5]
## [1,] 0.72284858 0.03428003 0.02475889 0.04051960 0.02397084
## [2,] 0.03428003 0.74335217 0.02098193 0.02772272 0.02125452
## [3,] 0.02475889 0.02098193 0.48389966 0.03247437 0.03382292
## [4,] 0.04051960 0.02772272 0.03247437 0.80364821 0.02771950
## [5,] 0.02397084 0.02125452 0.03382292 0.02771950 0.47465797</code></pre>
</div>
<div id="de-convoluting-ancestry-and-relatedness" class="section level2" number="6.3">
<h2><span class="header-section-number">6.3</span> De-convoluting ancestry and relatedness</h2>
<p>To disentangle distant ancestry (i.e. population structure) from recent kinship (i.e. familial relatedness), we implement the analysis described in <a href="https://www.cell.com/ajhg/fulltext/S0002-9297(15)00496-6">Conomos et al., 2016</a>. This approach uses the <a href="http://www.ncbi.nlm.nih.gov/pubmed/20926424">KING</a>, <a href="http://www.ncbi.nlm.nih.gov/pubmed/25810074">PC-AiR</a>, and <a href="http://www.ncbi.nlm.nih.gov/pubmed/26748516">PC-Relate</a> methods.</p>
<div id="king" class="section level3" number="6.3.1">
<h3><span class="header-section-number">6.3.1</span> KING</h3>
<p>Step 1 is to get initial kinship estimates using <a href="http://www.ncbi.nlm.nih.gov/pubmed/20926424">KING-robust</a>, which is robust to discrete population structure but not ancestry admixture. KING-robust will be able to identify close relatives (e.g. 1st and 2nd degree) reliably, but may identify spurious pairs or miss more distant pairs of relatives in the presence of admixture. KING is available as its own software, but the KING-robust algorithm is also available in SNPRelate.</p>
<div class="sourceCode" id="cb242"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb242-1"><a href="ancestry-and-relatedness-inference.html#cb242-1"></a><span class="co"># run KING-robust</span></span>
<span id="cb242-2"><a href="ancestry-and-relatedness-inference.html#cb242-2"></a>king &lt;-<span class="st"> </span><span class="kw">snpgdsIBDKING</span>(gds, <span class="dt">snp.id=</span>pruned)</span></code></pre></div>
<pre><code>## IBD analysis (KING method of moment) on genotypes:
## Calculating allele counts/frequencies ...
## 
[..................................................]  0%, ETC: ---    
[==================================================] 100%, completed, 0s
## # of selected variants: 21,287
##     # of samples: 1,126
##     # of SNVs: 21,287
##     using 1 thread
## No family is specified, and all individuals are treated as singletons.
## Relationship inference in the presence of population stratification.
## CPU capabilities: Double-Precision SSE2
## Tue Jul 20 14:19:40 2021    (internal increment: 10112)
## 
[..................................................]  0%, ETC: ---        
[==================================================] 100%, completed, 3s
## Tue Jul 20 14:19:43 2021    Done.</code></pre>
<div class="sourceCode" id="cb244"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb244-1"><a href="ancestry-and-relatedness-inference.html#cb244-1"></a><span class="kw">names</span>(king)</span></code></pre></div>
<pre><code>## [1] &quot;sample.id&quot; &quot;snp.id&quot;    &quot;afreq&quot;     &quot;IBS0&quot;      &quot;kinship&quot;</code></pre>
<div class="sourceCode" id="cb246"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb246-1"><a href="ancestry-and-relatedness-inference.html#cb246-1"></a><span class="kw">dim</span>(king<span class="op">$</span>kinship)</span></code></pre></div>
<pre><code>## [1] 1126 1126</code></pre>
<div class="sourceCode" id="cb248"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb248-1"><a href="ancestry-and-relatedness-inference.html#cb248-1"></a>kingMat &lt;-<span class="st"> </span>king<span class="op">$</span>kinship</span>
<span id="cb248-2"><a href="ancestry-and-relatedness-inference.html#cb248-2"></a><span class="kw">colnames</span>(kingMat) &lt;-<span class="st"> </span><span class="kw">rownames</span>(kingMat) &lt;-<span class="st"> </span>king<span class="op">$</span>sample.id</span>
<span id="cb248-3"><a href="ancestry-and-relatedness-inference.html#cb248-3"></a></span>
<span id="cb248-4"><a href="ancestry-and-relatedness-inference.html#cb248-4"></a><span class="co"># look at the top corner of the matrix</span></span>
<span id="cb248-5"><a href="ancestry-and-relatedness-inference.html#cb248-5"></a>kingMat[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>##             HG00096      HG00097     HG00099      HG00100     HG00101
## HG00096 0.500000000  0.013295347 0.021206409  0.003261883 0.021668220
## HG00097 0.013295347  0.500000000 0.007122507 -0.009021842 0.003561254
## HG00099 0.021206409  0.007122507 0.500000000  0.037700283 0.020028275
## HG00100 0.003261883 -0.009021842 0.037700283  0.500000000 0.016992551
## HG00101 0.021668220  0.003561254 0.020028275  0.016992551 0.500000000</code></pre>
<p>We extract pairwise kinship estimates and IBS0 values (the proportion of variants for which the pair of indivdiuals share 0 alleles identical by state) to plot. We use a hexbin plot to visualize the relatedness for all pairs of samples.</p>
<div class="sourceCode" id="cb250"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb250-1"><a href="ancestry-and-relatedness-inference.html#cb250-1"></a>kinship &lt;-<span class="st"> </span><span class="kw">snpgdsIBDSelection</span>(king)</span>
<span id="cb250-2"><a href="ancestry-and-relatedness-inference.html#cb250-2"></a><span class="kw">head</span>(kinship)</span></code></pre></div>
<pre><code>##       ID1     ID2        IBS0      kinship
## 1 HG00096 HG00097 0.007986095  0.013295347
## 2 HG00096 HG00099 0.007845164  0.021206409
## 3 HG00096 HG00100 0.008831681  0.003261883
## 4 HG00096 HG00101 0.007939118  0.021668220
## 5 HG00096 HG00102 0.008737727 -0.003362152
## 6 HG00096 HG00103 0.007939118  0.026241799</code></pre>
<div class="sourceCode" id="cb252"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb252-1"><a href="ancestry-and-relatedness-inference.html#cb252-1"></a><span class="kw">library</span>(ggplot2)</span>
<span id="cb252-2"><a href="ancestry-and-relatedness-inference.html#cb252-2"></a><span class="kw">ggplot</span>(kinship, <span class="kw">aes</span>(IBS0, kinship)) <span class="op">+</span></span>
<span id="cb252-3"><a href="ancestry-and-relatedness-inference.html#cb252-3"></a><span class="st">    </span><span class="kw">geom_hline</span>(<span class="dt">yintercept=</span><span class="dv">2</span><span class="op">^</span>(<span class="op">-</span><span class="kw">seq</span>(<span class="dv">3</span>,<span class="dv">9</span>,<span class="dv">2</span>)<span class="op">/</span><span class="dv">2</span>), <span class="dt">linetype=</span><span class="st">&quot;dashed&quot;</span>, <span class="dt">color=</span><span class="st">&quot;grey&quot;</span>) <span class="op">+</span></span>
<span id="cb252-4"><a href="ancestry-and-relatedness-inference.html#cb252-4"></a><span class="st">    </span><span class="kw">geom_hex</span>(<span class="dt">bins =</span> <span class="dv">100</span>) <span class="op">+</span></span>
<span id="cb252-5"><a href="ancestry-and-relatedness-inference.html#cb252-5"></a><span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;kinship estimate&quot;</span>) <span class="op">+</span></span>
<span id="cb252-6"><a href="ancestry-and-relatedness-inference.html#cb252-6"></a><span class="st">    </span><span class="kw">theme_bw</span>()</span></code></pre></div>
<p><img src="_main_files/figure-html/king_plot-1.png" width="672" /></p>
<p>We see a few parent-offspring, full sibling, 2nd degree, and 3rd degree relative pairs. The abundance of negative estimates represent pairs of individuals who have ancestry from different populations – the magnitude of the negative relationship is informative of how different their ancestries are; more on this below.</p>
</div>
<div id="pc-air" class="section level3" number="6.3.2">
<h3><span class="header-section-number">6.3.2</span> PC-AiR</h3>
<p>The next step is <a href="http://www.ncbi.nlm.nih.gov/pubmed/25810074">PC-AiR</a>, which provides robust population structure inference in samples with kinship and pedigree structure. PC-AiR is available in the GENESIS package via the function <code>pcair</code>.</p>
<p>First, PC-AiR partitions the full sample set into a set of mutually unrelated samples that is maximally informative about all ancestries in the sample (i.e. the unrelated set) and their relatives (i.e. the related set). We use a 3rd degree kinship threshold (<code>kin.thresh = 2^(-9/2)</code>), which corresponds to first cousins – this defines anyone less related than first cousins as “unrelated”. We use the negative KING-robust estimates as “ancestry divergence” measures (<code>divMat</code>) to identify pairs of samples with different ancestry – we preferentially select individuals with many negative estimates for the unrelated set to ensure ancestry representation. For now, we also use the KING-robust estimates as our kinship measures (<code>kinMat</code>); more on this below.</p>
<p>Once the unrelated and related sets are identified, PC-AiR performs a standard Principal Component Analysis (PCA) on the unrelated set of individuals and then projects the relatives onto the PCs. Under the hood, PC-AiR uses the SNPRelate package for efficient PC computation and projection.</p>
<div class="sourceCode" id="cb253"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb253-1"><a href="ancestry-and-relatedness-inference.html#cb253-1"></a><span class="co"># run PC-AiR</span></span>
<span id="cb253-2"><a href="ancestry-and-relatedness-inference.html#cb253-2"></a><span class="kw">library</span>(GENESIS)</span>
<span id="cb253-3"><a href="ancestry-and-relatedness-inference.html#cb253-3"></a>pca &lt;-<span class="st"> </span><span class="kw">pcair</span>(gds, </span>
<span id="cb253-4"><a href="ancestry-and-relatedness-inference.html#cb253-4"></a>            <span class="dt">kinobj =</span> kingMat,</span>
<span id="cb253-5"><a href="ancestry-and-relatedness-inference.html#cb253-5"></a>            <span class="dt">kin.thresh=</span><span class="dv">2</span><span class="op">^</span>(<span class="op">-</span><span class="dv">9</span><span class="op">/</span><span class="dv">2</span>),</span>
<span id="cb253-6"><a href="ancestry-and-relatedness-inference.html#cb253-6"></a>            <span class="dt">divobj =</span> kingMat,</span>
<span id="cb253-7"><a href="ancestry-and-relatedness-inference.html#cb253-7"></a>            <span class="dt">div.thresh=</span><span class="op">-</span><span class="dv">2</span><span class="op">^</span>(<span class="op">-</span><span class="dv">9</span><span class="op">/</span><span class="dv">2</span>))</span></code></pre></div>
<pre><code>## Principal Component Analysis (PCA) on genotypes:
## Excluding 1,120 SNVs on non-autosomes
## Calculating allele counts/frequencies ...
## 
[..................................................]  0%, ETC: ---    
[==================================================] 100%, completed, 0s
## # of selected variants: 23,779
## Excluding 861 SNVs (monomorphic: TRUE, MAF: NaN, missing rate: NaN)
##     # of samples: 1,040
##     # of SNVs: 23,779
##     using 1 thread
##     # of principal components: 32
## CPU capabilities: Double-Precision SSE2
## Tue Jul 20 14:19:52 2021    (internal increment: 340)
## 
[..................................................]  0%, ETC: ---        
[==================================================] 100%, completed, 12s
## Tue Jul 20 14:20:04 2021    Begin (eigenvalues and eigenvectors)
## Tue Jul 20 14:20:05 2021    Done.
## SNP Loading:
##     # of samples: 1,040
##     # of SNPs: 23,779
##     using 1 thread
##     using the top 32 eigenvectors
## Tue Jul 20 14:20:05 2021    (internal increment: 2740)
## 
[..................................................]  0%, ETC: ---        
[==================================================] 100%, completed, 1s
## Tue Jul 20 14:20:06 2021    Done.
## Sample Loading:
##     # of samples: 86
##     # of SNPs: 23,779
##     using 1 thread
##     using the top 32 eigenvectors
## Tue Jul 20 14:20:06 2021    (internal increment: 33148)
## 
[..................................................]  0%, ETC: ---        
[==================================================] 100%, completed, 1s
## Tue Jul 20 14:20:07 2021    Done.</code></pre>
<div class="sourceCode" id="cb255"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb255-1"><a href="ancestry-and-relatedness-inference.html#cb255-1"></a><span class="kw">names</span>(pca)</span></code></pre></div>
<pre><code>##  [1] &quot;vectors&quot;    &quot;values&quot;     &quot;rels&quot;       &quot;unrels&quot;     &quot;kin.thresh&quot;
##  [6] &quot;div.thresh&quot; &quot;sample.id&quot;  &quot;nsamp&quot;      &quot;nsnps&quot;      &quot;varprop&quot;   
## [11] &quot;call&quot;       &quot;method&quot;</code></pre>
<div class="sourceCode" id="cb257"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb257-1"><a href="ancestry-and-relatedness-inference.html#cb257-1"></a><span class="co"># the unrelated set of samples</span></span>
<span id="cb257-2"><a href="ancestry-and-relatedness-inference.html#cb257-2"></a><span class="kw">length</span>(pca<span class="op">$</span>unrels)</span></code></pre></div>
<pre><code>## [1] 1040</code></pre>
<div class="sourceCode" id="cb259"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb259-1"><a href="ancestry-and-relatedness-inference.html#cb259-1"></a><span class="kw">head</span>(pca<span class="op">$</span>unrels)</span></code></pre></div>
<pre><code>## [1] &quot;HG00096&quot; &quot;HG00097&quot; &quot;HG00099&quot; &quot;HG00100&quot; &quot;HG00101&quot; &quot;HG00102&quot;</code></pre>
<div class="sourceCode" id="cb261"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb261-1"><a href="ancestry-and-relatedness-inference.html#cb261-1"></a><span class="co"># the related set of samples</span></span>
<span id="cb261-2"><a href="ancestry-and-relatedness-inference.html#cb261-2"></a><span class="kw">length</span>(pca<span class="op">$</span>rels)</span></code></pre></div>
<pre><code>## [1] 86</code></pre>
<div class="sourceCode" id="cb263"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb263-1"><a href="ancestry-and-relatedness-inference.html#cb263-1"></a><span class="kw">head</span>(pca<span class="op">$</span>rels)</span></code></pre></div>
<pre><code>## [1] &quot;NA18553&quot; &quot;NA19010&quot; &quot;HG00240&quot; &quot;HG01958&quot; &quot;NA18974&quot; &quot;NA19720&quot;</code></pre>
<div class="sourceCode" id="cb265"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb265-1"><a href="ancestry-and-relatedness-inference.html#cb265-1"></a><span class="co"># extract the top 10 PCs and make a data.frame</span></span>
<span id="cb265-2"><a href="ancestry-and-relatedness-inference.html#cb265-2"></a>pcs &lt;-<span class="st"> </span><span class="kw">data.frame</span>(pca<span class="op">$</span>vectors[,<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>])</span>
<span id="cb265-3"><a href="ancestry-and-relatedness-inference.html#cb265-3"></a><span class="kw">colnames</span>(pcs) &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&#39;PC&#39;</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">10</span>)</span>
<span id="cb265-4"><a href="ancestry-and-relatedness-inference.html#cb265-4"></a>pcs<span class="op">$</span>sample.id &lt;-<span class="st"> </span>pca<span class="op">$</span>sample.id</span>
<span id="cb265-5"><a href="ancestry-and-relatedness-inference.html#cb265-5"></a><span class="kw">dim</span>(pcs)</span></code></pre></div>
<pre><code>## [1] 1126   11</code></pre>
<div class="sourceCode" id="cb267"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb267-1"><a href="ancestry-and-relatedness-inference.html#cb267-1"></a><span class="kw">head</span>(pcs)</span></code></pre></div>
<pre><code>##                 PC1         PC2         PC3          PC4           PC5
## HG00096 -0.02226416 -0.03706223 0.007784659 -0.006036777 -0.0002494770
## HG00097 -0.02069118 -0.03316550 0.010648323 -0.006725514  0.0004179001
## HG00099 -0.02102452 -0.03547247 0.009881201 -0.005776697  0.0005220132
## HG00100 -0.02120448 -0.04293781 0.014368290 -0.006435646  0.0004269673
## HG00101 -0.02108916 -0.03594252 0.010240805 -0.003099693  0.0012105916
## HG00102 -0.02209371 -0.04086700 0.013113725 -0.001004959 -0.0014970887
##                   PC6         PC7           PC8           PC9          PC10
## HG00096  0.0008316496 -0.03953455  6.079733e-05  0.0009154511  8.333681e-04
## HG00097 -0.0005008697 -0.03380661 -1.743989e-03 -0.0012662487 -2.869483e-03
## HG00099  0.0011478104 -0.03409698  6.145952e-07  0.0010956673 -2.258131e-04
## HG00100  0.0025670278 -0.04708289  1.140219e-03 -0.0007786959 -8.237878e-04
## HG00101  0.0028715482 -0.03505874  4.599234e-04  0.0068621094 -1.783242e-03
## HG00102 -0.0008704542 -0.03083835 -1.097275e-03  0.0030383585  6.466877e-05
##         sample.id
## HG00096   HG00096
## HG00097   HG00097
## HG00099   HG00099
## HG00100   HG00100
## HG00101   HG00101
## HG00102   HG00102</code></pre>
<p>We’d like to determine which PCs are ancestry informative. To do this we look at the PCs in conjunction with population information for the 1000 Genomes samples. This information is stored in an <code>AnnotatedDataFrame</code>. We make a parallel coordinates plot, color-coding by 1000 Genomes population.</p>
<div class="sourceCode" id="cb269"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb269-1"><a href="ancestry-and-relatedness-inference.html#cb269-1"></a><span class="kw">library</span>(Biobase)</span>
<span id="cb269-2"><a href="ancestry-and-relatedness-inference.html#cb269-2"></a>sampfile &lt;-<span class="st"> &quot;data/sample_annotation.RData&quot;</span></span>
<span id="cb269-3"><a href="ancestry-and-relatedness-inference.html#cb269-3"></a><span class="cf">if</span> (<span class="op">!</span><span class="kw">file.exists</span>(sampfile)) <span class="kw">download.file</span>(<span class="kw">file.path</span>(repo_path, sampfile), sampfile)</span>
<span id="cb269-4"><a href="ancestry-and-relatedness-inference.html#cb269-4"></a>annot &lt;-<span class="st"> </span><span class="kw">get</span>(<span class="kw">load</span>(sampfile))</span>
<span id="cb269-5"><a href="ancestry-and-relatedness-inference.html#cb269-5"></a></span>
<span id="cb269-6"><a href="ancestry-and-relatedness-inference.html#cb269-6"></a><span class="kw">library</span>(dplyr)</span>
<span id="cb269-7"><a href="ancestry-and-relatedness-inference.html#cb269-7"></a>annot &lt;-<span class="st"> </span><span class="kw">pData</span>(annot) <span class="op">%&gt;%</span></span>
<span id="cb269-8"><a href="ancestry-and-relatedness-inference.html#cb269-8"></a><span class="st">        </span>dplyr<span class="op">::</span><span class="kw">select</span>(sample.id, Population)</span>
<span id="cb269-9"><a href="ancestry-and-relatedness-inference.html#cb269-9"></a>pc.df &lt;-<span class="st"> </span><span class="kw">left_join</span>(pcs, annot, <span class="dt">by=</span><span class="st">&quot;sample.id&quot;</span>)</span>
<span id="cb269-10"><a href="ancestry-and-relatedness-inference.html#cb269-10"></a></span>
<span id="cb269-11"><a href="ancestry-and-relatedness-inference.html#cb269-11"></a><span class="kw">library</span>(GGally)</span>
<span id="cb269-12"><a href="ancestry-and-relatedness-inference.html#cb269-12"></a><span class="kw">library</span>(RColorBrewer)</span>
<span id="cb269-13"><a href="ancestry-and-relatedness-inference.html#cb269-13"></a>pop.cols &lt;-<span class="st"> </span><span class="kw">setNames</span>(<span class="kw">brewer.pal</span>(<span class="dv">12</span>, <span class="st">&quot;Paired&quot;</span>),</span>
<span id="cb269-14"><a href="ancestry-and-relatedness-inference.html#cb269-14"></a>                 <span class="kw">c</span>(<span class="st">&quot;ACB&quot;</span>, <span class="st">&quot;ASW&quot;</span>, <span class="st">&quot;CEU&quot;</span>, <span class="st">&quot;GBR&quot;</span>, <span class="st">&quot;CHB&quot;</span>, <span class="st">&quot;JPT&quot;</span>, <span class="st">&quot;CLM&quot;</span>, <span class="st">&quot;MXL&quot;</span>, <span class="st">&quot;LWK&quot;</span>, <span class="st">&quot;YRI&quot;</span>, <span class="st">&quot;GIH&quot;</span>, <span class="st">&quot;PUR&quot;</span>))</span>
<span id="cb269-15"><a href="ancestry-and-relatedness-inference.html#cb269-15"></a><span class="kw">ggparcoord</span>(pc.df, <span class="dt">columns=</span><span class="dv">1</span><span class="op">:</span><span class="dv">10</span>, <span class="dt">groupColumn=</span><span class="st">&quot;Population&quot;</span>, <span class="dt">scale=</span><span class="st">&quot;uniminmax&quot;</span>) <span class="op">+</span></span>
<span id="cb269-16"><a href="ancestry-and-relatedness-inference.html#cb269-16"></a><span class="st">    </span><span class="kw">scale_color_manual</span>(<span class="dt">values=</span>pop.cols) <span class="op">+</span></span>
<span id="cb269-17"><a href="ancestry-and-relatedness-inference.html#cb269-17"></a><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;PC&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/pcair1_parcoord-1.png" width="672" /></p>
</div>
<div id="pc-relate" class="section level3" number="6.3.3">
<h3><span class="header-section-number">6.3.3</span> PC-Relate</h3>
<p>The next step is <a href="http://www.ncbi.nlm.nih.gov/pubmed/26748516">PC-Relate</a>, which provides accurate kinship inference, even in the presence of population structure and ancestry admixture, by conditioning on ancestry informative PCs. As we saw above, the first 4 PCs separate populations in our study, so we condition on PCs 1-4 in our PC-Relate analysis. PC-Relate can be performed using the <code>pcrelate</code> function in GENESIS, which expects a <code>SeqVarIterator</code> object for the genotype data. The <code>training.set</code> argument allows for specification of which samples to use to “learn” the ancestry adjustment – we recommend the unrelated set from the PC-AiR analysis.</p>
<p>(NOTE: this will take a few minutes to run).</p>
<div class="sourceCode" id="cb270"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb270-1"><a href="ancestry-and-relatedness-inference.html#cb270-1"></a><span class="kw">seqResetFilter</span>(gds, <span class="dt">verbose=</span><span class="ot">FALSE</span>)</span>
<span id="cb270-2"><a href="ancestry-and-relatedness-inference.html#cb270-2"></a><span class="kw">library</span>(SeqVarTools)</span>
<span id="cb270-3"><a href="ancestry-and-relatedness-inference.html#cb270-3"></a>seqData &lt;-<span class="st"> </span><span class="kw">SeqVarData</span>(gds)</span>
<span id="cb270-4"><a href="ancestry-and-relatedness-inference.html#cb270-4"></a></span>
<span id="cb270-5"><a href="ancestry-and-relatedness-inference.html#cb270-5"></a><span class="co"># filter the GDS object to our LD-pruned variants</span></span>
<span id="cb270-6"><a href="ancestry-and-relatedness-inference.html#cb270-6"></a><span class="kw">seqSetFilter</span>(seqData, <span class="dt">variant.id=</span>pruned)</span></code></pre></div>
<pre><code>## # of selected variants: 21,287</code></pre>
<div class="sourceCode" id="cb272"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb272-1"><a href="ancestry-and-relatedness-inference.html#cb272-1"></a>iterator &lt;-<span class="st"> </span><span class="kw">SeqVarBlockIterator</span>(seqData, <span class="dt">verbose=</span><span class="ot">FALSE</span>)</span>
<span id="cb272-2"><a href="ancestry-and-relatedness-inference.html#cb272-2"></a></span>
<span id="cb272-3"><a href="ancestry-and-relatedness-inference.html#cb272-3"></a>pcrel &lt;-<span class="st"> </span><span class="kw">pcrelate</span>(iterator, </span>
<span id="cb272-4"><a href="ancestry-and-relatedness-inference.html#cb272-4"></a>                  <span class="dt">pcs=</span>pca<span class="op">$</span>vectors[,<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>], </span>
<span id="cb272-5"><a href="ancestry-and-relatedness-inference.html#cb272-5"></a>                  <span class="dt">training.set=</span>pca<span class="op">$</span>unrels)</span>
<span id="cb272-6"><a href="ancestry-and-relatedness-inference.html#cb272-6"></a><span class="kw">names</span>(pcrel)</span></code></pre></div>
<pre><code>## [1] &quot;kinBtwn&quot; &quot;kinSelf&quot;</code></pre>
<div class="sourceCode" id="cb274"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb274-1"><a href="ancestry-and-relatedness-inference.html#cb274-1"></a><span class="co"># relatedness between pairs of individuals</span></span>
<span id="cb274-2"><a href="ancestry-and-relatedness-inference.html#cb274-2"></a><span class="kw">dim</span>(pcrel<span class="op">$</span>kinBtwn)</span></code></pre></div>
<pre><code>## [1] 633375      6</code></pre>
<div class="sourceCode" id="cb276"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb276-1"><a href="ancestry-and-relatedness-inference.html#cb276-1"></a><span class="kw">head</span>(pcrel<span class="op">$</span>kinBtwn)</span></code></pre></div>
<pre><code>##       ID1     ID2          kin        k0          k2 nsnp
## 1 HG00096 HG00097 0.0195745605 0.8884407 -0.03326108 4238
## 2 HG00096 HG00099 0.0066373621 0.9415895 -0.03186106 4253
## 3 HG00096 HG00100 0.0008478914 0.9733286 -0.02327980 4200
## 4 HG00096 HG00101 0.0094582260 0.9347343 -0.02743277 4259
## 5 HG00096 HG00102 0.0143016544 0.9578218  0.01502844 4219
## 6 HG00096 HG00103 0.0168467786 0.9103588 -0.02225413 4252</code></pre>
<div class="sourceCode" id="cb278"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb278-1"><a href="ancestry-and-relatedness-inference.html#cb278-1"></a><span class="co"># self-kinship estimates</span></span>
<span id="cb278-2"><a href="ancestry-and-relatedness-inference.html#cb278-2"></a><span class="kw">dim</span>(pcrel<span class="op">$</span>kinSelf)</span></code></pre></div>
<pre><code>## [1] 1126    3</code></pre>
<div class="sourceCode" id="cb280"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb280-1"><a href="ancestry-and-relatedness-inference.html#cb280-1"></a><span class="kw">head</span>(pcrel<span class="op">$</span>kinSelf)</span></code></pre></div>
<pre><code>##        ID            f nsnp
## 1 HG00096 -0.010207002 4278
## 2 HG00097  0.003093096 4326
## 3 HG00099 -0.060436310 4303
## 4 HG00100  0.012633672 4270
## 5 HG00101 -0.033837780 4302
## 6 HG00102  0.030378865 4264</code></pre>
<p>We plot the pairwise kinship estimates againts the IBD0 (<code>k0</code>) estimates (the proportion of variants for which the pair of individuals share 0 alleles identical by descent). We use a hexbin plot to visualize the relatedness for all pairs of samples.</p>
<div class="sourceCode" id="cb282"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb282-1"><a href="ancestry-and-relatedness-inference.html#cb282-1"></a><span class="kw">ggplot</span>(pcrel<span class="op">$</span>kinBtwn, <span class="kw">aes</span>(k0, kin)) <span class="op">+</span></span>
<span id="cb282-2"><a href="ancestry-and-relatedness-inference.html#cb282-2"></a><span class="st">    </span><span class="kw">geom_hline</span>(<span class="dt">yintercept=</span><span class="dv">2</span><span class="op">^</span>(<span class="op">-</span><span class="kw">seq</span>(<span class="dv">3</span>,<span class="dv">9</span>,<span class="dv">2</span>)<span class="op">/</span><span class="dv">2</span>), <span class="dt">linetype=</span><span class="st">&quot;dashed&quot;</span>, <span class="dt">color=</span><span class="st">&quot;grey&quot;</span>) <span class="op">+</span></span>
<span id="cb282-3"><a href="ancestry-and-relatedness-inference.html#cb282-3"></a><span class="st">    </span><span class="kw">geom_hex</span>(<span class="dt">bins =</span> <span class="dv">100</span>) <span class="op">+</span></span>
<span id="cb282-4"><a href="ancestry-and-relatedness-inference.html#cb282-4"></a><span class="st">    </span><span class="kw">geom_abline</span>(<span class="dt">intercept =</span> <span class="fl">0.25</span>, <span class="dt">slope =</span> <span class="fl">-0.25</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb282-5"><a href="ancestry-and-relatedness-inference.html#cb282-5"></a><span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;kinship estimate&quot;</span>) <span class="op">+</span></span>
<span id="cb282-6"><a href="ancestry-and-relatedness-inference.html#cb282-6"></a><span class="st">    </span><span class="kw">theme_bw</span>()</span></code></pre></div>
<p><img src="_main_files/figure-html/pcrelate1_plot-1.png" width="672" /></p>
<p>We see that the PC-Relate relatedness estimates for unrelated pairs (i.e. kin ~ 0 and k0 ~ 1) are much closer to expectation than those from KING-robust.</p>
<p>We can use the <code>pcrelateToMatrix</code> function to transform the output into an (n x n) kinship matrix (KM).</p>
<div class="sourceCode" id="cb283"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb283-1"><a href="ancestry-and-relatedness-inference.html#cb283-1"></a>pcrelMat &lt;-<span class="st"> </span><span class="kw">pcrelateToMatrix</span>(pcrel, <span class="dt">scaleKin=</span><span class="dv">1</span>, <span class="dt">verbose=</span><span class="ot">FALSE</span>)</span>
<span id="cb283-2"><a href="ancestry-and-relatedness-inference.html#cb283-2"></a><span class="kw">dim</span>(pcrelMat)</span></code></pre></div>
<pre><code>## [1] 1126 1126</code></pre>
<div class="sourceCode" id="cb285"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb285-1"><a href="ancestry-and-relatedness-inference.html#cb285-1"></a><span class="co"># look at the top corner of the matrix</span></span>
<span id="cb285-2"><a href="ancestry-and-relatedness-inference.html#cb285-2"></a>pcrelMat[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>## 5 x 5 Matrix of class &quot;dsyMatrix&quot;
##              HG00096      HG00097       HG00099      HG00100       HG00101
## HG00096 0.4948964992 0.0195745605  0.0066373621 0.0008478914  0.0094582260
## HG00097 0.0195745605 0.5015465480  0.0008448096 0.0022230394  0.0048742994
## HG00099 0.0066373621 0.0008448096  0.4697818452 0.0290478189 -0.0002944969
## HG00100 0.0008478914 0.0022230394  0.0290478189 0.5063168358  0.0100841545
## HG00101 0.0094582260 0.0048742994 -0.0002944969 0.0100841545  0.4830811098</code></pre>
</div>
</div>
<div id="exercises-1" class="section level2" number="6.4">
<h2><span class="header-section-number">6.4</span> Exercises</h2>
<p>In small samples (such as this one), we recommend performing a second iteration of PC-AiR and PC-Relate. Now that we have the PC-Relate ancestry-adjusted kinship estimates, we can better partition our sample into unrelated and related sets. This can lead to better ancestry PCs from PC-AiR and better relatedness estimates from PC-Relate.</p>
<ol style="list-style-type: decimal">
<li><p>Perform a second PC-AiR analysis, this time using the PC-Relate kinship matrix as the kinship estimates (you should still use the KING-robust matrix for the ancestry divergence estimates). How does the unrelated set compare to the first PC-AiR analysis?</p></li>
<li><p>Make a parallel coordinates plot of the top 10 PC-AiR PCs. How does this compare to the plot from the first iteration? How many PCs seem to reflect ancestry?</p></li>
<li><p>Perform a second PC-Relate analysis, this time using the new PC-AiR PCs to adjust for ancestry. Make a hexbin plot of estimated kinship vs IBD0.</p></li>
</ol>
</div>
<div id="solutions-1" class="section level2" number="6.5">
<h2><span class="header-section-number">6.5</span> Solutions</h2>
<ol style="list-style-type: decimal">
<li>Perform a second PC-AiR analysis, this time using the PC-Relate kinship matrix as the kinship estimates (you should still use the KING-robust matrix for the ancestry divergence estimates). How does the unrelated set compare to the first PC-AiR analysis?</li>
</ol>
<p>We run the second PC-AiR analysis the same as the first, except using the PC-Relate kinship matrix we created above (<code>pcrelMat</code>) as the input to parameter <code>kinobj</code> – this means that we are using the PC-Relate estimates instead of the KING estimates to identify related pairs of samples. In the solution shown here, we also demonstrate that a <code>SeqVarData</code> object can be used as input, but we need to specify the variants to use in the analysis using the <code>snp.include</code> parameter.</p>
<div class="sourceCode" id="cb287"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb287-1"><a href="ancestry-and-relatedness-inference.html#cb287-1"></a><span class="kw">seqResetFilter</span>(seqData)</span></code></pre></div>
<pre><code>## # of selected samples: 1,126
## # of selected variants: 25,760</code></pre>
<div class="sourceCode" id="cb289"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb289-1"><a href="ancestry-and-relatedness-inference.html#cb289-1"></a><span class="co"># run PC-AiR</span></span>
<span id="cb289-2"><a href="ancestry-and-relatedness-inference.html#cb289-2"></a>pca2 &lt;-<span class="st"> </span><span class="kw">pcair</span>(seqData,</span>
<span id="cb289-3"><a href="ancestry-and-relatedness-inference.html#cb289-3"></a>             <span class="dt">kinobj=</span>pcrelMat, </span>
<span id="cb289-4"><a href="ancestry-and-relatedness-inference.html#cb289-4"></a>             <span class="dt">kin.thresh=</span><span class="dv">2</span><span class="op">^</span>(<span class="op">-</span><span class="dv">9</span><span class="op">/</span><span class="dv">2</span>),</span>
<span id="cb289-5"><a href="ancestry-and-relatedness-inference.html#cb289-5"></a>             <span class="dt">divobj=</span>kingMat, </span>
<span id="cb289-6"><a href="ancestry-and-relatedness-inference.html#cb289-6"></a>             <span class="dt">div.thresh=</span><span class="op">-</span><span class="dv">2</span><span class="op">^</span>(<span class="op">-</span><span class="dv">9</span><span class="op">/</span><span class="dv">2</span>),</span>
<span id="cb289-7"><a href="ancestry-and-relatedness-inference.html#cb289-7"></a>             <span class="dt">snp.include =</span> pruned)</span></code></pre></div>
<pre><code>## Principal Component Analysis (PCA) on genotypes:
## Calculating allele counts/frequencies ...
## 
[..................................................]  0%, ETC: ---    
[==================================================] 100%, completed, 1s
## # of selected variants: 20,945
## Excluding 342 SNVs (monomorphic: TRUE, MAF: NaN, missing rate: NaN)
##     # of samples: 1,070
##     # of SNVs: 20,945
##     using 1 thread
##     # of principal components: 32
## CPU capabilities: Double-Precision SSE2
## Tue Jul 20 14:31:32 2021    (internal increment: 332)
## 
[..................................................]  0%, ETC: ---        
[==================================================] 100%, completed, 35s
## Tue Jul 20 14:32:07 2021    Begin (eigenvalues and eigenvectors)
## Tue Jul 20 14:32:09 2021    Done.
## SNP Loading:
##     # of samples: 1,070
##     # of SNPs: 20,945
##     using 1 thread
##     using the top 32 eigenvectors
## Tue Jul 20 14:32:09 2021    (internal increment: 2664)
## 
[..................................................]  0%, ETC: ---        
[==================================================] 100%, completed, 1s
## Tue Jul 20 14:32:10 2021    Done.
## Sample Loading:
##     # of samples: 56
##     # of SNPs: 20,945
##     using 1 thread
##     using the top 32 eigenvectors
## Tue Jul 20 14:32:10 2021    (internal increment: 50904)
## 
[..................................................]  0%, ETC: ---        
[==================================================] 100%, completed, 0s
## Tue Jul 20 14:32:10 2021    Done.</code></pre>
<div class="sourceCode" id="cb291"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb291-1"><a href="ancestry-and-relatedness-inference.html#cb291-1"></a><span class="kw">names</span>(pca2)</span></code></pre></div>
<pre><code>##  [1] &quot;vectors&quot;    &quot;values&quot;     &quot;rels&quot;       &quot;unrels&quot;     &quot;kin.thresh&quot;
##  [6] &quot;div.thresh&quot; &quot;sample.id&quot;  &quot;nsamp&quot;      &quot;nsnps&quot;      &quot;varprop&quot;   
## [11] &quot;call&quot;       &quot;method&quot;</code></pre>
<div class="sourceCode" id="cb293"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb293-1"><a href="ancestry-and-relatedness-inference.html#cb293-1"></a><span class="co"># the unrelated set of samples</span></span>
<span id="cb293-2"><a href="ancestry-and-relatedness-inference.html#cb293-2"></a><span class="kw">length</span>(pca2<span class="op">$</span>unrels)</span></code></pre></div>
<pre><code>## [1] 1070</code></pre>
<div class="sourceCode" id="cb295"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb295-1"><a href="ancestry-and-relatedness-inference.html#cb295-1"></a><span class="co"># the related set of samples</span></span>
<span id="cb295-2"><a href="ancestry-and-relatedness-inference.html#cb295-2"></a><span class="kw">length</span>(pca2<span class="op">$</span>rels)</span></code></pre></div>
<pre><code>## [1] 56</code></pre>
<div class="sourceCode" id="cb297"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb297-1"><a href="ancestry-and-relatedness-inference.html#cb297-1"></a><span class="co"># extract the top 10 PCs and make a data.frame</span></span>
<span id="cb297-2"><a href="ancestry-and-relatedness-inference.html#cb297-2"></a>pcs2 &lt;-<span class="st"> </span><span class="kw">data.frame</span>(pca2<span class="op">$</span>vectors[,<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>])</span>
<span id="cb297-3"><a href="ancestry-and-relatedness-inference.html#cb297-3"></a><span class="kw">colnames</span>(pcs2) &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&#39;PC&#39;</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">10</span>)</span>
<span id="cb297-4"><a href="ancestry-and-relatedness-inference.html#cb297-4"></a>pcs2<span class="op">$</span>sample.id &lt;-<span class="st"> </span>pca2<span class="op">$</span>sample.id</span>
<span id="cb297-5"><a href="ancestry-and-relatedness-inference.html#cb297-5"></a><span class="kw">dim</span>(pcs2)</span></code></pre></div>
<pre><code>## [1] 1126   11</code></pre>
<div class="sourceCode" id="cb299"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb299-1"><a href="ancestry-and-relatedness-inference.html#cb299-1"></a><span class="kw">save</span>(pcs2, <span class="dt">file=</span><span class="st">&quot;data/pcs.RData&quot;</span>)</span></code></pre></div>
<p>We see that there are now 1,070 samples in the unrelated set, as opposed to 1,040 in the first PC-AiR analysis. This indicates that KING-robust likely overestimated relatedness for some pairs due to bias from ancestry admixture.</p>
<ol start="2" style="list-style-type: decimal">
<li>Make a parallel coordinates plot of the top 10 PC-AiR PCs. How does this compare to the plot from the first iteration? How many PCs seem to reflect ancestry?</li>
</ol>
<p>We can reuse the code from above to make the parallel coordinates plot.</p>
<div class="sourceCode" id="cb300"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb300-1"><a href="ancestry-and-relatedness-inference.html#cb300-1"></a><span class="kw">library</span>(Biobase)</span>
<span id="cb300-2"><a href="ancestry-and-relatedness-inference.html#cb300-2"></a>sampfile &lt;-<span class="st"> &quot;data/sample_annotation.RData&quot;</span></span>
<span id="cb300-3"><a href="ancestry-and-relatedness-inference.html#cb300-3"></a><span class="cf">if</span> (<span class="op">!</span><span class="kw">file.exists</span>(sampfile)) <span class="kw">download.file</span>(<span class="kw">file.path</span>(repo_path, sampfile), sampfile)</span>
<span id="cb300-4"><a href="ancestry-and-relatedness-inference.html#cb300-4"></a>annot &lt;-<span class="st"> </span><span class="kw">get</span>(<span class="kw">load</span>(sampfile))</span>
<span id="cb300-5"><a href="ancestry-and-relatedness-inference.html#cb300-5"></a></span>
<span id="cb300-6"><a href="ancestry-and-relatedness-inference.html#cb300-6"></a>pc.df &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(pcs2)</span>
<span id="cb300-7"><a href="ancestry-and-relatedness-inference.html#cb300-7"></a><span class="kw">names</span>(pc.df) &lt;-<span class="st"> </span><span class="dv">1</span><span class="op">:</span><span class="kw">ncol</span>(pcs2)</span>
<span id="cb300-8"><a href="ancestry-and-relatedness-inference.html#cb300-8"></a>pc.df<span class="op">$</span>sample.id &lt;-<span class="st"> </span><span class="kw">row.names</span>(pcs2)</span>
<span id="cb300-9"><a href="ancestry-and-relatedness-inference.html#cb300-9"></a></span>
<span id="cb300-10"><a href="ancestry-and-relatedness-inference.html#cb300-10"></a><span class="kw">library</span>(dplyr)</span>
<span id="cb300-11"><a href="ancestry-and-relatedness-inference.html#cb300-11"></a>annot &lt;-<span class="st"> </span><span class="kw">pData</span>(annot) <span class="op">%&gt;%</span></span>
<span id="cb300-12"><a href="ancestry-and-relatedness-inference.html#cb300-12"></a><span class="st">        </span>dplyr<span class="op">::</span><span class="kw">select</span>(sample.id, Population)</span>
<span id="cb300-13"><a href="ancestry-and-relatedness-inference.html#cb300-13"></a>pc.df &lt;-<span class="st"> </span><span class="kw">left_join</span>(pcs2, annot, <span class="dt">by=</span><span class="st">&quot;sample.id&quot;</span>)</span>
<span id="cb300-14"><a href="ancestry-and-relatedness-inference.html#cb300-14"></a></span>
<span id="cb300-15"><a href="ancestry-and-relatedness-inference.html#cb300-15"></a><span class="kw">library</span>(GGally)</span>
<span id="cb300-16"><a href="ancestry-and-relatedness-inference.html#cb300-16"></a><span class="kw">library</span>(RColorBrewer)</span>
<span id="cb300-17"><a href="ancestry-and-relatedness-inference.html#cb300-17"></a>pop.cols &lt;-<span class="st"> </span><span class="kw">setNames</span>(<span class="kw">brewer.pal</span>(<span class="dv">12</span>, <span class="st">&quot;Paired&quot;</span>),</span>
<span id="cb300-18"><a href="ancestry-and-relatedness-inference.html#cb300-18"></a>                 <span class="kw">c</span>(<span class="st">&quot;ACB&quot;</span>, <span class="st">&quot;ASW&quot;</span>, <span class="st">&quot;CEU&quot;</span>, <span class="st">&quot;GBR&quot;</span>, <span class="st">&quot;CHB&quot;</span>, <span class="st">&quot;JPT&quot;</span>, <span class="st">&quot;CLM&quot;</span>, <span class="st">&quot;MXL&quot;</span>, <span class="st">&quot;LWK&quot;</span>, <span class="st">&quot;YRI&quot;</span>, <span class="st">&quot;GIH&quot;</span>, <span class="st">&quot;PUR&quot;</span>))</span>
<span id="cb300-19"><a href="ancestry-and-relatedness-inference.html#cb300-19"></a><span class="kw">ggparcoord</span>(pc.df, <span class="dt">columns=</span><span class="dv">1</span><span class="op">:</span><span class="dv">10</span>, <span class="dt">groupColumn=</span><span class="st">&quot;Population&quot;</span>, <span class="dt">scale=</span><span class="st">&quot;uniminmax&quot;</span>) <span class="op">+</span></span>
<span id="cb300-20"><a href="ancestry-and-relatedness-inference.html#cb300-20"></a><span class="st">    </span><span class="kw">scale_color_manual</span>(<span class="dt">values=</span>pop.cols) <span class="op">+</span></span>
<span id="cb300-21"><a href="ancestry-and-relatedness-inference.html#cb300-21"></a><span class="st">    </span><span class="kw">xlab</span>(<span class="st">&quot;PC&quot;</span>) <span class="op">+</span><span class="st"> </span><span class="kw">ylab</span>(<span class="st">&quot;&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/pcair2_parcoord-1.png" width="672" /></p>
<p>The plot looks a bit cleaner than the one from the first PC-AiR analysis. Clearly, PCs 1-4 are reflective of ancestry here. In the prior analysis, PC7 also seemed to pick up some component of ancestry.</p>
<ol start="3" style="list-style-type: decimal">
<li>Perform a second PC-Relate analysis, this time using the new PC-AiR PCs to adjust for ancestry. Make a hexbin plot of estimated kinship vs IBD0.</li>
</ol>
<p>We run the second PC-Relate analysis the same as the first, except using the new PC-AiR PCs that we just generated to adjust for ancestry, and using the new PC-AiR unrelated set as our <code>training.set</code>.</p>
<div class="sourceCode" id="cb301"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb301-1"><a href="ancestry-and-relatedness-inference.html#cb301-1"></a><span class="co"># filter the GDS object to our LD-pruned variants</span></span>
<span id="cb301-2"><a href="ancestry-and-relatedness-inference.html#cb301-2"></a><span class="kw">seqSetFilter</span>(seqData, <span class="dt">variant.id=</span>pruned)</span></code></pre></div>
<pre><code>## # of selected variants: 21,287</code></pre>
<div class="sourceCode" id="cb303"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb303-1"><a href="ancestry-and-relatedness-inference.html#cb303-1"></a>iterator &lt;-<span class="st"> </span><span class="kw">SeqVarBlockIterator</span>(seqData, <span class="dt">verbose=</span><span class="ot">FALSE</span>)</span>
<span id="cb303-2"><a href="ancestry-and-relatedness-inference.html#cb303-2"></a></span>
<span id="cb303-3"><a href="ancestry-and-relatedness-inference.html#cb303-3"></a><span class="co"># run PC-Relate</span></span>
<span id="cb303-4"><a href="ancestry-and-relatedness-inference.html#cb303-4"></a>pcrel2 &lt;-<span class="st"> </span><span class="kw">pcrelate</span>(iterator, </span>
<span id="cb303-5"><a href="ancestry-and-relatedness-inference.html#cb303-5"></a>                  <span class="dt">pcs=</span>pca2<span class="op">$</span>vectors[,<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>], </span>
<span id="cb303-6"><a href="ancestry-and-relatedness-inference.html#cb303-6"></a>                  <span class="dt">training.set=</span>pca2<span class="op">$</span>unrels)</span>
<span id="cb303-7"><a href="ancestry-and-relatedness-inference.html#cb303-7"></a><span class="kw">save</span>(pcrel2, <span class="dt">file=</span><span class="st">&quot;data/pcrelate_kinship.RData&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb304"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb304-1"><a href="ancestry-and-relatedness-inference.html#cb304-1"></a><span class="kw">ggplot</span>(pcrel2<span class="op">$</span>kinBtwn, <span class="kw">aes</span>(k0, kin)) <span class="op">+</span></span>
<span id="cb304-2"><a href="ancestry-and-relatedness-inference.html#cb304-2"></a><span class="st">    </span><span class="kw">geom_hline</span>(<span class="dt">yintercept=</span><span class="dv">2</span><span class="op">^</span>(<span class="op">-</span><span class="kw">seq</span>(<span class="dv">3</span>,<span class="dv">9</span>,<span class="dv">2</span>)<span class="op">/</span><span class="dv">2</span>), <span class="dt">linetype=</span><span class="st">&quot;dashed&quot;</span>, <span class="dt">color=</span><span class="st">&quot;grey&quot;</span>) <span class="op">+</span></span>
<span id="cb304-3"><a href="ancestry-and-relatedness-inference.html#cb304-3"></a><span class="st">    </span><span class="kw">geom_hex</span>(<span class="dt">bins =</span> <span class="dv">100</span>) <span class="op">+</span></span>
<span id="cb304-4"><a href="ancestry-and-relatedness-inference.html#cb304-4"></a><span class="st">    </span><span class="kw">geom_abline</span>(<span class="dt">intercept =</span> <span class="fl">0.25</span>, <span class="dt">slope =</span> <span class="fl">-0.25</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb304-5"><a href="ancestry-and-relatedness-inference.html#cb304-5"></a><span class="st">    </span><span class="kw">ylab</span>(<span class="st">&quot;kinship estimate&quot;</span>) <span class="op">+</span></span>
<span id="cb304-6"><a href="ancestry-and-relatedness-inference.html#cb304-6"></a><span class="st">    </span><span class="kw">theme_bw</span>()</span></code></pre></div>
<p><img src="_main_files/figure-html/pcrelate2_plot-1.png" width="672" /></p>
<div class="sourceCode" id="cb305"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb305-1"><a href="ancestry-and-relatedness-inference.html#cb305-1"></a><span class="kw">seqClose</span>(seqData)</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="association-tests-part-ii.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="mixed-models.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
