<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>4 Association tests | SISG 2021 Module 16: Computational Pipeline for WGS Data</title>
  <meta name="description" content="Course materials for SISG Module 16" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="4 Association tests | SISG 2021 Module 16: Computational Pipeline for WGS Data" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Course materials for SISG Module 16" />
  <meta name="github-repo" content="UW-GAC/SISG_2021" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="4 Association tests | SISG 2021 Module 16: Computational Pipeline for WGS Data" />
  
  <meta name="twitter:description" content="Course materials for SISG Module 16" />
  



<meta name="date" content="2021-07-09" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="phenotype-harmonization.html"/>
<link rel="next" href="computing-a-grm.html"/>
<script src="libs/header-attrs-2.9/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#schedule"><i class="fa fa-check"></i><b>1.1</b> Schedule</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#r-packages-used"><i class="fa fa-check"></i><b>1.2</b> R packages used</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#resources"><i class="fa fa-check"></i><b>1.3</b> Resources</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="gds-format.html"><a href="gds-format.html"><i class="fa fa-check"></i><b>2</b> GDS format</a>
<ul>
<li class="chapter" data-level="2.1" data-path="gds-format.html"><a href="gds-format.html#convert-a-vcf-to-gds"><i class="fa fa-check"></i><b>2.1</b> Convert a VCF to GDS</a></li>
<li class="chapter" data-level="2.2" data-path="gds-format.html"><a href="gds-format.html#exploring-a-gds-file"><i class="fa fa-check"></i><b>2.2</b> Exploring a GDS file</a></li>
<li class="chapter" data-level="2.3" data-path="gds-format.html"><a href="gds-format.html#exercises"><i class="fa fa-check"></i><b>2.3</b> Exercises</a></li>
<li class="chapter" data-level="2.4" data-path="gds-format.html"><a href="gds-format.html#solutions"><i class="fa fa-check"></i><b>2.4</b> Solutions</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="phenotype-harmonization.html"><a href="phenotype-harmonization.html"><i class="fa fa-check"></i><b>3</b> Phenotype Harmonization</a>
<ul>
<li class="chapter" data-level="3.1" data-path="phenotype-harmonization.html"><a href="phenotype-harmonization.html#inspect-individual-study-data-in-r"><i class="fa fa-check"></i><b>3.1</b> Inspect individual study data in R</a></li>
<li class="chapter" data-level="3.2" data-path="phenotype-harmonization.html"><a href="phenotype-harmonization.html#compare-study-values"><i class="fa fa-check"></i><b>3.2</b> Compare study values</a></li>
<li class="chapter" data-level="3.3" data-path="phenotype-harmonization.html"><a href="phenotype-harmonization.html#using-null-models-to-compare-studies"><i class="fa fa-check"></i><b>3.3</b> Using null models to compare studies</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="phenotype-harmonization.html"><a href="phenotype-harmonization.html#create-an-annotated-data-frame"><i class="fa fa-check"></i><b>3.3.1</b> Create an Annotated Data Frame</a></li>
<li class="chapter" data-level="3.3.2" data-path="phenotype-harmonization.html"><a href="phenotype-harmonization.html#fit-a-null-model-without-study"><i class="fa fa-check"></i><b>3.3.2</b> Fit a null model without study</a></li>
<li class="chapter" data-level="3.3.3" data-path="phenotype-harmonization.html"><a href="phenotype-harmonization.html#fit-a-model-with-study"><i class="fa fa-check"></i><b>3.3.3</b> Fit a model with study</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="phenotype-harmonization.html"><a href="phenotype-harmonization.html#final-considerations"><i class="fa fa-check"></i><b>3.4</b> Final considerations</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="association-tests.html"><a href="association-tests.html"><i class="fa fa-check"></i><b>4</b> Association tests</a>
<ul>
<li class="chapter" data-level="4.1" data-path="association-tests.html"><a href="association-tests.html#null-model"><i class="fa fa-check"></i><b>4.1</b> Null model</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="association-tests.html"><a href="association-tests.html#prepare-the-data"><i class="fa fa-check"></i><b>4.1.1</b> Prepare the data</a></li>
<li class="chapter" data-level="4.1.2" data-path="association-tests.html"><a href="association-tests.html#fit-the-null-model"><i class="fa fa-check"></i><b>4.1.2</b> Fit the null model</a></li>
<li class="chapter" data-level="4.1.3" data-path="association-tests.html"><a href="association-tests.html#exercise"><i class="fa fa-check"></i><b>4.1.3</b> Exercise</a></li>
<li class="chapter" data-level="4.1.4" data-path="association-tests.html"><a href="association-tests.html#solution"><i class="fa fa-check"></i><b>4.1.4</b> Solution</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="association-tests.html"><a href="association-tests.html#single-variant-tests"><i class="fa fa-check"></i><b>4.2</b> Single-variant tests</a></li>
<li class="chapter" data-level="4.3" data-path="association-tests.html"><a href="association-tests.html#exercises-1"><i class="fa fa-check"></i><b>4.3</b> Exercises</a></li>
<li class="chapter" data-level="4.4" data-path="association-tests.html"><a href="association-tests.html#sliding-window-tests"><i class="fa fa-check"></i><b>4.4</b> Sliding window tests</a></li>
<li class="chapter" data-level="4.5" data-path="association-tests.html"><a href="association-tests.html#exercise-1"><i class="fa fa-check"></i><b>4.5</b> Exercise</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="association-tests.html"><a href="association-tests.html#association-tests---solutions"><i class="fa fa-check"></i><b>4.5.1</b> Association tests - Solutions</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="computing-a-grm.html"><a href="computing-a-grm.html"><i class="fa fa-check"></i><b>5</b> Computing a GRM</a>
<ul>
<li class="chapter" data-level="5.1" data-path="computing-a-grm.html"><a href="computing-a-grm.html#king"><i class="fa fa-check"></i><b>5.1</b> KING</a></li>
<li class="chapter" data-level="5.2" data-path="computing-a-grm.html"><a href="computing-a-grm.html#pc-air"><i class="fa fa-check"></i><b>5.2</b> PC-AiR</a></li>
<li class="chapter" data-level="5.3" data-path="computing-a-grm.html"><a href="computing-a-grm.html#pc-relate"><i class="fa fa-check"></i><b>5.3</b> PC-Relate</a></li>
<li class="chapter" data-level="5.4" data-path="computing-a-grm.html"><a href="computing-a-grm.html#comparison-with-pedigree"><i class="fa fa-check"></i><b>5.4</b> Comparison with pedigree</a></li>
<li class="chapter" data-level="5.5" data-path="computing-a-grm.html"><a href="computing-a-grm.html#exercise-2"><i class="fa fa-check"></i><b>5.5</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="mixed-models.html"><a href="mixed-models.html"><i class="fa fa-check"></i><b>6</b> Mixed models</a>
<ul>
<li class="chapter" data-level="6.1" data-path="mixed-models.html"><a href="mixed-models.html#null-model-1"><i class="fa fa-check"></i><b>6.1</b> Null model</a></li>
<li class="chapter" data-level="6.2" data-path="mixed-models.html"><a href="mixed-models.html#single-variant-tests-1"><i class="fa fa-check"></i><b>6.2</b> Single-variant tests</a></li>
<li class="chapter" data-level="6.3" data-path="mixed-models.html"><a href="mixed-models.html#exercise-3"><i class="fa fa-check"></i><b>6.3</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="variant-annotation.html"><a href="variant-annotation.html"><i class="fa fa-check"></i><b>7</b> Variant annotation</a>
<ul>
<li class="chapter" data-level="7.1" data-path="variant-annotation.html"><a href="variant-annotation.html#using-bioconductor-annotation-resources"><i class="fa fa-check"></i><b>7.1</b> Using Bioconductor annotation resources</a></li>
<li class="chapter" data-level="7.2" data-path="variant-annotation.html"><a href="variant-annotation.html#aggregating-and-filtering-variants-using-annotation"><i class="fa fa-check"></i><b>7.2</b> Aggregating and filtering variants using annotation</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="aggregate-tests.html"><a href="aggregate-tests.html"><i class="fa fa-check"></i><b>8</b> Aggregate tests</a>
<ul>
<li class="chapter" data-level="8.1" data-path="aggregate-tests.html"><a href="aggregate-tests.html#aggregate-unit-for-association-testing-exercise"><i class="fa fa-check"></i><b>8.1</b> Aggregate unit for association testing exercise</a></li>
<li class="chapter" data-level="8.2" data-path="aggregate-tests.html"><a href="aggregate-tests.html#association-testing-with-aggregate-units"><i class="fa fa-check"></i><b>8.2</b> Association testing with aggregate units</a></li>
<li class="chapter" data-level="8.3" data-path="aggregate-tests.html"><a href="aggregate-tests.html#exercise-4"><i class="fa fa-check"></i><b>8.3</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="exercises-running-a-gwas-workflow.html"><a href="exercises-running-a-gwas-workflow.html"><i class="fa fa-check"></i><b>9</b> Exercises: Running a GWAS workflow</a>
<ul>
<li class="chapter" data-level="9.1" data-path="exercises-running-a-gwas-workflow.html"><a href="exercises-running-a-gwas-workflow.html#vcf-to-gds"><i class="fa fa-check"></i><b>9.1</b> VCF to GDS</a></li>
<li class="chapter" data-level="9.2" data-path="exercises-running-a-gwas-workflow.html"><a href="exercises-running-a-gwas-workflow.html#null-model-2"><i class="fa fa-check"></i><b>9.2</b> Null Model</a></li>
<li class="chapter" data-level="9.3" data-path="exercises-running-a-gwas-workflow.html"><a href="exercises-running-a-gwas-workflow.html#single-variant-test"><i class="fa fa-check"></i><b>9.3</b> Single variant test</a></li>
<li class="chapter" data-level="9.4" data-path="exercises-running-a-gwas-workflow.html"><a href="exercises-running-a-gwas-workflow.html#analysis-follow-up"><i class="fa fa-check"></i><b>9.4</b> Analysis follow-up</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">SISG 2021 Module 16: Computational Pipeline for WGS Data</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="association-tests" class="section level1" number="4">
<h1><span class="header-section-number">4</span> Association tests</h1>
<p>These exercises introduce genetic association testing: how to identify which genetic variants are associated with a phenotype.</p>
<div id="null-model" class="section level2" number="4.1">
<h2><span class="header-section-number">4.1</span> Null model</h2>
<p>The first step in our association testing procedure is to fit the “null model” – i.e., a model fit under the null hypothesis of no individual variant association. Operationally, this is fitting a regression model with the desired outcome phenotype, fixed effect covariates, and random effects.</p>
<div id="prepare-the-data" class="section level3" number="4.1.1">
<h3><span class="header-section-number">4.1.1</span> Prepare the data</h3>
<p>To fit the null model, we will need to create an <code>AnnotatedDataFrame</code> with sample information and phenotype data. We will merge our sample annotation file, which is indexed by a <code>sample.id</code> column matched to the GDS file, with our phenotype file, which is indexed by a <code>subject_id</code> column.</p>
<p>NOTE: In this example, we use the 1000 Genomes IDs for both sample and subject IDs, though we would generally advise using separate IDs for samples (sequencing instances) and subjects (individuals).</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="association-tests.html#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># sample annotation</span></span>
<span id="cb1-2"><a href="association-tests.html#cb1-2" aria-hidden="true" tabindex="-1"></a>repo_path <span class="ot">&lt;-</span> <span class="st">&quot;https://github.com/UW-GAC/SISG_2021/raw/master&quot;</span></span>
<span id="cb1-3"><a href="association-tests.html#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">dir.exists</span>(<span class="st">&quot;data&quot;</span>)) <span class="fu">dir.create</span>(<span class="st">&quot;data&quot;</span>)</span>
<span id="cb1-4"><a href="association-tests.html#cb1-4" aria-hidden="true" tabindex="-1"></a>sampfile <span class="ot">&lt;-</span> <span class="st">&quot;data/sample_annotation.RData&quot;</span></span>
<span id="cb1-5"><a href="association-tests.html#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(sampfile)) <span class="fu">download.file</span>(<span class="fu">file.path</span>(repo_path, sampfile), sampfile)</span>
<span id="cb1-6"><a href="association-tests.html#cb1-6" aria-hidden="true" tabindex="-1"></a>samp <span class="ot">&lt;-</span> <span class="fu">get</span>(<span class="fu">load</span>(sampfile))</span>
<span id="cb1-7"><a href="association-tests.html#cb1-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-8"><a href="association-tests.html#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Biobase)</span></code></pre></div>
<pre><code>## Loading required package: BiocGenerics</code></pre>
<pre><code>## Loading required package: parallel</code></pre>
<pre><code>## 
## Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:parallel&#39;:
## 
##     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
##     clusterExport, clusterMap, parApply, parCapply, parLapply,
##     parLapplyLB, parRapply, parSapply, parSapplyLB</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     IQR, mad, sd, var, xtabs</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     anyDuplicated, append, as.data.frame, basename, cbind, colnames,
##     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
##     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
##     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
##     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
##     union, unique, unsplit, which.max, which.min</code></pre>
<pre><code>## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view with
##     &#39;browseVignettes()&#39;. To cite Bioconductor, see
##     &#39;citation(&quot;Biobase&quot;)&#39;, and for packages &#39;citation(&quot;pkgname&quot;)&#39;.</code></pre>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="association-tests.html#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># access the data with the pData() function</span></span>
<span id="cb9-2"><a href="association-tests.html#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">pData</span>(samp))</span></code></pre></div>
<pre><code>##   sample.id subject.id Population          Population.Description sex status
## 1   HG00096    HG00096        GBR British in England and Scotland   M      0
## 2   HG00097    HG00097        GBR British in England and Scotland   F      1
## 3   HG00099    HG00099        GBR British in England and Scotland   F      0
## 4   HG00100    HG00100        GBR British in England and Scotland   F      1
## 5   HG00101    HG00101        GBR British in England and Scotland   M      0
## 6   HG00102    HG00102        GBR British in England and Scotland   F      0</code></pre>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="association-tests.html#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># access the metadata with the varMetadata() function</span></span>
<span id="cb11-2"><a href="association-tests.html#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="fu">varMetadata</span>(samp)</span></code></pre></div>
<pre><code>##                                     labelDescription
## sample.id                          sample identifier
## subject.id                        subject identifier
## Population                   population abbreviation
## Population.Description        population description
## sex                                              sex
## status                 simulated case/control status</code></pre>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="association-tests.html#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># phenotype data</span></span>
<span id="cb13-2"><a href="association-tests.html#cb13-2" aria-hidden="true" tabindex="-1"></a>phenfile <span class="ot">&lt;-</span> <span class="st">&quot;data/phenotype_annotation.RData&quot;</span></span>
<span id="cb13-3"><a href="association-tests.html#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(phenfile)) <span class="fu">download.file</span>(<span class="fu">file.path</span>(repo_path, phenfile), phenfile)</span>
<span id="cb13-4"><a href="association-tests.html#cb13-4" aria-hidden="true" tabindex="-1"></a>phen <span class="ot">&lt;-</span> <span class="fu">get</span>(<span class="fu">load</span>(phenfile))</span>
<span id="cb13-5"><a href="association-tests.html#cb13-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-6"><a href="association-tests.html#cb13-6" aria-hidden="true" tabindex="-1"></a><span class="co"># access the data with the pData() function</span></span>
<span id="cb13-7"><a href="association-tests.html#cb13-7" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">pData</span>(phen))</span></code></pre></div>
<pre><code>##   subject_id sex age height   study
## 1    HG00096   M  47  165.3 study_1
## 2    HG00102   F  49  169.1 study_1
## 3    HG00112   M  46  167.9 study_1
## 4    HG00114   M  49  169.5 study_1
## 5    HG00115   M  35  161.1 study_1
## 6    HG00116   M  37  182.2 study_1</code></pre>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="association-tests.html#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># access the metadata with the varMetadata() function</span></span>
<span id="cb15-2"><a href="association-tests.html#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="fu">varMetadata</span>(phen)</span></code></pre></div>
<pre><code>##                        labelDescription
## subject_id           subject identifier
## sex                       subject&#39;s sex
## age        age at measurement of height
## height           subject&#39;s height in cm
## study                  study identifier</code></pre>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="association-tests.html#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># merge sample annotation with phenotype data</span></span>
<span id="cb17-2"><a href="association-tests.html#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span></code></pre></div>
<pre><code>## 
## Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:Biobase&#39;:
## 
##     combine</code></pre>
<pre><code>## The following objects are masked from &#39;package:BiocGenerics&#39;:
## 
##     combine, intersect, setdiff, union</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     filter, lag</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     intersect, setdiff, setequal, union</code></pre>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="association-tests.html#cb23-1" aria-hidden="true" tabindex="-1"></a>dat <span class="ot">&lt;-</span> <span class="fu">pData</span>(samp) <span class="sc">%&gt;%</span></span>
<span id="cb23-2"><a href="association-tests.html#cb23-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(<span class="fu">pData</span>(phen), <span class="at">by=</span><span class="fu">c</span>(<span class="st">&quot;subject.id&quot;</span><span class="ot">=</span><span class="st">&quot;subject_id&quot;</span>, <span class="st">&quot;sex&quot;</span><span class="ot">=</span><span class="st">&quot;sex&quot;</span>))</span>
<span id="cb23-3"><a href="association-tests.html#cb23-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(dat)</span></code></pre></div>
<pre><code>##   sample.id subject.id Population          Population.Description sex status
## 1   HG00096    HG00096        GBR British in England and Scotland   M      0
## 2   HG00097    HG00097        GBR British in England and Scotland   F      1
## 3   HG00099    HG00099        GBR British in England and Scotland   F      0
## 4   HG00100    HG00100        GBR British in England and Scotland   F      1
## 5   HG00101    HG00101        GBR British in England and Scotland   M      0
## 6   HG00102    HG00102        GBR British in England and Scotland   F      0
##   age  height   study
## 1  47 165.300 study_1
## 2  47 144.780 study_3
## 3  40 185.500 study_2
## 4  45 150.622 study_3
## 5  40 177.800 study_3
## 6  49 169.100 study_1</code></pre>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="association-tests.html#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co"># merge the metadata</span></span>
<span id="cb25-2"><a href="association-tests.html#cb25-2" aria-hidden="true" tabindex="-1"></a>meta <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(<span class="fu">varMetadata</span>(samp), <span class="fu">varMetadata</span>(phen)[<span class="dv">3</span><span class="sc">:</span><span class="dv">5</span>,,<span class="at">drop=</span><span class="cn">FALSE</span>])</span>
<span id="cb25-3"><a href="association-tests.html#cb25-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-4"><a href="association-tests.html#cb25-4" aria-hidden="true" tabindex="-1"></a><span class="co"># make an AnnotatedDataFrame</span></span>
<span id="cb25-5"><a href="association-tests.html#cb25-5" aria-hidden="true" tabindex="-1"></a>annot <span class="ot">&lt;-</span> <span class="fu">AnnotatedDataFrame</span>(dat, meta)</span>
<span id="cb25-6"><a href="association-tests.html#cb25-6" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(annot, <span class="at">file=</span><span class="st">&quot;data/sample_phenotype_annotation.RData&quot;</span>)</span></code></pre></div>
</div>
<div id="fit-the-null-model" class="section level3" number="4.1.2">
<h3><span class="header-section-number">4.1.2</span> Fit the null model</h3>
<p>We will test for an association between variant genotypes and height, adjusting for sex, age, and study as fixed effect covariates. If the sample set involves multiple distinct groups with different variances for the phenotype, we recommend allowing for heterogeneous residual variance among groups with the <code>group.var</code> parameter. We saw in a previous exercise that the variance of height differs by study.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="association-tests.html#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GENESIS)</span>
<span id="cb26-2"><a href="association-tests.html#cb26-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-3"><a href="association-tests.html#cb26-3" aria-hidden="true" tabindex="-1"></a><span class="co"># fit the null model</span></span>
<span id="cb26-4"><a href="association-tests.html#cb26-4" aria-hidden="true" tabindex="-1"></a>nullmod <span class="ot">&lt;-</span> <span class="fu">fitNullModel</span>(annot, </span>
<span id="cb26-5"><a href="association-tests.html#cb26-5" aria-hidden="true" tabindex="-1"></a>                        <span class="at">outcome=</span><span class="st">&quot;height&quot;</span>, </span>
<span id="cb26-6"><a href="association-tests.html#cb26-6" aria-hidden="true" tabindex="-1"></a>                        <span class="at">covars=</span><span class="fu">c</span>(<span class="st">&quot;sex&quot;</span>, <span class="st">&quot;age&quot;</span>, <span class="st">&quot;study&quot;</span>), </span>
<span id="cb26-7"><a href="association-tests.html#cb26-7" aria-hidden="true" tabindex="-1"></a>                        <span class="at">group.var=</span><span class="st">&quot;study&quot;</span>, </span>
<span id="cb26-8"><a href="association-tests.html#cb26-8" aria-hidden="true" tabindex="-1"></a>                        <span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb26-9"><a href="association-tests.html#cb26-9" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(nullmod, <span class="at">file=</span><span class="st">&quot;data/null_model.RData&quot;</span>)</span></code></pre></div>
<p>The <code>fitNullModel</code> function returns a lot of information about the model that was fit. We examine some of that information below; to see all of the components, try <code>names(nullmod)</code>.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="association-tests.html#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co"># description of the model we fit</span></span>
<span id="cb27-2"><a href="association-tests.html#cb27-2" aria-hidden="true" tabindex="-1"></a>nullmod<span class="sc">$</span>model</span></code></pre></div>
<pre><code>## $hetResid
## [1] TRUE
## 
## $family
## 
## Family: gaussian 
## Link function: identity 
## 
## 
## $outcome
## [1] &quot;height&quot;
## 
## $covars
## [1] &quot;sex&quot;   &quot;age&quot;   &quot;study&quot;
## 
## $formula
## [1] &quot;height ~ sex + age + study + var(study)&quot;</code></pre>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="association-tests.html#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co"># fixed effect regression estimates</span></span>
<span id="cb29-2"><a href="association-tests.html#cb29-2" aria-hidden="true" tabindex="-1"></a>nullmod<span class="sc">$</span>fixef</span></code></pre></div>
<pre><code>##                       Est         SE        Stat         pval
## (Intercept)  163.67175933 3.18936046 2633.542247 0.000000e+00
## sexM           6.28764509 0.68812251   83.491932 6.397653e-20
## age            0.07519782 0.06921691    1.180283 2.772984e-01
## studystudy_2  10.63152325 0.82176939  167.375182 2.769992e-38
## studystudy_3  -8.96183691 0.84479021  112.537257 2.724959e-26</code></pre>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="association-tests.html#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co"># variance component estimates by group.var</span></span>
<span id="cb31-2"><a href="association-tests.html#cb31-2" aria-hidden="true" tabindex="-1"></a>nullmod<span class="sc">$</span>varComp</span></code></pre></div>
<pre><code>## V_study_1 V_study_3 V_study_2 
##  98.20191 168.82044 155.70722</code></pre>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="association-tests.html#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="co"># model fit: fitted values, residuals</span></span>
<span id="cb33-2"><a href="association-tests.html#cb33-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(nullmod<span class="sc">$</span>fit)</span></code></pre></div>
<pre><code>##         outcome workingY fitted.values resid.marginal    resid.PY
## HG00096 165.300  165.300      173.4937      -8.193702 -0.08343729
## HG00097 144.780  144.780      158.2442     -13.464220 -0.07975468
## HG00099 185.500  185.500      177.3112       8.188805  0.05259104
## HG00100 150.622  150.622      158.0938      -7.471824 -0.04425900
## HG00101 177.800  177.800      164.0055      13.794520  0.08171119
## HG00102 169.100  169.100      167.3565       1.743547  0.01775472
##         resid.cholesky sample.id
## HG00096     -0.8268376   HG00096
## HG00097     -1.0362599   HG00097
## HG00099      0.6562452   HG00099
## HG00100     -0.5750613   HG00100
## HG00101      1.0616810   HG00101
## HG00102      0.1759437   HG00102</code></pre>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="association-tests.html#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb35-2"><a href="association-tests.html#cb35-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(nullmod<span class="sc">$</span>fit, <span class="fu">aes</span>(<span class="at">x =</span> fitted.values, <span class="at">y =</span> resid.marginal)) <span class="sc">+</span> </span>
<span id="cb35-3"><a href="association-tests.html#cb35-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span> </span>
<span id="cb35-4"><a href="association-tests.html#cb35-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>) <span class="sc">+</span> </span>
<span id="cb35-5"><a href="association-tests.html#cb35-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">&#39;lm&#39;</span>)</span></code></pre></div>
<pre><code>## `geom_smooth()` using formula &#39;y ~ x&#39;</code></pre>
<p><img src="_main_files/figure-html/null_model_results-1.png" width="672" /></p>
</div>
<div id="exercise" class="section level3" number="4.1.3">
<h3><span class="header-section-number">4.1.3</span> Exercise</h3>
<ol style="list-style-type: decimal">
<li>Inverse Normal transform. As discussed in the lecture, we recommend a fully adjusted two-stage inverse Normalization procedure for fitting the null model when phenotypes have non-Normal distributions. Using the <code>two.stage</code> option in <code>fitNullModel</code>, fit a two-stage null model. Compare these residuals with the residuals from the original null model.</li>
</ol>
</div>
<div id="solution" class="section level3" number="4.1.4">
<h3><span class="header-section-number">4.1.4</span> Solution</h3>
<p>To run the fully adjusted two.stage null model, we simply set the <code>two.stage</code> option to <code>TRUE</code>. The <code>norm.option</code> parameter determines if the inverse Normalization should be done with all samples together (<code>"all"</code>) or within each <code>group.var</code> group separately (<code>"by.group"</code>).</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="association-tests.html#cb37-1" aria-hidden="true" tabindex="-1"></a>nullmod.twostage <span class="ot">&lt;-</span> <span class="fu">fitNullModel</span>(annot, </span>
<span id="cb37-2"><a href="association-tests.html#cb37-2" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">outcome=</span><span class="st">&quot;height&quot;</span>, </span>
<span id="cb37-3"><a href="association-tests.html#cb37-3" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">covars=</span><span class="fu">c</span>(<span class="st">&quot;sex&quot;</span>, <span class="st">&quot;age&quot;</span>, <span class="st">&quot;study&quot;</span>), </span>
<span id="cb37-4"><a href="association-tests.html#cb37-4" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">group.var=</span><span class="st">&quot;study&quot;</span>, </span>
<span id="cb37-5"><a href="association-tests.html#cb37-5" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">two.stage =</span> <span class="cn">TRUE</span>,</span>
<span id="cb37-6"><a href="association-tests.html#cb37-6" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">norm.option =</span> <span class="st">&quot;by.group&quot;</span>,</span>
<span id="cb37-7"><a href="association-tests.html#cb37-7" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb37-8"><a href="association-tests.html#cb37-8" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(nullmod.twostage, <span class="at">file=</span><span class="st">&quot;data/null_model_two_stage.RData&quot;</span>)</span>
<span id="cb37-9"><a href="association-tests.html#cb37-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-10"><a href="association-tests.html#cb37-10" aria-hidden="true" tabindex="-1"></a><span class="co"># description of the model we fit</span></span>
<span id="cb37-11"><a href="association-tests.html#cb37-11" aria-hidden="true" tabindex="-1"></a>nullmod.twostage<span class="sc">$</span>model</span></code></pre></div>
<pre><code>## $hetResid
## [1] TRUE
## 
## $family
## 
## Family: gaussian 
## Link function: identity 
## 
## 
## $outcome
## [1] &quot;height&quot;
## 
## $covars
## [1] &quot;sex&quot;   &quot;age&quot;   &quot;study&quot;
## 
## $formula
## [1] &quot;rankInvNorm(resid(height)) ~ sex + age + study + var(study)&quot;</code></pre>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="association-tests.html#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="co"># compare the marginal residuals</span></span>
<span id="cb39-2"><a href="association-tests.html#cb39-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-3"><a href="association-tests.html#cb39-3" aria-hidden="true" tabindex="-1"></a><span class="co"># merge the data for plotting</span></span>
<span id="cb39-4"><a href="association-tests.html#cb39-4" aria-hidden="true" tabindex="-1"></a>pdat <span class="ot">&lt;-</span> <span class="fu">merge</span>(nullmod<span class="sc">$</span>fit, nullmod.twostage<span class="sc">$</span>fit, </span>
<span id="cb39-5"><a href="association-tests.html#cb39-5" aria-hidden="true" tabindex="-1"></a>              <span class="at">by =</span> <span class="st">&#39;sample.id&#39;</span>, <span class="at">suffixes =</span> <span class="fu">c</span>(<span class="st">&#39;.orig&#39;</span>, <span class="st">&#39;.twostage&#39;</span>))</span>
<span id="cb39-6"><a href="association-tests.html#cb39-6" aria-hidden="true" tabindex="-1"></a>pdat <span class="ot">&lt;-</span> <span class="fu">merge</span>(pdat, <span class="fu">pData</span>(annot), <span class="at">by =</span> <span class="st">&#39;sample.id&#39;</span>)</span>
<span id="cb39-7"><a href="association-tests.html#cb39-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-8"><a href="association-tests.html#cb39-8" aria-hidden="true" tabindex="-1"></a><span class="co"># distribution of residuals - original null model</span></span>
<span id="cb39-9"><a href="association-tests.html#cb39-9" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(pdat, <span class="fu">aes</span>(<span class="at">x =</span> resid.marginal.orig, <span class="at">color =</span> study)) <span class="sc">+</span> </span>
<span id="cb39-10"><a href="association-tests.html#cb39-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_density</span>()</span></code></pre></div>
<p><img src="_main_files/figure-html/null_model_two_stage-1.png" width="672" /></p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="association-tests.html#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="co"># distribution of residuals - two stage null model</span></span>
<span id="cb40-2"><a href="association-tests.html#cb40-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(pdat, <span class="fu">aes</span>(<span class="at">x =</span> resid.marginal.twostage, <span class="at">color =</span> study)) <span class="sc">+</span> </span>
<span id="cb40-3"><a href="association-tests.html#cb40-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_density</span>()</span></code></pre></div>
<p><img src="_main_files/figure-html/null_model_two_stage-2.png" width="672" /></p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="association-tests.html#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="co"># compare residuals</span></span>
<span id="cb41-2"><a href="association-tests.html#cb41-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(pdat, <span class="fu">aes</span>(<span class="at">x =</span> resid.marginal.orig, <span class="at">y =</span> resid.marginal.twostage, <span class="at">color =</span> study)) <span class="sc">+</span> </span>
<span id="cb41-3"><a href="association-tests.html#cb41-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb41-4"><a href="association-tests.html#cb41-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_abline</span>(<span class="at">intercept =</span> <span class="dv">0</span>, <span class="at">slope =</span> <span class="dv">1</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/null_model_two_stage-3.png" width="672" /></p>
<p>See <a href="https://onlinelibrary.wiley.com/doi/10.1002/gepi.22188">Sofer et al.</a> for more information on the fully adjusted two-stage model.</p>
</div>
</div>
<div id="single-variant-tests" class="section level2" number="4.2">
<h2><span class="header-section-number">4.2</span> Single-variant tests</h2>
<p>After fitting our null model, we can run association tests to look for a
Now that we have a null model adjusting height for covariates, we can run an association test to look for genetic effects on height.</p>
<p>Single-variant tests are the same as in GWAS. We use the <code>assocTestSingle</code> function in GENESIS. First, we have to create a <code>SeqVarData</code> object including both the GDS file and the sample annotation containing phenotypes. We then create a <code>SeqVarBlockIterator</code> object to iterate over blocks of variants.</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="association-tests.html#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SeqVarTools)</span></code></pre></div>
<pre><code>## Loading required package: SeqArray</code></pre>
<pre><code>## Loading required package: gdsfmt</code></pre>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="association-tests.html#cb45-1" aria-hidden="true" tabindex="-1"></a>gdsfile <span class="ot">&lt;-</span> <span class="st">&quot;data/1KG_phase3_subset_chr1.gds&quot;</span></span>
<span id="cb45-2"><a href="association-tests.html#cb45-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(gdsfile)) <span class="fu">download.file</span>(<span class="fu">file.path</span>(repo_path, gdsfile), gdsfile)</span>
<span id="cb45-3"><a href="association-tests.html#cb45-3" aria-hidden="true" tabindex="-1"></a>gdsfmt<span class="sc">::</span><span class="fu">showfile.gds</span>(<span class="at">closeall=</span><span class="cn">TRUE</span>) <span class="co"># make sure file is not already open</span></span>
<span id="cb45-4"><a href="association-tests.html#cb45-4" aria-hidden="true" tabindex="-1"></a>gds <span class="ot">&lt;-</span> <span class="fu">seqOpen</span>(gdsfile)</span>
<span id="cb45-5"><a href="association-tests.html#cb45-5" aria-hidden="true" tabindex="-1"></a>seqData <span class="ot">&lt;-</span> <span class="fu">SeqVarData</span>(gds, <span class="at">sampleData=</span>annot)</span>
<span id="cb45-6"><a href="association-tests.html#cb45-6" aria-hidden="true" tabindex="-1"></a>iterator <span class="ot">&lt;-</span> <span class="fu">SeqVarBlockIterator</span>(seqData, <span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb45-7"><a href="association-tests.html#cb45-7" aria-hidden="true" tabindex="-1"></a>assoc <span class="ot">&lt;-</span> <span class="fu">assocTestSingle</span>(iterator, nullmod)</span></code></pre></div>
<pre><code>## # of selected samples: 1,126</code></pre>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="association-tests.html#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(assoc)</span></code></pre></div>
<pre><code>##   variant.id chr     pos allele.index n.obs         freq MAC      Score
## 1          1   1  970546            1  1126 0.0039964476   9 -0.1191236
## 2          2   1  985900            1  1126 0.0492895204 111 -1.6707553
## 3          3   1 1025045            1  1126 0.0004440497   1 -0.2795838
## 4          4   1 1265550            1  1126 0.0008880995   2 -0.1105487
## 5          5   1 1472676            1  1126 0.0071047957  16  0.3630992
## 6          6   1 1735725            1  1126 0.0022202487   5 -0.1300405
##    Score.SE Score.Stat  Score.pval        Est   Est.SE          PVE
## 1 0.2577712 -0.4621292 0.643988693  -1.792788 3.879410 0.0001905115
## 2 0.8841849 -1.8895996 0.058811535  -2.137109 1.130985 0.0031851797
## 3 0.1007173 -2.7759261 0.005504472 -27.561563 9.928781 0.0068740102
## 4 0.1085480 -1.0184319 0.308472744  -9.382319 9.212515 0.0009252485
## 5 0.3456555  1.0504657 0.293504065   3.039054 2.893054 0.0009843694
## 6 0.1973175 -0.6590420 0.509868790  -3.340007 5.067973 0.0003874544</code></pre>
<p>We make a QQ plot to examine the results.</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="association-tests.html#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb49-2"><a href="association-tests.html#cb49-2" aria-hidden="true" tabindex="-1"></a>qqPlot <span class="ot">&lt;-</span> <span class="cf">function</span>(pval) {</span>
<span id="cb49-3"><a href="association-tests.html#cb49-3" aria-hidden="true" tabindex="-1"></a>    pval <span class="ot">&lt;-</span> pval[<span class="sc">!</span><span class="fu">is.na</span>(pval)]</span>
<span id="cb49-4"><a href="association-tests.html#cb49-4" aria-hidden="true" tabindex="-1"></a>    n <span class="ot">&lt;-</span> <span class="fu">length</span>(pval)</span>
<span id="cb49-5"><a href="association-tests.html#cb49-5" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">:</span>n</span>
<span id="cb49-6"><a href="association-tests.html#cb49-6" aria-hidden="true" tabindex="-1"></a>    dat <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">obs=</span><span class="fu">sort</span>(pval),</span>
<span id="cb49-7"><a href="association-tests.html#cb49-7" aria-hidden="true" tabindex="-1"></a>                      <span class="at">exp=</span>x<span class="sc">/</span>n,</span>
<span id="cb49-8"><a href="association-tests.html#cb49-8" aria-hidden="true" tabindex="-1"></a>                      <span class="at">upper=</span><span class="fu">qbeta</span>(<span class="fl">0.025</span>, x, <span class="fu">rev</span>(x)),</span>
<span id="cb49-9"><a href="association-tests.html#cb49-9" aria-hidden="true" tabindex="-1"></a>                      <span class="at">lower=</span><span class="fu">qbeta</span>(<span class="fl">0.975</span>, x, <span class="fu">rev</span>(x)))</span>
<span id="cb49-10"><a href="association-tests.html#cb49-10" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb49-11"><a href="association-tests.html#cb49-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(dat, <span class="fu">aes</span>(<span class="sc">-</span><span class="fu">log10</span>(exp), <span class="sc">-</span><span class="fu">log10</span>(obs))) <span class="sc">+</span></span>
<span id="cb49-12"><a href="association-tests.html#cb49-12" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="sc">-</span><span class="fu">log10</span>(exp), <span class="sc">-</span><span class="fu">log10</span>(upper)), <span class="at">color=</span><span class="st">&quot;gray&quot;</span>) <span class="sc">+</span></span>
<span id="cb49-13"><a href="association-tests.html#cb49-13" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="sc">-</span><span class="fu">log10</span>(exp), <span class="sc">-</span><span class="fu">log10</span>(lower)), <span class="at">color=</span><span class="st">&quot;gray&quot;</span>) <span class="sc">+</span></span>
<span id="cb49-14"><a href="association-tests.html#cb49-14" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb49-15"><a href="association-tests.html#cb49-15" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_abline</span>(<span class="at">intercept=</span><span class="dv">0</span>, <span class="at">slope=</span><span class="dv">1</span>, <span class="at">color=</span><span class="st">&quot;red&quot;</span>) <span class="sc">+</span></span>
<span id="cb49-16"><a href="association-tests.html#cb49-16" aria-hidden="true" tabindex="-1"></a>        <span class="fu">xlab</span>(<span class="fu">expression</span>(<span class="fu">paste</span>(<span class="sc">-</span>log[<span class="dv">10</span>], <span class="st">&quot;(expected P)&quot;</span>))) <span class="sc">+</span></span>
<span id="cb49-17"><a href="association-tests.html#cb49-17" aria-hidden="true" tabindex="-1"></a>        <span class="fu">ylab</span>(<span class="fu">expression</span>(<span class="fu">paste</span>(<span class="sc">-</span>log[<span class="dv">10</span>], <span class="st">&quot;(observed P)&quot;</span>))) <span class="sc">+</span></span>
<span id="cb49-18"><a href="association-tests.html#cb49-18" aria-hidden="true" tabindex="-1"></a>        <span class="fu">theme_bw</span>()</span>
<span id="cb49-19"><a href="association-tests.html#cb49-19" aria-hidden="true" tabindex="-1"></a>}    </span>
<span id="cb49-20"><a href="association-tests.html#cb49-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-21"><a href="association-tests.html#cb49-21" aria-hidden="true" tabindex="-1"></a><span class="fu">qqPlot</span>(assoc<span class="sc">$</span>Score.pval)</span></code></pre></div>
<p><img src="_main_files/figure-html/assoc_single_qq-1.png" width="672" /></p>
</div>
<div id="exercises-1" class="section level2" number="4.3">
<h2><span class="header-section-number">4.3</span> Exercises</h2>
<ol style="list-style-type: decimal">
<li>Logistic regression: <code>fitNullModel</code> can use a binary phenotype as the outcome variable by specifying the argument <code>family=binomial</code>. Use the <code>status</code> column in the sample annotation to fit a null model for simulated case/control status, with <code>sex</code> and <code>Population</code> as covariates. Then run a single-variant test using this model.</li>
</ol>
</div>
<div id="sliding-window-tests" class="section level2" number="4.4">
<h2><span class="header-section-number">4.4</span> Sliding window tests</h2>
<p>For rare variants, we can do burden tests or SKAT using the GENESIS function <code>assocTestAggregate</code>. We restrict the test to variants with alternate allele frequency &lt; 0.1. (For real data, this threshold would be lower.) We use a flat weighting scheme. We define a sliding window across the genome using a <code>SeqVarWindowIterator</code>.</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="association-tests.html#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="fu">seqResetFilter</span>(seqData, <span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb50-2"><a href="association-tests.html#cb50-2" aria-hidden="true" tabindex="-1"></a>iterator <span class="ot">&lt;-</span> <span class="fu">SeqVarWindowIterator</span>(seqData, <span class="at">windowSize=</span><span class="dv">5000</span>, <span class="at">windowShift=</span><span class="dv">2000</span>, <span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb50-3"><a href="association-tests.html#cb50-3" aria-hidden="true" tabindex="-1"></a>assoc <span class="ot">&lt;-</span> <span class="fu">assocTestAggregate</span>(iterator, nullmod, <span class="at">test=</span><span class="st">&quot;Burden&quot;</span>, <span class="at">AF.max=</span><span class="fl">0.1</span>, <span class="at">weight.beta=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>))</span></code></pre></div>
<pre><code>## # of selected samples: 1,126</code></pre>
<pre><code>## Iteration 12 of 1127 completed</code></pre>
<pre><code>## Iteration 24 of 1127 completed</code></pre>
<pre><code>## Iteration 36 of 1127 completed</code></pre>
<pre><code>## Iteration 48 of 1127 completed</code></pre>
<pre><code>## Iteration 60 of 1127 completed</code></pre>
<pre><code>## Iteration 72 of 1127 completed</code></pre>
<pre><code>## Iteration 84 of 1127 completed</code></pre>
<pre><code>## Iteration 96 of 1127 completed</code></pre>
<pre><code>## Iteration 108 of 1127 completed</code></pre>
<pre><code>## Iteration 120 of 1127 completed</code></pre>
<pre><code>## Iteration 132 of 1127 completed</code></pre>
<pre><code>## Iteration 144 of 1127 completed</code></pre>
<pre><code>## Iteration 156 of 1127 completed</code></pre>
<pre><code>## Iteration 168 of 1127 completed</code></pre>
<pre><code>## Iteration 180 of 1127 completed</code></pre>
<pre><code>## Iteration 192 of 1127 completed</code></pre>
<pre><code>## Iteration 204 of 1127 completed</code></pre>
<pre><code>## Iteration 216 of 1127 completed</code></pre>
<pre><code>## Iteration 228 of 1127 completed</code></pre>
<pre><code>## Iteration 240 of 1127 completed</code></pre>
<pre><code>## Iteration 252 of 1127 completed</code></pre>
<pre><code>## Iteration 264 of 1127 completed</code></pre>
<pre><code>## Iteration 276 of 1127 completed</code></pre>
<pre><code>## Iteration 288 of 1127 completed</code></pre>
<pre><code>## Iteration 300 of 1127 completed</code></pre>
<pre><code>## Iteration 312 of 1127 completed</code></pre>
<pre><code>## Iteration 324 of 1127 completed</code></pre>
<pre><code>## Iteration 336 of 1127 completed</code></pre>
<pre><code>## Iteration 348 of 1127 completed</code></pre>
<pre><code>## Iteration 360 of 1127 completed</code></pre>
<pre><code>## Iteration 372 of 1127 completed</code></pre>
<pre><code>## Iteration 384 of 1127 completed</code></pre>
<pre><code>## Iteration 396 of 1127 completed</code></pre>
<pre><code>## Iteration 408 of 1127 completed</code></pre>
<pre><code>## Iteration 420 of 1127 completed</code></pre>
<pre><code>## Iteration 432 of 1127 completed</code></pre>
<pre><code>## Iteration 444 of 1127 completed</code></pre>
<pre><code>## Iteration 456 of 1127 completed</code></pre>
<pre><code>## Iteration 468 of 1127 completed</code></pre>
<pre><code>## Iteration 480 of 1127 completed</code></pre>
<pre><code>## Iteration 492 of 1127 completed</code></pre>
<pre><code>## Iteration 504 of 1127 completed</code></pre>
<pre><code>## Iteration 516 of 1127 completed</code></pre>
<pre><code>## Iteration 528 of 1127 completed</code></pre>
<pre><code>## Iteration 540 of 1127 completed</code></pre>
<pre><code>## Iteration 552 of 1127 completed</code></pre>
<pre><code>## Iteration 564 of 1127 completed</code></pre>
<pre><code>## Iteration 576 of 1127 completed</code></pre>
<pre><code>## Iteration 588 of 1127 completed</code></pre>
<pre><code>## Iteration 600 of 1127 completed</code></pre>
<pre><code>## Iteration 612 of 1127 completed</code></pre>
<pre><code>## Iteration 624 of 1127 completed</code></pre>
<pre><code>## Iteration 636 of 1127 completed</code></pre>
<pre><code>## Iteration 648 of 1127 completed</code></pre>
<pre><code>## Iteration 660 of 1127 completed</code></pre>
<pre><code>## Iteration 672 of 1127 completed</code></pre>
<pre><code>## Iteration 684 of 1127 completed</code></pre>
<pre><code>## Iteration 696 of 1127 completed</code></pre>
<pre><code>## Iteration 708 of 1127 completed</code></pre>
<pre><code>## Iteration 720 of 1127 completed</code></pre>
<pre><code>## Iteration 732 of 1127 completed</code></pre>
<pre><code>## Iteration 744 of 1127 completed</code></pre>
<pre><code>## Iteration 756 of 1127 completed</code></pre>
<pre><code>## Iteration 768 of 1127 completed</code></pre>
<pre><code>## Iteration 780 of 1127 completed</code></pre>
<pre><code>## Iteration 792 of 1127 completed</code></pre>
<pre><code>## Iteration 804 of 1127 completed</code></pre>
<pre><code>## Iteration 816 of 1127 completed</code></pre>
<pre><code>## Iteration 828 of 1127 completed</code></pre>
<pre><code>## Iteration 840 of 1127 completed</code></pre>
<pre><code>## Iteration 852 of 1127 completed</code></pre>
<pre><code>## Iteration 864 of 1127 completed</code></pre>
<pre><code>## Iteration 876 of 1127 completed</code></pre>
<pre><code>## Iteration 888 of 1127 completed</code></pre>
<pre><code>## Iteration 900 of 1127 completed</code></pre>
<pre><code>## Iteration 912 of 1127 completed</code></pre>
<pre><code>## Iteration 924 of 1127 completed</code></pre>
<pre><code>## Iteration 936 of 1127 completed</code></pre>
<pre><code>## Iteration 948 of 1127 completed</code></pre>
<pre><code>## Iteration 960 of 1127 completed</code></pre>
<pre><code>## Iteration 972 of 1127 completed</code></pre>
<pre><code>## Iteration 984 of 1127 completed</code></pre>
<pre><code>## Iteration 996 of 1127 completed</code></pre>
<pre><code>## Iteration 1008 of 1127 completed</code></pre>
<pre><code>## Iteration 1020 of 1127 completed</code></pre>
<pre><code>## Iteration 1032 of 1127 completed</code></pre>
<pre><code>## Iteration 1044 of 1127 completed</code></pre>
<pre><code>## Iteration 1056 of 1127 completed</code></pre>
<pre><code>## Iteration 1068 of 1127 completed</code></pre>
<pre><code>## Iteration 1080 of 1127 completed</code></pre>
<pre><code>## Iteration 1092 of 1127 completed</code></pre>
<pre><code>## Iteration 1104 of 1127 completed</code></pre>
<pre><code>## Iteration 1116 of 1127 completed</code></pre>
<div class="sourceCode" id="cb145"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb145-1"><a href="association-tests.html#cb145-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(assoc)</span></code></pre></div>
<pre><code>## [1] &quot;results&quot;     &quot;variantInfo&quot;</code></pre>
<div class="sourceCode" id="cb147"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb147-1"><a href="association-tests.html#cb147-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(assoc<span class="sc">$</span>results)</span></code></pre></div>
<pre><code>##   chr   start     end n.site n.alt n.sample.alt      Score  Score.SE Score.Stat
## 1   1  966001  971000      1     9            9 -0.1191236 0.2577712 -0.4621292
## 2   1  982001  987000      1   111          107 -1.6707553 0.8841849 -1.8895996
## 3   1 1022001 1027000      1     1            1 -0.2795838 0.1007173 -2.7759261
## 4   1 1262001 1267000      1     2            2 -0.1105487 0.1085480 -1.0184319
## 5   1 1468001 1473000      1    16           16  0.3630992 0.3456555  1.0504657
## 6   1 1732001 1737000      1     5            5 -0.1300405 0.1973175 -0.6590420
##    Score.pval        Est   Est.SE          PVE
## 1 0.643988693  -1.792788 3.879410 0.0001905115
## 2 0.058811535  -2.137109 1.130985 0.0031851797
## 3 0.005504472 -27.561563 9.928781 0.0068740102
## 4 0.308472744  -9.382319 9.212515 0.0009252485
## 5 0.293504065   3.039054 2.893054 0.0009843694
## 6 0.509868790  -3.340007 5.067973 0.0003874544</code></pre>
<div class="sourceCode" id="cb149"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb149-1"><a href="association-tests.html#cb149-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(assoc<span class="sc">$</span>variantInfo)</span></code></pre></div>
<pre><code>## [[1]]
##   variant.id chr    pos allele.index n.obs        freq MAC weight
## 1          1   1 970546            1  1126 0.003996448   9      1
## 
## [[2]]
##   variant.id chr    pos allele.index n.obs       freq MAC weight
## 1          2   1 985900            1  1126 0.04928952 111      1
## 
## [[3]]
##   variant.id chr     pos allele.index n.obs         freq MAC weight
## 1          3   1 1025045            1  1126 0.0004440497   1      1
## 
## [[4]]
##   variant.id chr     pos allele.index n.obs         freq MAC weight
## 1          4   1 1265550            1  1126 0.0008880995   2      1
## 
## [[5]]
##   variant.id chr     pos allele.index n.obs        freq MAC weight
## 1          5   1 1472676            1  1126 0.007104796  16      1
## 
## [[6]]
##   variant.id chr     pos allele.index n.obs        freq MAC weight
## 1          6   1 1735725            1  1126 0.002220249   5      1</code></pre>
<div class="sourceCode" id="cb151"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb151-1"><a href="association-tests.html#cb151-1" aria-hidden="true" tabindex="-1"></a><span class="fu">qqPlot</span>(assoc<span class="sc">$</span>results<span class="sc">$</span>Score.pval)</span></code></pre></div>
<p><img src="_main_files/figure-html/assoc_window_burden-1.png" width="672" /></p>
<p>For SKAT, we use the Wu weights.</p>
<div class="sourceCode" id="cb152"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb152-1"><a href="association-tests.html#cb152-1" aria-hidden="true" tabindex="-1"></a><span class="fu">seqResetFilter</span>(seqData, <span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb152-2"><a href="association-tests.html#cb152-2" aria-hidden="true" tabindex="-1"></a>iterator <span class="ot">&lt;-</span> <span class="fu">SeqVarWindowIterator</span>(seqData, <span class="at">windowSize=</span><span class="dv">5000</span>, <span class="at">windowShift=</span><span class="dv">2000</span>, <span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb152-3"><a href="association-tests.html#cb152-3" aria-hidden="true" tabindex="-1"></a>assoc <span class="ot">&lt;-</span> <span class="fu">assocTestAggregate</span>(iterator, nullmod, <span class="at">test=</span><span class="st">&quot;SKAT&quot;</span>, <span class="at">AF.max=</span><span class="fl">0.1</span>, <span class="at">weight.beta=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">25</span>))</span></code></pre></div>
<pre><code>## # of selected samples: 1,126</code></pre>
<pre><code>## Loading required namespace: survey</code></pre>
<pre><code>## Loading required namespace: CompQuadForm</code></pre>
<pre><code>## Iteration 12 of 1127 completed</code></pre>
<pre><code>## Iteration 24 of 1127 completed</code></pre>
<pre><code>## Iteration 36 of 1127 completed</code></pre>
<pre><code>## Iteration 48 of 1127 completed</code></pre>
<pre><code>## Iteration 60 of 1127 completed</code></pre>
<pre><code>## Iteration 72 of 1127 completed</code></pre>
<pre><code>## Iteration 84 of 1127 completed</code></pre>
<pre><code>## Iteration 96 of 1127 completed</code></pre>
<pre><code>## Iteration 108 of 1127 completed</code></pre>
<pre><code>## Iteration 120 of 1127 completed</code></pre>
<pre><code>## Iteration 132 of 1127 completed</code></pre>
<pre><code>## Iteration 144 of 1127 completed</code></pre>
<pre><code>## Iteration 156 of 1127 completed</code></pre>
<pre><code>## Iteration 168 of 1127 completed</code></pre>
<pre><code>## Iteration 180 of 1127 completed</code></pre>
<pre><code>## Iteration 192 of 1127 completed</code></pre>
<pre><code>## Iteration 204 of 1127 completed</code></pre>
<pre><code>## Iteration 216 of 1127 completed</code></pre>
<pre><code>## Iteration 228 of 1127 completed</code></pre>
<pre><code>## Iteration 240 of 1127 completed</code></pre>
<pre><code>## Iteration 252 of 1127 completed</code></pre>
<pre><code>## Iteration 264 of 1127 completed</code></pre>
<pre><code>## Iteration 276 of 1127 completed</code></pre>
<pre><code>## Iteration 288 of 1127 completed</code></pre>
<pre><code>## Iteration 300 of 1127 completed</code></pre>
<pre><code>## Iteration 312 of 1127 completed</code></pre>
<pre><code>## Iteration 324 of 1127 completed</code></pre>
<pre><code>## Iteration 336 of 1127 completed</code></pre>
<pre><code>## Iteration 348 of 1127 completed</code></pre>
<pre><code>## Iteration 360 of 1127 completed</code></pre>
<pre><code>## Iteration 372 of 1127 completed</code></pre>
<pre><code>## Iteration 384 of 1127 completed</code></pre>
<pre><code>## Iteration 396 of 1127 completed</code></pre>
<pre><code>## Iteration 408 of 1127 completed</code></pre>
<pre><code>## Iteration 420 of 1127 completed</code></pre>
<pre><code>## Iteration 432 of 1127 completed</code></pre>
<pre><code>## Iteration 444 of 1127 completed</code></pre>
<pre><code>## Iteration 456 of 1127 completed</code></pre>
<pre><code>## Iteration 468 of 1127 completed</code></pre>
<pre><code>## Iteration 480 of 1127 completed</code></pre>
<pre><code>## Iteration 492 of 1127 completed</code></pre>
<pre><code>## Iteration 504 of 1127 completed</code></pre>
<pre><code>## Iteration 516 of 1127 completed</code></pre>
<pre><code>## Iteration 528 of 1127 completed</code></pre>
<pre><code>## Iteration 540 of 1127 completed</code></pre>
<pre><code>## Iteration 552 of 1127 completed</code></pre>
<pre><code>## Iteration 564 of 1127 completed</code></pre>
<pre><code>## Iteration 576 of 1127 completed</code></pre>
<pre><code>## Iteration 588 of 1127 completed</code></pre>
<pre><code>## Iteration 600 of 1127 completed</code></pre>
<pre><code>## Iteration 612 of 1127 completed</code></pre>
<pre><code>## Iteration 624 of 1127 completed</code></pre>
<pre><code>## Iteration 636 of 1127 completed</code></pre>
<pre><code>## Iteration 648 of 1127 completed</code></pre>
<pre><code>## Iteration 660 of 1127 completed</code></pre>
<pre><code>## Iteration 672 of 1127 completed</code></pre>
<pre><code>## Iteration 684 of 1127 completed</code></pre>
<pre><code>## Iteration 696 of 1127 completed</code></pre>
<pre><code>## Iteration 708 of 1127 completed</code></pre>
<pre><code>## Iteration 720 of 1127 completed</code></pre>
<pre><code>## Iteration 732 of 1127 completed</code></pre>
<pre><code>## Iteration 744 of 1127 completed</code></pre>
<pre><code>## Iteration 756 of 1127 completed</code></pre>
<pre><code>## Iteration 768 of 1127 completed</code></pre>
<pre><code>## Iteration 780 of 1127 completed</code></pre>
<pre><code>## Iteration 792 of 1127 completed</code></pre>
<pre><code>## Iteration 804 of 1127 completed</code></pre>
<pre><code>## Iteration 816 of 1127 completed</code></pre>
<pre><code>## Iteration 828 of 1127 completed</code></pre>
<pre><code>## Iteration 840 of 1127 completed</code></pre>
<pre><code>## Iteration 852 of 1127 completed</code></pre>
<pre><code>## Iteration 864 of 1127 completed</code></pre>
<pre><code>## Iteration 876 of 1127 completed</code></pre>
<pre><code>## Iteration 888 of 1127 completed</code></pre>
<pre><code>## Iteration 900 of 1127 completed</code></pre>
<pre><code>## Iteration 912 of 1127 completed</code></pre>
<pre><code>## Iteration 924 of 1127 completed</code></pre>
<pre><code>## Iteration 936 of 1127 completed</code></pre>
<pre><code>## Iteration 948 of 1127 completed</code></pre>
<pre><code>## Iteration 960 of 1127 completed</code></pre>
<pre><code>## Iteration 972 of 1127 completed</code></pre>
<pre><code>## Iteration 984 of 1127 completed</code></pre>
<pre><code>## Iteration 996 of 1127 completed</code></pre>
<pre><code>## Iteration 1008 of 1127 completed</code></pre>
<pre><code>## Iteration 1020 of 1127 completed</code></pre>
<pre><code>## Iteration 1032 of 1127 completed</code></pre>
<pre><code>## Iteration 1044 of 1127 completed</code></pre>
<pre><code>## Iteration 1056 of 1127 completed</code></pre>
<pre><code>## Iteration 1068 of 1127 completed</code></pre>
<pre><code>## Iteration 1080 of 1127 completed</code></pre>
<pre><code>## Iteration 1092 of 1127 completed</code></pre>
<pre><code>## Iteration 1104 of 1127 completed</code></pre>
<pre><code>## Iteration 1116 of 1127 completed</code></pre>
<div class="sourceCode" id="cb249"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb249-1"><a href="association-tests.html#cb249-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(assoc<span class="sc">$</span>results)</span></code></pre></div>
<pre><code>##   chr   start     end n.site n.alt n.sample.alt          Q        pval err
## 1   1  966001  971000      1     9            9   7.318095 0.643988693   0
## 2   1  982001  987000      1   111          107 154.178289 0.058811535   0
## 3   1 1022001 1027000      1     1            1  47.823918 0.005504472   0
## 4   1 1262001 1267000      1     2            2   7.319239 0.308472744   0
## 5   1 1468001 1473000      1    16           16  58.518665 0.293504065   0
## 6   1 1732001 1737000      1     5            5   9.499539 0.509868790   0
##   pval.method
## 1 integration
## 2 integration
## 3 integration
## 4 integration
## 5 integration
## 6 integration</code></pre>
<div class="sourceCode" id="cb251"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb251-1"><a href="association-tests.html#cb251-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(assoc<span class="sc">$</span>variantInfo)</span></code></pre></div>
<pre><code>## [[1]]
##   variant.id chr    pos allele.index n.obs        freq MAC   weight
## 1          1   1 970546            1  1126 0.003996448   9 22.70917
## 
## [[2]]
##   variant.id chr    pos allele.index n.obs       freq MAC   weight
## 1          2   1 985900            1  1126 0.04928952 111 7.431881
## 
## [[3]]
##   variant.id chr     pos allele.index n.obs         freq MAC   weight
## 1          3   1 1025045            1  1126 0.0004440497   1 24.73493
## 
## [[4]]
##   variant.id chr     pos allele.index n.obs         freq MAC   weight
## 1          4   1 1265550            1  1126 0.0008880995   2 24.47255
## 
## [[5]]
##   variant.id chr     pos allele.index n.obs        freq MAC   weight
## 1          5   1 1472676            1  1126 0.007104796  16 21.06793
## 
## [[6]]
##   variant.id chr     pos allele.index n.obs        freq MAC   weight
## 1          6   1 1735725            1  1126 0.002220249   5 23.70132</code></pre>
<div class="sourceCode" id="cb253"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb253-1"><a href="association-tests.html#cb253-1" aria-hidden="true" tabindex="-1"></a><span class="fu">qqPlot</span>(assoc<span class="sc">$</span>results<span class="sc">$</span>pval)</span></code></pre></div>
<p><img src="_main_files/figure-html/assoc_window_skat-1.png" width="672" /></p>
</div>
<div id="exercise-1" class="section level2" number="4.5">
<h2><span class="header-section-number">4.5</span> Exercise</h2>
<ol start="3" style="list-style-type: decimal">
<li>Repeat the previous exercise on logistic regression, this time running a sliding-window test.</li>
</ol>
<div id="association-tests---solutions" class="section level3" number="4.5.1">
<h3><span class="header-section-number">4.5.1</span> Association tests - Solutions</h3>
<ol style="list-style-type: decimal">
<li>Logistic regression: <code>fitNullModel</code> can use a binary phenotype as the outcome variable by specifying the argument <code>family=binomial</code>. Use the <code>status</code> column in the sample annotation to fit a null model for simulated case/control status, with <code>sex</code> and <code>Population</code> as covariates. Then run a single-variant test using this model.</li>
</ol>
<div class="sourceCode" id="cb254"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb254-1"><a href="association-tests.html#cb254-1" aria-hidden="true" tabindex="-1"></a>nullmod.status <span class="ot">&lt;-</span> <span class="fu">fitNullModel</span>(annot, <span class="at">outcome=</span><span class="st">&quot;status&quot;</span>, <span class="at">covars=</span><span class="fu">c</span>(<span class="st">&quot;sex&quot;</span>, <span class="st">&quot;Population&quot;</span>), </span>
<span id="cb254-2"><a href="association-tests.html#cb254-2" aria-hidden="true" tabindex="-1"></a>                               <span class="at">family=</span>binomial, <span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb254-3"><a href="association-tests.html#cb254-3" aria-hidden="true" tabindex="-1"></a><span class="fu">resetIterator</span>(iterator, <span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb254-4"><a href="association-tests.html#cb254-4" aria-hidden="true" tabindex="-1"></a>assoc <span class="ot">&lt;-</span> <span class="fu">assocTestSingle</span>(iterator, nullmod.status, <span class="at">test=</span><span class="st">&quot;Score&quot;</span>)</span></code></pre></div>
<pre><code>## # of selected samples: 1,126</code></pre>
<pre><code>## Iteration 12 of 1127 completed</code></pre>
<pre><code>## Iteration 24 of 1127 completed</code></pre>
<pre><code>## Iteration 36 of 1127 completed</code></pre>
<pre><code>## Iteration 48 of 1127 completed</code></pre>
<pre><code>## Iteration 60 of 1127 completed</code></pre>
<pre><code>## Iteration 72 of 1127 completed</code></pre>
<pre><code>## Iteration 84 of 1127 completed</code></pre>
<pre><code>## Iteration 96 of 1127 completed</code></pre>
<pre><code>## Iteration 108 of 1127 completed</code></pre>
<pre><code>## Iteration 120 of 1127 completed</code></pre>
<pre><code>## Iteration 132 of 1127 completed</code></pre>
<pre><code>## Iteration 144 of 1127 completed</code></pre>
<pre><code>## Iteration 156 of 1127 completed</code></pre>
<pre><code>## Iteration 168 of 1127 completed</code></pre>
<pre><code>## Iteration 180 of 1127 completed</code></pre>
<pre><code>## Iteration 192 of 1127 completed</code></pre>
<pre><code>## Iteration 204 of 1127 completed</code></pre>
<pre><code>## Iteration 216 of 1127 completed</code></pre>
<pre><code>## Iteration 228 of 1127 completed</code></pre>
<pre><code>## Iteration 240 of 1127 completed</code></pre>
<pre><code>## Iteration 252 of 1127 completed</code></pre>
<pre><code>## Iteration 264 of 1127 completed</code></pre>
<pre><code>## Iteration 276 of 1127 completed</code></pre>
<pre><code>## Iteration 288 of 1127 completed</code></pre>
<pre><code>## Iteration 300 of 1127 completed</code></pre>
<pre><code>## Iteration 312 of 1127 completed</code></pre>
<pre><code>## Iteration 324 of 1127 completed</code></pre>
<pre><code>## Iteration 336 of 1127 completed</code></pre>
<pre><code>## Iteration 348 of 1127 completed</code></pre>
<pre><code>## Iteration 360 of 1127 completed</code></pre>
<pre><code>## Iteration 372 of 1127 completed</code></pre>
<pre><code>## Iteration 384 of 1127 completed</code></pre>
<pre><code>## Iteration 396 of 1127 completed</code></pre>
<pre><code>## Iteration 408 of 1127 completed</code></pre>
<pre><code>## Iteration 420 of 1127 completed</code></pre>
<pre><code>## Iteration 432 of 1127 completed</code></pre>
<pre><code>## Iteration 444 of 1127 completed</code></pre>
<pre><code>## Iteration 456 of 1127 completed</code></pre>
<pre><code>## Iteration 468 of 1127 completed</code></pre>
<pre><code>## Iteration 480 of 1127 completed</code></pre>
<pre><code>## Iteration 492 of 1127 completed</code></pre>
<pre><code>## Iteration 504 of 1127 completed</code></pre>
<pre><code>## Iteration 516 of 1127 completed</code></pre>
<pre><code>## Iteration 528 of 1127 completed</code></pre>
<pre><code>## Iteration 540 of 1127 completed</code></pre>
<pre><code>## Iteration 552 of 1127 completed</code></pre>
<pre><code>## Iteration 564 of 1127 completed</code></pre>
<pre><code>## Iteration 576 of 1127 completed</code></pre>
<pre><code>## Iteration 588 of 1127 completed</code></pre>
<pre><code>## Iteration 600 of 1127 completed</code></pre>
<pre><code>## Iteration 612 of 1127 completed</code></pre>
<pre><code>## Iteration 624 of 1127 completed</code></pre>
<pre><code>## Iteration 636 of 1127 completed</code></pre>
<pre><code>## Iteration 648 of 1127 completed</code></pre>
<pre><code>## Iteration 660 of 1127 completed</code></pre>
<pre><code>## Iteration 672 of 1127 completed</code></pre>
<pre><code>## Iteration 684 of 1127 completed</code></pre>
<pre><code>## Iteration 696 of 1127 completed</code></pre>
<pre><code>## Iteration 708 of 1127 completed</code></pre>
<pre><code>## Iteration 720 of 1127 completed</code></pre>
<pre><code>## Iteration 732 of 1127 completed</code></pre>
<pre><code>## Iteration 744 of 1127 completed</code></pre>
<pre><code>## Iteration 756 of 1127 completed</code></pre>
<pre><code>## Iteration 768 of 1127 completed</code></pre>
<pre><code>## Iteration 780 of 1127 completed</code></pre>
<pre><code>## Iteration 792 of 1127 completed</code></pre>
<pre><code>## Iteration 804 of 1127 completed</code></pre>
<pre><code>## Iteration 816 of 1127 completed</code></pre>
<pre><code>## Iteration 828 of 1127 completed</code></pre>
<pre><code>## Iteration 840 of 1127 completed</code></pre>
<pre><code>## Iteration 852 of 1127 completed</code></pre>
<pre><code>## Iteration 864 of 1127 completed</code></pre>
<pre><code>## Iteration 876 of 1127 completed</code></pre>
<pre><code>## Iteration 888 of 1127 completed</code></pre>
<pre><code>## Iteration 900 of 1127 completed</code></pre>
<pre><code>## Iteration 912 of 1127 completed</code></pre>
<pre><code>## Iteration 924 of 1127 completed</code></pre>
<pre><code>## Iteration 936 of 1127 completed</code></pre>
<pre><code>## Iteration 948 of 1127 completed</code></pre>
<pre><code>## Iteration 960 of 1127 completed</code></pre>
<pre><code>## Iteration 972 of 1127 completed</code></pre>
<pre><code>## Iteration 984 of 1127 completed</code></pre>
<pre><code>## Iteration 996 of 1127 completed</code></pre>
<pre><code>## Iteration 1008 of 1127 completed</code></pre>
<pre><code>## Iteration 1020 of 1127 completed</code></pre>
<pre><code>## Iteration 1032 of 1127 completed</code></pre>
<pre><code>## Iteration 1044 of 1127 completed</code></pre>
<pre><code>## Iteration 1056 of 1127 completed</code></pre>
<pre><code>## Iteration 1068 of 1127 completed</code></pre>
<pre><code>## Iteration 1080 of 1127 completed</code></pre>
<pre><code>## Iteration 1092 of 1127 completed</code></pre>
<pre><code>## Iteration 1104 of 1127 completed</code></pre>
<pre><code>## Iteration 1116 of 1127 completed</code></pre>
<div class="sourceCode" id="cb349"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb349-1"><a href="association-tests.html#cb349-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(assoc)</span></code></pre></div>
<pre><code>##   variant.id chr     pos allele.index n.obs         freq MAC       Score
## 1          1   1  970546            1  1126 0.0039964476   9  0.20256722
## 2          2   1  985900            1  1126 0.0492895204 111 -2.64169956
## 3          3   1 1025045            1  1126 0.0004440497   1 -0.09916904
## 4          4   1 1265550            1  1126 0.0008880995   2  0.81717324
## 5          5   1 1472676            1  1126 0.0071047957  16  0.64418361
## 6          6   1 1735725            1  1126 0.0022202487   5 -0.46319177
##    Score.SE Score.Stat Score.pval        Est    Est.SE          PVE
## 1 0.8351783  0.2425437 0.80835892  0.2904095 1.1973491 5.242118e-05
## 2 2.6522412 -0.9960254 0.31923781 -0.3755410 0.3770396 8.840314e-04
## 3 0.2972472 -0.3336248 0.73866267 -1.1223819 3.3642035 9.918446e-05
## 4 0.4033577  2.0259271 0.04277226  5.0226565 2.4791892 3.657417e-03
## 5 1.0778277  0.5976685 0.55006117  0.5545121 0.9277921 3.183080e-04
## 6 0.6396675 -0.7241134 0.46899613 -1.1320153 1.5633123 4.672400e-04</code></pre>
<ol start="3" style="list-style-type: decimal">
<li>Repeat the previous exercise on logistic regression, this time running a sliding-window test.</li>
</ol>
<div class="sourceCode" id="cb351"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb351-1"><a href="association-tests.html#cb351-1" aria-hidden="true" tabindex="-1"></a>nullmod.status <span class="ot">&lt;-</span> <span class="fu">fitNullModel</span>(annot, <span class="at">outcome=</span><span class="st">&quot;status&quot;</span>, <span class="at">covars=</span><span class="fu">c</span>(<span class="st">&quot;sex&quot;</span>, <span class="st">&quot;Population&quot;</span>), </span>
<span id="cb351-2"><a href="association-tests.html#cb351-2" aria-hidden="true" tabindex="-1"></a>                               <span class="at">family=</span>binomial, <span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb351-3"><a href="association-tests.html#cb351-3" aria-hidden="true" tabindex="-1"></a><span class="fu">seqResetFilter</span>(seqData, <span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb351-4"><a href="association-tests.html#cb351-4" aria-hidden="true" tabindex="-1"></a>iterator <span class="ot">&lt;-</span> <span class="fu">SeqVarWindowIterator</span>(seqData, <span class="at">windowSize=</span><span class="dv">5000</span>, <span class="at">windowShift=</span><span class="dv">2000</span>, <span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb351-5"><a href="association-tests.html#cb351-5" aria-hidden="true" tabindex="-1"></a>assoc <span class="ot">&lt;-</span> <span class="fu">assocTestAggregate</span>(iterator, nullmod, <span class="at">test=</span><span class="st">&quot;SKAT&quot;</span>, <span class="at">AF.max=</span><span class="fl">0.1</span>, <span class="at">weight.beta=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">25</span>))</span></code></pre></div>
<pre><code>## # of selected samples: 1,126</code></pre>
<pre><code>## Iteration 12 of 1127 completed</code></pre>
<pre><code>## Iteration 24 of 1127 completed</code></pre>
<pre><code>## Iteration 36 of 1127 completed</code></pre>
<pre><code>## Iteration 48 of 1127 completed</code></pre>
<pre><code>## Iteration 60 of 1127 completed</code></pre>
<pre><code>## Iteration 72 of 1127 completed</code></pre>
<pre><code>## Iteration 84 of 1127 completed</code></pre>
<pre><code>## Iteration 96 of 1127 completed</code></pre>
<pre><code>## Iteration 108 of 1127 completed</code></pre>
<pre><code>## Iteration 120 of 1127 completed</code></pre>
<pre><code>## Iteration 132 of 1127 completed</code></pre>
<pre><code>## Iteration 144 of 1127 completed</code></pre>
<pre><code>## Iteration 156 of 1127 completed</code></pre>
<pre><code>## Iteration 168 of 1127 completed</code></pre>
<pre><code>## Iteration 180 of 1127 completed</code></pre>
<pre><code>## Iteration 192 of 1127 completed</code></pre>
<pre><code>## Iteration 204 of 1127 completed</code></pre>
<pre><code>## Iteration 216 of 1127 completed</code></pre>
<pre><code>## Iteration 228 of 1127 completed</code></pre>
<pre><code>## Iteration 240 of 1127 completed</code></pre>
<pre><code>## Iteration 252 of 1127 completed</code></pre>
<pre><code>## Iteration 264 of 1127 completed</code></pre>
<pre><code>## Iteration 276 of 1127 completed</code></pre>
<pre><code>## Iteration 288 of 1127 completed</code></pre>
<pre><code>## Iteration 300 of 1127 completed</code></pre>
<pre><code>## Iteration 312 of 1127 completed</code></pre>
<pre><code>## Iteration 324 of 1127 completed</code></pre>
<pre><code>## Iteration 336 of 1127 completed</code></pre>
<pre><code>## Iteration 348 of 1127 completed</code></pre>
<pre><code>## Iteration 360 of 1127 completed</code></pre>
<pre><code>## Iteration 372 of 1127 completed</code></pre>
<pre><code>## Iteration 384 of 1127 completed</code></pre>
<pre><code>## Iteration 396 of 1127 completed</code></pre>
<pre><code>## Iteration 408 of 1127 completed</code></pre>
<pre><code>## Iteration 420 of 1127 completed</code></pre>
<pre><code>## Iteration 432 of 1127 completed</code></pre>
<pre><code>## Iteration 444 of 1127 completed</code></pre>
<pre><code>## Iteration 456 of 1127 completed</code></pre>
<pre><code>## Iteration 468 of 1127 completed</code></pre>
<pre><code>## Iteration 480 of 1127 completed</code></pre>
<pre><code>## Iteration 492 of 1127 completed</code></pre>
<pre><code>## Iteration 504 of 1127 completed</code></pre>
<pre><code>## Iteration 516 of 1127 completed</code></pre>
<pre><code>## Iteration 528 of 1127 completed</code></pre>
<pre><code>## Iteration 540 of 1127 completed</code></pre>
<pre><code>## Iteration 552 of 1127 completed</code></pre>
<pre><code>## Iteration 564 of 1127 completed</code></pre>
<pre><code>## Iteration 576 of 1127 completed</code></pre>
<pre><code>## Iteration 588 of 1127 completed</code></pre>
<pre><code>## Iteration 600 of 1127 completed</code></pre>
<pre><code>## Iteration 612 of 1127 completed</code></pre>
<pre><code>## Iteration 624 of 1127 completed</code></pre>
<pre><code>## Iteration 636 of 1127 completed</code></pre>
<pre><code>## Iteration 648 of 1127 completed</code></pre>
<pre><code>## Iteration 660 of 1127 completed</code></pre>
<pre><code>## Iteration 672 of 1127 completed</code></pre>
<pre><code>## Iteration 684 of 1127 completed</code></pre>
<pre><code>## Iteration 696 of 1127 completed</code></pre>
<pre><code>## Iteration 708 of 1127 completed</code></pre>
<pre><code>## Iteration 720 of 1127 completed</code></pre>
<pre><code>## Iteration 732 of 1127 completed</code></pre>
<pre><code>## Iteration 744 of 1127 completed</code></pre>
<pre><code>## Iteration 756 of 1127 completed</code></pre>
<pre><code>## Iteration 768 of 1127 completed</code></pre>
<pre><code>## Iteration 780 of 1127 completed</code></pre>
<pre><code>## Iteration 792 of 1127 completed</code></pre>
<pre><code>## Iteration 804 of 1127 completed</code></pre>
<pre><code>## Iteration 816 of 1127 completed</code></pre>
<pre><code>## Iteration 828 of 1127 completed</code></pre>
<pre><code>## Iteration 840 of 1127 completed</code></pre>
<pre><code>## Iteration 852 of 1127 completed</code></pre>
<pre><code>## Iteration 864 of 1127 completed</code></pre>
<pre><code>## Iteration 876 of 1127 completed</code></pre>
<pre><code>## Iteration 888 of 1127 completed</code></pre>
<pre><code>## Iteration 900 of 1127 completed</code></pre>
<pre><code>## Iteration 912 of 1127 completed</code></pre>
<pre><code>## Iteration 924 of 1127 completed</code></pre>
<pre><code>## Iteration 936 of 1127 completed</code></pre>
<pre><code>## Iteration 948 of 1127 completed</code></pre>
<pre><code>## Iteration 960 of 1127 completed</code></pre>
<pre><code>## Iteration 972 of 1127 completed</code></pre>
<pre><code>## Iteration 984 of 1127 completed</code></pre>
<pre><code>## Iteration 996 of 1127 completed</code></pre>
<pre><code>## Iteration 1008 of 1127 completed</code></pre>
<pre><code>## Iteration 1020 of 1127 completed</code></pre>
<pre><code>## Iteration 1032 of 1127 completed</code></pre>
<pre><code>## Iteration 1044 of 1127 completed</code></pre>
<pre><code>## Iteration 1056 of 1127 completed</code></pre>
<pre><code>## Iteration 1068 of 1127 completed</code></pre>
<pre><code>## Iteration 1080 of 1127 completed</code></pre>
<pre><code>## Iteration 1092 of 1127 completed</code></pre>
<pre><code>## Iteration 1104 of 1127 completed</code></pre>
<pre><code>## Iteration 1116 of 1127 completed</code></pre>
<div class="sourceCode" id="cb446"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb446-1"><a href="association-tests.html#cb446-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(assoc<span class="sc">$</span>results)</span></code></pre></div>
<pre><code>##   chr   start     end n.site n.alt n.sample.alt          Q        pval err
## 1   1  966001  971000      1     9            9   7.318095 0.643988693   0
## 2   1  982001  987000      1   111          107 154.178289 0.058811535   0
## 3   1 1022001 1027000      1     1            1  47.823918 0.005504472   0
## 4   1 1262001 1267000      1     2            2   7.319239 0.308472744   0
## 5   1 1468001 1473000      1    16           16  58.518665 0.293504065   0
## 6   1 1732001 1737000      1     5            5   9.499539 0.509868790   0
##   pval.method
## 1 integration
## 2 integration
## 3 integration
## 4 integration
## 5 integration
## 6 integration</code></pre>
<div class="sourceCode" id="cb448"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb448-1"><a href="association-tests.html#cb448-1" aria-hidden="true" tabindex="-1"></a><span class="fu">seqClose</span>(gds)</span></code></pre></div>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="phenotype-harmonization.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="computing-a-grm.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
