<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>2 GDS format | SISG 2021 Module 16: Computational Pipeline for WGS Data</title>
  <meta name="description" content="Course materials for SISG 2021 Module 16" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="2 GDS format | SISG 2021 Module 16: Computational Pipeline for WGS Data" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Course materials for SISG 2021 Module 16" />
  <meta name="github-repo" content="UW-GAC/SISG_2021" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="2 GDS format | SISG 2021 Module 16: Computational Pipeline for WGS Data" />
  
  <meta name="twitter:description" content="Course materials for SISG 2021 Module 16" />
  



<meta name="date" content="2021-07-20" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="phenotype-harmonization.html"/>
<script src="libs/header-attrs-2.9/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#schedule"><i class="fa fa-check"></i><b>1.1</b> Schedule</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#r-packages-used"><i class="fa fa-check"></i><b>1.2</b> R packages used</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#resources"><i class="fa fa-check"></i><b>1.3</b> Resources</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="gds-format.html"><a href="gds-format.html"><i class="fa fa-check"></i><b>2</b> GDS format</a>
<ul>
<li class="chapter" data-level="2.1" data-path="gds-format.html"><a href="gds-format.html#convert-a-vcf-to-gds"><i class="fa fa-check"></i><b>2.1</b> Convert a VCF to GDS</a></li>
<li class="chapter" data-level="2.2" data-path="gds-format.html"><a href="gds-format.html#exploring-a-gds-file"><i class="fa fa-check"></i><b>2.2</b> Exploring a GDS file</a></li>
<li class="chapter" data-level="2.3" data-path="gds-format.html"><a href="gds-format.html#exercises"><i class="fa fa-check"></i><b>2.3</b> Exercises</a></li>
<li class="chapter" data-level="2.4" data-path="gds-format.html"><a href="gds-format.html#solutions"><i class="fa fa-check"></i><b>2.4</b> Solutions</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="phenotype-harmonization.html"><a href="phenotype-harmonization.html"><i class="fa fa-check"></i><b>3</b> Phenotype Harmonization</a>
<ul>
<li class="chapter" data-level="3.1" data-path="phenotype-harmonization.html"><a href="phenotype-harmonization.html#inspect-individual-study-data-in-r"><i class="fa fa-check"></i><b>3.1</b> Inspect individual study data in R</a></li>
<li class="chapter" data-level="3.2" data-path="phenotype-harmonization.html"><a href="phenotype-harmonization.html#compare-study-values"><i class="fa fa-check"></i><b>3.2</b> Compare study values</a></li>
<li class="chapter" data-level="3.3" data-path="phenotype-harmonization.html"><a href="phenotype-harmonization.html#using-regression-models-to-compare-studies"><i class="fa fa-check"></i><b>3.3</b> Using regression models to compare studies</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="phenotype-harmonization.html"><a href="phenotype-harmonization.html#create-an-annotated-data-frame"><i class="fa fa-check"></i><b>3.3.1</b> Create an Annotated Data Frame</a></li>
<li class="chapter" data-level="3.3.2" data-path="phenotype-harmonization.html"><a href="phenotype-harmonization.html#fit-a-regression-model-without-study"><i class="fa fa-check"></i><b>3.3.2</b> Fit a regression model without study</a></li>
<li class="chapter" data-level="3.3.3" data-path="phenotype-harmonization.html"><a href="phenotype-harmonization.html#fit-a-model-with-study"><i class="fa fa-check"></i><b>3.3.3</b> Fit a model with study</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="phenotype-harmonization.html"><a href="phenotype-harmonization.html#final-considerations"><i class="fa fa-check"></i><b>3.4</b> Final considerations</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="association-tests-part-i.html"><a href="association-tests-part-i.html"><i class="fa fa-check"></i><b>4</b> Association tests - Part I</a>
<ul>
<li class="chapter" data-level="4.1" data-path="association-tests-part-i.html"><a href="association-tests-part-i.html#null-model"><i class="fa fa-check"></i><b>4.1</b> Null model</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="association-tests-part-i.html"><a href="association-tests-part-i.html#prepare-the-data"><i class="fa fa-check"></i><b>4.1.1</b> Prepare the data</a></li>
<li class="chapter" data-level="4.1.2" data-path="association-tests-part-i.html"><a href="association-tests-part-i.html#fit-the-null-model"><i class="fa fa-check"></i><b>4.1.2</b> Fit the null model</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="association-tests-part-i.html"><a href="association-tests-part-i.html#exercise"><i class="fa fa-check"></i><b>4.2</b> Exercise</a></li>
<li class="chapter" data-level="4.3" data-path="association-tests-part-i.html"><a href="association-tests-part-i.html#solution"><i class="fa fa-check"></i><b>4.3</b> Solution</a></li>
<li class="chapter" data-level="4.4" data-path="association-tests-part-i.html"><a href="association-tests-part-i.html#single-variant-association-tests"><i class="fa fa-check"></i><b>4.4</b> Single-variant association tests</a></li>
<li class="chapter" data-level="4.5" data-path="association-tests-part-i.html"><a href="association-tests-part-i.html#exercise-1"><i class="fa fa-check"></i><b>4.5</b> Exercise</a></li>
<li class="chapter" data-level="4.6" data-path="association-tests-part-i.html"><a href="association-tests-part-i.html#solution-1"><i class="fa fa-check"></i><b>4.6</b> Solution</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="association-tests-part-ii.html"><a href="association-tests-part-ii.html"><i class="fa fa-check"></i><b>5</b> Association tests - Part II</a>
<ul>
<li class="chapter" data-level="5.1" data-path="association-tests-part-ii.html"><a href="association-tests-part-ii.html#sliding-window-tests"><i class="fa fa-check"></i><b>5.1</b> Sliding window tests</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="association-tests-part-ii.html"><a href="association-tests-part-ii.html#burden-test"><i class="fa fa-check"></i><b>5.1.1</b> Burden test</a></li>
<li class="chapter" data-level="5.1.2" data-path="association-tests-part-ii.html"><a href="association-tests-part-ii.html#skat-test"><i class="fa fa-check"></i><b>5.1.2</b> SKAT test</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="association-tests-part-ii.html"><a href="association-tests-part-ii.html#exercise-2"><i class="fa fa-check"></i><b>5.2</b> Exercise</a></li>
<li class="chapter" data-level="5.3" data-path="association-tests-part-ii.html"><a href="association-tests-part-ii.html#solution-2"><i class="fa fa-check"></i><b>5.3</b> Solution</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="ancestry-and-relatedness-inference.html"><a href="ancestry-and-relatedness-inference.html"><i class="fa fa-check"></i><b>6</b> Ancestry and Relatedness Inference</a>
<ul>
<li class="chapter" data-level="6.1" data-path="ancestry-and-relatedness-inference.html"><a href="ancestry-and-relatedness-inference.html#ld-pruning"><i class="fa fa-check"></i><b>6.1</b> LD-pruning</a></li>
<li class="chapter" data-level="6.2" data-path="ancestry-and-relatedness-inference.html"><a href="ancestry-and-relatedness-inference.html#computing-a-grm"><i class="fa fa-check"></i><b>6.2</b> Computing a GRM</a></li>
<li class="chapter" data-level="6.3" data-path="ancestry-and-relatedness-inference.html"><a href="ancestry-and-relatedness-inference.html#de-convoluting-ancestry-and-relatedness"><i class="fa fa-check"></i><b>6.3</b> De-convoluting ancestry and relatedness</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="ancestry-and-relatedness-inference.html"><a href="ancestry-and-relatedness-inference.html#king"><i class="fa fa-check"></i><b>6.3.1</b> KING</a></li>
<li class="chapter" data-level="6.3.2" data-path="ancestry-and-relatedness-inference.html"><a href="ancestry-and-relatedness-inference.html#pc-air"><i class="fa fa-check"></i><b>6.3.2</b> PC-AiR</a></li>
<li class="chapter" data-level="6.3.3" data-path="ancestry-and-relatedness-inference.html"><a href="ancestry-and-relatedness-inference.html#pc-relate"><i class="fa fa-check"></i><b>6.3.3</b> PC-Relate</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="ancestry-and-relatedness-inference.html"><a href="ancestry-and-relatedness-inference.html#exercises-1"><i class="fa fa-check"></i><b>6.4</b> Exercises</a></li>
<li class="chapter" data-level="6.5" data-path="ancestry-and-relatedness-inference.html"><a href="ancestry-and-relatedness-inference.html#solutions-1"><i class="fa fa-check"></i><b>6.5</b> Solutions</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="mixed-models.html"><a href="mixed-models.html"><i class="fa fa-check"></i><b>7</b> Mixed models</a>
<ul>
<li class="chapter" data-level="7.1" data-path="mixed-models.html"><a href="mixed-models.html#null-model-1"><i class="fa fa-check"></i><b>7.1</b> Null model</a></li>
<li class="chapter" data-level="7.2" data-path="mixed-models.html"><a href="mixed-models.html#single-variant-tests"><i class="fa fa-check"></i><b>7.2</b> Single-variant tests</a></li>
<li class="chapter" data-level="7.3" data-path="mixed-models.html"><a href="mixed-models.html#exercise-3"><i class="fa fa-check"></i><b>7.3</b> Exercise</a></li>
<li class="chapter" data-level="7.4" data-path="mixed-models.html"><a href="mixed-models.html#solution-3"><i class="fa fa-check"></i><b>7.4</b> Solution</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="variant-annotation.html"><a href="variant-annotation.html"><i class="fa fa-check"></i><b>8</b> Variant annotation</a>
<ul>
<li class="chapter" data-level="8.1" data-path="variant-annotation.html"><a href="variant-annotation.html#use-cases"><i class="fa fa-check"></i><b>8.1</b> Use cases</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="variant-annotation.html"><a href="variant-annotation.html#use-case-1"><i class="fa fa-check"></i><b>8.1.1</b> Use case 1</a></li>
<li class="chapter" data-level="8.1.2" data-path="variant-annotation.html"><a href="variant-annotation.html#use-case-2"><i class="fa fa-check"></i><b>8.1.2</b> Use case 2</a></li>
<li class="chapter" data-level="8.1.3" data-path="variant-annotation.html"><a href="variant-annotation.html#use-case-3"><i class="fa fa-check"></i><b>8.1.3</b> Use case 3</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="variant-annotation.html"><a href="variant-annotation.html#exercise-4"><i class="fa fa-check"></i><b>8.2</b> Exercise</a></li>
<li class="chapter" data-level="8.3" data-path="variant-annotation.html"><a href="variant-annotation.html#solution-4"><i class="fa fa-check"></i><b>8.3</b> Solution</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="annotation-informed-aggregate-association-tests.html"><a href="annotation-informed-aggregate-association-tests.html"><i class="fa fa-check"></i><b>9</b> Annotation informed aggregate association tests</a>
<ul>
<li class="chapter" data-level="9.1" data-path="annotation-informed-aggregate-association-tests.html"><a href="annotation-informed-aggregate-association-tests.html#aggregate-unit-for-association-testing-exercise"><i class="fa fa-check"></i><b>9.1</b> Aggregate unit for association testing exercise</a></li>
<li class="chapter" data-level="9.2" data-path="annotation-informed-aggregate-association-tests.html"><a href="annotation-informed-aggregate-association-tests.html#association-testing-with-aggregate-units"><i class="fa fa-check"></i><b>9.2</b> Association testing with aggregate units</a></li>
<li class="chapter" data-level="9.3" data-path="annotation-informed-aggregate-association-tests.html"><a href="annotation-informed-aggregate-association-tests.html#exercise-5"><i class="fa fa-check"></i><b>9.3</b> Exercise</a></li>
<li class="chapter" data-level="9.4" data-path="annotation-informed-aggregate-association-tests.html"><a href="annotation-informed-aggregate-association-tests.html#solution-5"><i class="fa fa-check"></i><b>9.4</b> Solution</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="running-a-gwas-workflow-on-biodata-catalyst.html"><a href="running-a-gwas-workflow-on-biodata-catalyst.html"><i class="fa fa-check"></i><b>10</b> Running a GWAS workflow on BioData Catalyst</a>
<ul>
<li class="chapter" data-level="10.1" data-path="running-a-gwas-workflow-on-biodata-catalyst.html"><a href="running-a-gwas-workflow-on-biodata-catalyst.html#preparing-the-genotype-data"><i class="fa fa-check"></i><b>10.1</b> Preparing the genotype data</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="running-a-gwas-workflow-on-biodata-catalyst.html"><a href="running-a-gwas-workflow-on-biodata-catalyst.html#bcftools-merge-and-filter"><i class="fa fa-check"></i><b>10.1.1</b> Bcftools Merge and Filter</a></li>
<li class="chapter" data-level="10.1.2" data-path="running-a-gwas-workflow-on-biodata-catalyst.html"><a href="running-a-gwas-workflow-on-biodata-catalyst.html#vcf-to-gds-converter"><i class="fa fa-check"></i><b>10.1.2</b> VCF to GDS Converter</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="running-a-gwas-workflow-on-biodata-catalyst.html"><a href="running-a-gwas-workflow-on-biodata-catalyst.html#ancestry-and-relatedness"><i class="fa fa-check"></i><b>10.2</b> Ancestry and Relatedness</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="running-a-gwas-workflow-on-biodata-catalyst.html"><a href="running-a-gwas-workflow-on-biodata-catalyst.html#king-robust"><i class="fa fa-check"></i><b>10.2.1</b> KING robust</a></li>
<li class="chapter" data-level="10.2.2" data-path="running-a-gwas-workflow-on-biodata-catalyst.html"><a href="running-a-gwas-workflow-on-biodata-catalyst.html#pc-air-1"><i class="fa fa-check"></i><b>10.2.2</b> PC-AiR</a></li>
<li class="chapter" data-level="10.2.3" data-path="running-a-gwas-workflow-on-biodata-catalyst.html"><a href="running-a-gwas-workflow-on-biodata-catalyst.html#pc-relate-1"><i class="fa fa-check"></i><b>10.2.3</b> PC-Relate</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="running-a-gwas-workflow-on-biodata-catalyst.html"><a href="running-a-gwas-workflow-on-biodata-catalyst.html#association-testing"><i class="fa fa-check"></i><b>10.3</b> Association Testing</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="running-a-gwas-workflow-on-biodata-catalyst.html"><a href="running-a-gwas-workflow-on-biodata-catalyst.html#null-model-2"><i class="fa fa-check"></i><b>10.3.1</b> Null Model</a></li>
<li class="chapter" data-level="10.3.2" data-path="running-a-gwas-workflow-on-biodata-catalyst.html"><a href="running-a-gwas-workflow-on-biodata-catalyst.html#single-variant-association-test"><i class="fa fa-check"></i><b>10.3.2</b> Single variant association test</a></li>
<li class="chapter" data-level="10.3.3" data-path="running-a-gwas-workflow-on-biodata-catalyst.html"><a href="running-a-gwas-workflow-on-biodata-catalyst.html#aggregate-variant-test"><i class="fa fa-check"></i><b>10.3.3</b> Aggregate variant test</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="running-a-gwas-workflow-on-biodata-catalyst.html"><a href="running-a-gwas-workflow-on-biodata-catalyst.html#analysis-follow-up"><i class="fa fa-check"></i><b>10.4</b> Analysis follow-up</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">SISG 2021 Module 16: Computational Pipeline for WGS Data</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="gds-format" class="section level1" number="2">
<h1><span class="header-section-number">2</span> GDS format</h1>
<p>GDS is Genomic Data Structure, a storage format that can efficiently store genomic data and provide fast random access to subsets of the data. For more information on GDS for sequence data, read the <a href="https://github.com/zhengxwen/SeqArray/blob/master/vignettes/SeqArrayTutorial.Rmd">SeqArray package vignette</a>.</p>
<div id="convert-a-vcf-to-gds" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> Convert a VCF to GDS</h2>
<p>To use the R packages developed at the University of Washington Genetic Analysis Center for sequence data, we first need to convert a VCF file to GDS. (If the file is BCF, use <a href="bcftools">https://samtools.github.io/bcftools/bcftools.html</a> to convert to VCF.)</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="gds-format.html#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SeqArray)</span>
<span id="cb1-2"><a href="gds-format.html#cb1-2" aria-hidden="true" tabindex="-1"></a>repo_path <span class="ot">&lt;-</span> <span class="st">&quot;https://github.com/UW-GAC/SISG_2021/raw/master&quot;</span></span>
<span id="cb1-3"><a href="gds-format.html#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">dir.exists</span>(<span class="st">&quot;data&quot;</span>)) <span class="fu">dir.create</span>(<span class="st">&quot;data&quot;</span>)</span>
<span id="cb1-4"><a href="gds-format.html#cb1-4" aria-hidden="true" tabindex="-1"></a>vcffile <span class="ot">&lt;-</span> <span class="st">&quot;data/1KG_phase3_subset_chr1.vcf.gz&quot;</span></span>
<span id="cb1-5"><a href="gds-format.html#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(vcffile)) <span class="fu">download.file</span>(<span class="fu">file.path</span>(repo_path, vcffile), vcffile)</span>
<span id="cb1-6"><a href="gds-format.html#cb1-6" aria-hidden="true" tabindex="-1"></a>gdsfile <span class="ot">&lt;-</span> <span class="st">&quot;data/1KG_phase3_subset_chr1.gds&quot;</span></span>
<span id="cb1-7"><a href="gds-format.html#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="co"># convert the VCF to GDS</span></span>
<span id="cb1-8"><a href="gds-format.html#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">seqVCF2GDS</span>(vcffile, gdsfile, <span class="at">fmt.import=</span><span class="st">&quot;GT&quot;</span>, <span class="at">storage.option=</span><span class="st">&quot;LZMA_RA&quot;</span>)</span></code></pre></div>
<pre><code>## Tue Jul 20 17:39:19 2021
## Variant Call Format (VCF) Import:
##     file(s):
##         1KG_phase3_subset_chr1.vcf.gz (120.7K)
##     file format: VCFv4.1
##     genome reference: ftp://ftp.1000genomes.ebi.ac.uk//vol1/ftp/technical/reference/phase2_reference_assembly_sequence/hs37d5.fa.gz
##     the number of sets of chromosomes (ploidy): 2
##     the number of samples: 1,126
##     genotype storage: bit2
##     compression method: LZMA_RA
##     # of samples: 1126
## Output:
##     data/1KG_phase3_subset_chr1.gds
## Parsing &#39;1KG_phase3_subset_chr1.vcf.gz&#39;:
## + genotype/data   { Bit2 2x1126x1121 LZMA_ra(0.01%), 42B }
## Digests:
##     sample.id  [md5: 6b04799356320063276ad84f70713429]
##     variant.id  [md5: 35a6feaa59ff5659e2d16d2afb6923d4]
##     position  [md5: 868601c0a2f333d113be2040295602b4]
##     chromosome  [md5: 774667f087db893e987ce02dc9188ee2]
##     allele  [md5: bb36e628c2faf01dfc2792615b6c2350]
##     genotype  [md5: 59be1f80c30eee4a93b82077530eba87]
##     phase  [md5: ae6a90ef5cceae947052b0c8999a4a94]
##     annotation/id  [md5: 045802d7856c955f4710288e4764fcbe]
##     annotation/qual  [md5: d6b8ed9bd5cac7dd7898bb679675fc7c]
##     annotation/filter  [md5: 8294edaf60fffc1e9aa693b4bc423dd1]
## Done.
## Tue Jul 20 17:39:20 2021
## Optimize the access efficiency ...
## Clean up the fragments of GDS file:
##     open the file &#39;data/1KG_phase3_subset_chr1.gds&#39; (71.3K)
##     # of fragments: 123
##     save to &#39;data/1KG_phase3_subset_chr1.gds.tmp&#39;
##     rename &#39;data/1KG_phase3_subset_chr1.gds.tmp&#39; (70.5K, reduced: 744B)
##     # of fragments: 61
## Tue Jul 20 17:39:20 2021</code></pre>
</div>
<div id="exploring-a-gds-file" class="section level2" number="2.2">
<h2><span class="header-section-number">2.2</span> Exploring a GDS file</h2>
<p>We can interact with the GDS file using the SeqArray package.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="gds-format.html#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># open a connection to the GDS file</span></span>
<span id="cb3-2"><a href="gds-format.html#cb3-2" aria-hidden="true" tabindex="-1"></a>gds <span class="ot">&lt;-</span> <span class="fu">seqOpen</span>(gdsfile)</span>
<span id="cb3-3"><a href="gds-format.html#cb3-3" aria-hidden="true" tabindex="-1"></a>gds</span></code></pre></div>
<pre><code>## Object of class &quot;SeqVarGDSClass&quot;
## File: /Users/mconomos/Documents/SISG_2021/data/1KG_phase3_subset_chr1.gds (70.5K)
## +    [  ] *
## |--+ description   [  ] *
## |--+ sample.id   { Str8 1126 LZMA_ra(9.66%), 877B } *
## |--+ variant.id   { Int32 1120 LZMA_ra(17.5%), 793B } *
## |--+ position   { Int32 1120 LZMA_ra(78.5%), 3.4K } *
## |--+ chromosome   { Str8 1120 LZMA_ra(4.55%), 109B } *
## |--+ allele   { Str8 1120 LZMA_ra(26.0%), 1.2K } *
## |--+ genotype   [  ] *
## |  |--+ data   { Bit2 2x1126x1121 LZMA_ra(8.34%), 51.4K } *
## |  |--+ extra.index   { Int32 3x0 LZMA_ra, 18B } *
## |  \--+ extra   { Int16 0 LZMA_ra, 18B }
## |--+ phase   [  ]
## |  |--+ data   { Bit1 1126x1120 LZMA_ra(0.11%), 177B } *
## |  |--+ extra.index   { Int32 3x0 LZMA_ra, 18B } *
## |  \--+ extra   { Bit1 0 LZMA_ra, 18B }
## |--+ annotation   [  ]
## |  |--+ id   { Str8 1120 LZMA_ra(40.4%), 3.6K } *
## |  |--+ qual   { Float32 1120 LZMA_ra(2.46%), 117B } *
## |  |--+ filter   { Int32,factor 1120 LZMA_ra(2.46%), 117B } *
## |  |--+ info   [  ]
## |  \--+ format   [  ]
## \--+ sample.annotation   [  ]</code></pre>
<p>The <code>seqGetData</code> function is the basic function for reading in data from a GDS file</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="gds-format.html#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># the unique sample identifier comes from the VCF header</span></span>
<span id="cb5-2"><a href="gds-format.html#cb5-2" aria-hidden="true" tabindex="-1"></a>sample.id <span class="ot">&lt;-</span> <span class="fu">seqGetData</span>(gds, <span class="st">&quot;sample.id&quot;</span>)</span>
<span id="cb5-3"><a href="gds-format.html#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(sample.id)</span></code></pre></div>
<pre><code>## [1] 1126</code></pre>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="gds-format.html#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(sample.id)</span></code></pre></div>
<pre><code>## [1] &quot;HG00096&quot; &quot;HG00097&quot; &quot;HG00099&quot; &quot;HG00100&quot; &quot;HG00101&quot; &quot;HG00102&quot;</code></pre>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="gds-format.html#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># a unique integer ID is assigned to each variant</span></span>
<span id="cb9-2"><a href="gds-format.html#cb9-2" aria-hidden="true" tabindex="-1"></a>variant.id <span class="ot">&lt;-</span> <span class="fu">seqGetData</span>(gds, <span class="st">&quot;variant.id&quot;</span>)</span>
<span id="cb9-3"><a href="gds-format.html#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(variant.id)</span></code></pre></div>
<pre><code>## [1] 1120</code></pre>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="gds-format.html#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(variant.id)</span></code></pre></div>
<pre><code>## [1] 1 2 3 4 5 6</code></pre>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="gds-format.html#cb13-1" aria-hidden="true" tabindex="-1"></a>chr <span class="ot">&lt;-</span> <span class="fu">seqGetData</span>(gds, <span class="st">&quot;chromosome&quot;</span>)</span>
<span id="cb13-2"><a href="gds-format.html#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(chr)</span></code></pre></div>
<pre><code>## [1] &quot;1&quot; &quot;1&quot; &quot;1&quot; &quot;1&quot; &quot;1&quot; &quot;1&quot;</code></pre>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="gds-format.html#cb15-1" aria-hidden="true" tabindex="-1"></a>pos <span class="ot">&lt;-</span> <span class="fu">seqGetData</span>(gds, <span class="st">&quot;position&quot;</span>)</span>
<span id="cb15-2"><a href="gds-format.html#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(pos)</span></code></pre></div>
<pre><code>## [1]  970546  985900 1025045 1265550 1472676 1735725</code></pre>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="gds-format.html#cb17-1" aria-hidden="true" tabindex="-1"></a>id <span class="ot">&lt;-</span> <span class="fu">seqGetData</span>(gds, <span class="st">&quot;annotation/id&quot;</span>)</span>
<span id="cb17-2"><a href="gds-format.html#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(id)</span></code></pre></div>
<pre><code>## [1] &quot;&quot;           &quot;rs17160776&quot; &quot;&quot;           &quot;&quot;           &quot;rs78293298&quot; &quot;&quot;</code></pre>
<p>There are additional useful functions for summary level data.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="gds-format.html#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co"># reference allele frequency of each variant</span></span>
<span id="cb19-2"><a href="gds-format.html#cb19-2" aria-hidden="true" tabindex="-1"></a>afreq <span class="ot">&lt;-</span> <span class="fu">seqAlleleFreq</span>(gds)</span>
<span id="cb19-3"><a href="gds-format.html#cb19-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(afreq)</span></code></pre></div>
<pre><code>## [1] 0.9960036 0.9507105 0.9995560 0.9991119 0.9928952 0.9977798</code></pre>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="gds-format.html#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(afreq)</span></code></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##  0.0000  0.9866  0.9987  0.9436  0.9996  1.0000</code></pre>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="gds-format.html#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(afreq, <span class="at">breaks=</span><span class="dv">50</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/ref_freq-1.png" width="672" /></p>
<p>We can define a filter on the <code>gds</code> object. After using the <code>seqSetFilter</code> command, all subsequent reads from the <code>gds</code> object are restricted to the selected subset of data, until a new filter is defined or <code>seqResetFilter</code> is called.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="gds-format.html#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">seqSetFilter</span>(gds, <span class="at">variant.id=</span><span class="dv">91</span><span class="sc">:</span><span class="dv">100</span>, <span class="at">sample.id=</span>sample.id[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>])</span></code></pre></div>
<pre><code>## # of selected samples: 5
## # of selected variants: 10</code></pre>
<p>Genotype data is stored in a 3-dimensional array, where the first dimension is always length 2 for diploid genotypes. The second and third dimensions are samples and variants, respectively. The values of the array denote alleles: <code>0</code> is the reference allele and <code>1</code> is the alternate allele. For multiallelic variants, other alternate alleles are represented as integers <code>&gt; 1</code>.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="gds-format.html#cb26-1" aria-hidden="true" tabindex="-1"></a>geno <span class="ot">&lt;-</span> <span class="fu">seqGetData</span>(gds, <span class="st">&quot;genotype&quot;</span>)</span>
<span id="cb26-2"><a href="gds-format.html#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(geno)</span></code></pre></div>
<pre><code>## [1]  2  5 10</code></pre>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="gds-format.html#cb28-1" aria-hidden="true" tabindex="-1"></a>geno[,,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]</span></code></pre></div>
<pre><code>## , , 1
## 
##       sample
## allele [,1] [,2] [,3] [,4] [,5]
##   [1,]    1    1    1    1    1
##   [2,]    1    1    1    1    1
## 
## , , 2
## 
##       sample
## allele [,1] [,2] [,3] [,4] [,5]
##   [1,]    0    0    0    0    0
##   [2,]    0    1    0    0    1</code></pre>
<p>The <a href="http://bioconductor.org/packages/SeqVarTools">SeqVarTools package</a> has some additional functions for interacting with SeqArray-format GDS files. There are functions providing more intuitive ways to read in genotypes.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="gds-format.html#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SeqVarTools)</span>
<span id="cb30-2"><a href="gds-format.html#cb30-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-3"><a href="gds-format.html#cb30-3" aria-hidden="true" tabindex="-1"></a><span class="co"># return genotypes in matrix format</span></span>
<span id="cb30-4"><a href="gds-format.html#cb30-4" aria-hidden="true" tabindex="-1"></a><span class="fu">getGenotype</span>(gds)</span></code></pre></div>
<pre><code>##          variant
## sample    91    92    93    94    95    96    97    98    99    100  
##   HG00096 &quot;1|1&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;1|1&quot; &quot;0|0&quot; &quot;0|0&quot;
##   HG00097 &quot;1|1&quot; &quot;0|1&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;1|0&quot; &quot;0|0&quot; &quot;0|0&quot;
##   HG00099 &quot;1|1&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;1|0&quot; &quot;0|0&quot; &quot;0|0&quot;
##   HG00100 &quot;1|1&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot;
##   HG00101 &quot;1|1&quot; &quot;0|1&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot;</code></pre>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="gds-format.html#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fu">getGenotypeAlleles</span>(gds)</span></code></pre></div>
<pre><code>##          variant
## sample    91    92    93    94    95    96    97    98    99    100  
##   HG00096 &quot;G|G&quot; &quot;T|T&quot; &quot;G|G&quot; &quot;G|G&quot; &quot;G|G&quot; &quot;A|A&quot; &quot;C|C&quot; &quot;G|G&quot; &quot;C|C&quot; &quot;T|T&quot;
##   HG00097 &quot;G|G&quot; &quot;T|C&quot; &quot;G|G&quot; &quot;G|G&quot; &quot;G|G&quot; &quot;A|A&quot; &quot;C|C&quot; &quot;G|C&quot; &quot;C|C&quot; &quot;T|T&quot;
##   HG00099 &quot;G|G&quot; &quot;T|T&quot; &quot;G|G&quot; &quot;G|G&quot; &quot;G|G&quot; &quot;A|A&quot; &quot;C|C&quot; &quot;G|C&quot; &quot;C|C&quot; &quot;T|T&quot;
##   HG00100 &quot;G|G&quot; &quot;T|T&quot; &quot;G|G&quot; &quot;G|G&quot; &quot;G|G&quot; &quot;A|A&quot; &quot;C|C&quot; &quot;C|C&quot; &quot;C|C&quot; &quot;T|T&quot;
##   HG00101 &quot;G|G&quot; &quot;T|C&quot; &quot;G|G&quot; &quot;G|G&quot; &quot;G|G&quot; &quot;A|A&quot; &quot;C|C&quot; &quot;C|C&quot; &quot;C|C&quot; &quot;T|T&quot;</code></pre>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="gds-format.html#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="fu">refDosage</span>(gds)</span></code></pre></div>
<pre><code>##          variant
## sample    91 92 93 94 95 96 97 98 99 100
##   HG00096  0  2  2  2  2  2  2  0  2   2
##   HG00097  0  1  2  2  2  2  2  1  2   2
##   HG00099  0  2  2  2  2  2  2  1  2   2
##   HG00100  0  2  2  2  2  2  2  2  2   2
##   HG00101  0  1  2  2  2  2  2  2  2   2</code></pre>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="gds-format.html#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="fu">altDosage</span>(gds)</span></code></pre></div>
<pre><code>##          variant
## sample    91 92 93 94 95 96 97 98 99 100
##   HG00096  2  0  0  0  0  0  0  2  0   0
##   HG00097  2  1  0  0  0  0  0  1  0   0
##   HG00099  2  0  0  0  0  0  0  1  0   0
##   HG00100  2  0  0  0  0  0  0  0  0   0
##   HG00101  2  1  0  0  0  0  0  0  0   0</code></pre>
<p>There are functions to extract variant information.</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="gds-format.html#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="co"># look at reference and alternate alleles</span></span>
<span id="cb38-2"><a href="gds-format.html#cb38-2" aria-hidden="true" tabindex="-1"></a><span class="fu">refChar</span>(gds)</span></code></pre></div>
<pre><code>##  [1] &quot;A&quot; &quot;T&quot; &quot;G&quot; &quot;G&quot; &quot;G&quot; &quot;A&quot; &quot;C&quot; &quot;C&quot; &quot;C&quot; &quot;T&quot;</code></pre>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="gds-format.html#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="fu">altChar</span>(gds)</span></code></pre></div>
<pre><code>##  [1] &quot;G&quot; &quot;C&quot; &quot;A&quot; &quot;A&quot; &quot;A&quot; &quot;G&quot; &quot;T&quot; &quot;G&quot; &quot;A&quot; &quot;C&quot;</code></pre>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="gds-format.html#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="co"># data.frame of variant information</span></span>
<span id="cb42-2"><a href="gds-format.html#cb42-2" aria-hidden="true" tabindex="-1"></a><span class="fu">variantInfo</span>(gds)</span></code></pre></div>
<pre><code>##    variant.id chr      pos ref alt
## 1          91   1 15461365   A   G
## 2          92   1 15817319   T   C
## 3          93   1 16150840   G   A
## 4          94   1 16272296   G   A
## 5          95   1 16300899   G   A
## 6          96   1 16353896   A   G
## 7          97   1 16414772   C   T
## 8          98   1 16446764   C   G
## 9          99   1 16448267   C   A
## 10        100   1 16699413   T   C</code></pre>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="gds-format.html#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="co"># reset the filter to all variants and samples</span></span>
<span id="cb44-2"><a href="gds-format.html#cb44-2" aria-hidden="true" tabindex="-1"></a><span class="fu">seqResetFilter</span>(gds)</span></code></pre></div>
<pre><code>## # of selected samples: 1,126
## # of selected variants: 1,120</code></pre>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="gds-format.html#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="co"># how many alleles for each variant?</span></span>
<span id="cb46-2"><a href="gds-format.html#cb46-2" aria-hidden="true" tabindex="-1"></a>n <span class="ot">&lt;-</span> <span class="fu">seqNumAllele</span>(gds)</span>
<span id="cb46-3"><a href="gds-format.html#cb46-3" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(n)</span></code></pre></div>
<pre><code>## n
##    2    3    4 
## 1099   20    1</code></pre>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="gds-format.html#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="co"># some variants have more than one alternate allele</span></span>
<span id="cb48-2"><a href="gds-format.html#cb48-2" aria-hidden="true" tabindex="-1"></a>multi.allelic <span class="ot">&lt;-</span> <span class="fu">which</span>(n <span class="sc">&gt;</span> <span class="dv">2</span>)</span>
<span id="cb48-3"><a href="gds-format.html#cb48-3" aria-hidden="true" tabindex="-1"></a><span class="fu">altChar</span>(gds)[multi.allelic]</span></code></pre></div>
<pre><code>##  [1] &quot;GT,G&quot;            &quot;G,T&quot;             &quot;A,T&quot;             &quot;A,T&quot;             &quot;ATG,ATGTG&quot;       &quot;C,G&quot;             &quot;A,T&quot;             &quot;C,T&quot;             &quot;A,C&quot;            
## [10] &quot;TAA,T&quot;           &quot;GTTA,GTTT&quot;       &quot;GCC,GCCC,G&quot;      &quot;A,C&quot;             &quot;A,C&quot;             &quot;A,C&quot;             &quot;CAAGCAT,CGAGCAT&quot; &quot;CATTATT,C&quot;       &quot;AT,C&quot;           
## [19] &quot;TGTGA,C&quot;         &quot;CCATT,CCATTCATT&quot; &quot;C,G&quot;</code></pre>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="gds-format.html#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="co"># extract a particular alternate allele</span></span>
<span id="cb50-2"><a href="gds-format.html#cb50-2" aria-hidden="true" tabindex="-1"></a><span class="co"># first alternate</span></span>
<span id="cb50-3"><a href="gds-format.html#cb50-3" aria-hidden="true" tabindex="-1"></a><span class="fu">altChar</span>(gds, <span class="at">n=</span><span class="dv">1</span>)[multi.allelic]</span></code></pre></div>
<pre><code>##  [1] &quot;GT&quot;      &quot;G&quot;       &quot;A&quot;       &quot;A&quot;       &quot;ATG&quot;     &quot;C&quot;       &quot;A&quot;       &quot;C&quot;       &quot;A&quot;       &quot;TAA&quot;     &quot;GTTA&quot;    &quot;GCC&quot;     &quot;A&quot;       &quot;A&quot;       &quot;A&quot;       &quot;CAAGCAT&quot;
## [17] &quot;CATTATT&quot; &quot;AT&quot;      &quot;TGTGA&quot;   &quot;CCATT&quot;   &quot;C&quot;</code></pre>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="gds-format.html#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="co"># second alternate</span></span>
<span id="cb52-2"><a href="gds-format.html#cb52-2" aria-hidden="true" tabindex="-1"></a><span class="fu">altChar</span>(gds, <span class="at">n=</span><span class="dv">2</span>)[multi.allelic]</span></code></pre></div>
<pre><code>##  [1] &quot;G&quot;         &quot;T&quot;         &quot;T&quot;         &quot;T&quot;         &quot;ATGTG&quot;     &quot;G&quot;         &quot;T&quot;         &quot;T&quot;         &quot;C&quot;         &quot;T&quot;         &quot;GTTT&quot;      &quot;GCCC&quot;      &quot;C&quot;        
## [14] &quot;C&quot;         &quot;C&quot;         &quot;CGAGCAT&quot;   &quot;C&quot;         &quot;C&quot;         &quot;C&quot;         &quot;CCATTCATT&quot; &quot;G&quot;</code></pre>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="gds-format.html#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="co"># how many variants are biallelic SNVs?</span></span>
<span id="cb54-2"><a href="gds-format.html#cb54-2" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">isSNV</span>(gds, <span class="at">biallelic=</span><span class="cn">TRUE</span>))</span></code></pre></div>
<pre><code>## 
## FALSE  TRUE 
##   110  1010</code></pre>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="gds-format.html#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="co"># how many variants are SNVs vs INDELs?</span></span>
<span id="cb56-2"><a href="gds-format.html#cb56-2" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">isSNV</span>(gds, <span class="at">biallelic=</span><span class="cn">FALSE</span>))</span></code></pre></div>
<pre><code>## 
## FALSE  TRUE 
##    99  1021</code></pre>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="gds-format.html#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 11 SNVs are multi-allelic</span></span></code></pre></div>
<p>We can also return variant information as a <code>GRanges</code> object from the <a href="https://bioconductor.org/packages/release/bioc/manuals/GenomicRanges/man/GenomicRanges.pdf">GenomicRanges package</a>. This format for representing sequence data is common across many Bioconductor packages. Chromosome is stored in the <code>seqnames</code> column. The <code>ranges</code> column has variant position, which can be a single base pair or a range.</p>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="gds-format.html#cb59-1" aria-hidden="true" tabindex="-1"></a>gr <span class="ot">&lt;-</span> <span class="fu">granges</span>(gds)</span>
<span id="cb59-2"><a href="gds-format.html#cb59-2" aria-hidden="true" tabindex="-1"></a>gr</span></code></pre></div>
<pre><code>## GRanges object with 1120 ranges and 0 metadata columns:
##        seqnames              ranges strand
##           &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt;
##      1        1              970546      *
##      2        1              985900      *
##      3        1             1025045      *
##      4        1             1265550      *
##      5        1             1472676      *
##    ...      ...                 ...    ...
##   1116        1           248715186      *
##   1117        1 248715606-248715610      *
##   1118        1           248761613      *
##   1119        1           248894546      *
##   1120        1           249149558      *
##   -------
##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
<p>Always use the <code>seqClose</code> command to close your connection to a GDS file when you are done working with it. Trying to open an already opened GDS will result in an error.</p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="gds-format.html#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="fu">seqClose</span>(gds)</span></code></pre></div>
</div>
<div id="exercises" class="section level2" number="2.3">
<h2><span class="header-section-number">2.3</span> Exercises</h2>
<ol style="list-style-type: decimal">
<li><p>Set a filter selecting only multi-allelic variants. Inspect their genotypes using the different methods you learned above. Use the <code>alleleDosage</code> method to find dosage for the second (and third, etc.) alternate allele.</p></li>
<li><p>Use the <code>hwe</code> function in SeqVarTools to run a Hardy-Weinberg Equilibrium test on each variant. Identify a variant with low p-value and inspect its genotypes. (Note that the HWE test is only valid for biallelic variants, and will return <code>NA</code> for multiallelic variants.)</p></li>
</ol>
</div>
<div id="solutions" class="section level2" number="2.4">
<h2><span class="header-section-number">2.4</span> Solutions</h2>
<ol style="list-style-type: decimal">
<li>Set a filter selecting only multi-allelic variants. Inspect their genotypes using the different methods you learned above. Use the <code>alleleDosage</code> method to find dosage for the second (and third, etc.) alternate allele.</li>
</ol>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="gds-format.html#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="co"># open a connection to the GDS file again</span></span>
<span id="cb62-2"><a href="gds-format.html#cb62-2" aria-hidden="true" tabindex="-1"></a>gds <span class="ot">&lt;-</span> <span class="fu">seqOpen</span>(gdsfile)</span>
<span id="cb62-3"><a href="gds-format.html#cb62-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-4"><a href="gds-format.html#cb62-4" aria-hidden="true" tabindex="-1"></a><span class="co"># set your filter</span></span>
<span id="cb62-5"><a href="gds-format.html#cb62-5" aria-hidden="true" tabindex="-1"></a><span class="fu">seqSetFilter</span>(gds, <span class="at">variant.sel=</span>multi.allelic)</span></code></pre></div>
<pre><code>## # of selected variants: 21</code></pre>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="gds-format.html#cb64-1" aria-hidden="true" tabindex="-1"></a>geno <span class="ot">&lt;-</span> <span class="fu">seqGetData</span>(gds, <span class="st">&quot;genotype&quot;</span>)</span>
<span id="cb64-2"><a href="gds-format.html#cb64-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(geno)</span></code></pre></div>
<pre><code>## [1]    2 1126   21</code></pre>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="gds-format.html#cb66-1" aria-hidden="true" tabindex="-1"></a>geno[,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>## , , 1
## 
##       sample
## allele [,1] [,2] [,3] [,4] [,5]
##   [1,]    1    1    0    1    1
##   [2,]    0    1    1    1    1
## 
## , , 2
## 
##       sample
## allele [,1] [,2] [,3] [,4] [,5]
##   [1,]    0    0    0    0    0
##   [2,]    0    0    0    0    0
## 
## , , 3
## 
##       sample
## allele [,1] [,2] [,3] [,4] [,5]
##   [1,]    0    0    0    0    0
##   [2,]    0    0    0    0    0
## 
## , , 4
## 
##       sample
## allele [,1] [,2] [,3] [,4] [,5]
##   [1,]    1    0    0    1    0
##   [2,]    0    0    0    0    1
## 
## , , 5
## 
##       sample
## allele [,1] [,2] [,3] [,4] [,5]
##   [1,]    0    0    0    0    0
##   [2,]    0    0    0    0    0</code></pre>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="gds-format.html#cb68-1" aria-hidden="true" tabindex="-1"></a>geno <span class="ot">&lt;-</span> <span class="fu">getGenotype</span>(gds)</span>
<span id="cb68-2"><a href="gds-format.html#cb68-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(geno)</span></code></pre></div>
<pre><code>## [1] 1126   21</code></pre>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="gds-format.html#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(geno)</span></code></pre></div>
<pre><code>##          variant
## sample    30    69    73    161   162   195   243   253   407   431   434   610   627   645   689   756   765   814   988   1014  1056 
##   HG00096 &quot;1|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;1|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;1|0&quot; &quot;0|1&quot; &quot;3|3&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;2|0&quot; &quot;2|2&quot; &quot;2|2&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot;
##   HG00097 &quot;1|1&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;2|2&quot; &quot;0|1&quot; &quot;1|3&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;2|2&quot; &quot;2|2&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot;
##   HG00099 &quot;0|1&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;1|0&quot; &quot;1|3&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;2|2&quot; &quot;2|2&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot;
##   HG00100 &quot;1|1&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;1|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|2&quot; &quot;0|1&quot; &quot;1|1&quot; &quot;2|0&quot; &quot;0|0&quot; &quot;0|2&quot; &quot;2|2&quot; &quot;2|2&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|2&quot;
##   HG00101 &quot;1|1&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|1&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;2|0&quot; &quot;1|1&quot; &quot;0|0&quot; &quot;3|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;2|0&quot; &quot;2|2&quot; &quot;2|2&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;1|0&quot; &quot;0|0&quot;
##   HG00102 &quot;0|1&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;1|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;2|2&quot; &quot;0|1&quot; &quot;3|3&quot; &quot;0|2&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;2|2&quot; &quot;1|2&quot; &quot;0|0&quot; &quot;0|0&quot; &quot;0|1&quot; &quot;0|2&quot;</code></pre>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="gds-format.html#cb72-1" aria-hidden="true" tabindex="-1"></a>geno <span class="ot">&lt;-</span> <span class="fu">getGenotypeAlleles</span>(gds)</span>
<span id="cb72-2"><a href="gds-format.html#cb72-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(geno)</span></code></pre></div>
<pre><code>##          variant
## sample    30       69    73    161   162   195   243   253   407   431       434      610       627   645   689   756               765         814     988          
##   HG00096 &quot;GT|GTT&quot; &quot;A|A&quot; &quot;G|G&quot; &quot;A|G&quot; &quot;A|A&quot; &quot;A|A&quot; &quot;C|C&quot; &quot;G|G&quot; &quot;T|T&quot; &quot;TAA|TA&quot;  &quot;G|GTTA&quot; &quot;G|G&quot;     &quot;G|G&quot; &quot;G|G&quot; &quot;C|G&quot; &quot;CGAGCAT|CGAGCAT&quot; &quot;C|C&quot;       &quot;CT|CT&quot; &quot;CGTGA|CGTGA&quot;
##   HG00097 &quot;GT|GT&quot;  &quot;A|A&quot; &quot;G|G&quot; &quot;G|G&quot; &quot;A|A&quot; &quot;A|A&quot; &quot;C|C&quot; &quot;G|G&quot; &quot;T|T&quot; &quot;T|T&quot;     &quot;G|GTTA&quot; &quot;GCC|G&quot;   &quot;G|G&quot; &quot;G|G&quot; &quot;G|G&quot; &quot;CGAGCAT|CGAGCAT&quot; &quot;C|C&quot;       &quot;CT|CT&quot; &quot;CGTGA|CGTGA&quot;
##   HG00099 &quot;GTT|GT&quot; &quot;A|A&quot; &quot;G|G&quot; &quot;G|G&quot; &quot;A|A&quot; &quot;A|A&quot; &quot;C|C&quot; &quot;G|G&quot; &quot;T|T&quot; &quot;TA|TA&quot;   &quot;GTTA|G&quot; &quot;GCC|G&quot;   &quot;G|G&quot; &quot;G|G&quot; &quot;G|G&quot; &quot;CGAGCAT|CGAGCAT&quot; &quot;C|C&quot;       &quot;CT|CT&quot; &quot;CGTGA|CGTGA&quot;
##   HG00100 &quot;GT|GT&quot;  &quot;A|A&quot; &quot;G|G&quot; &quot;A|G&quot; &quot;A|A&quot; &quot;A|A&quot; &quot;C|C&quot; &quot;G|G&quot; &quot;T|T&quot; &quot;TA|T&quot;    &quot;G|GTTA&quot; &quot;GCC|GCC&quot; &quot;C|G&quot; &quot;G|G&quot; &quot;G|C&quot; &quot;CGAGCAT|CGAGCAT&quot; &quot;C|C&quot;       &quot;CT|CT&quot; &quot;CGTGA|CGTGA&quot;
##   HG00101 &quot;GT|GT&quot;  &quot;A|A&quot; &quot;G|G&quot; &quot;G|A&quot; &quot;A|A&quot; &quot;A|A&quot; &quot;C|C&quot; &quot;G|G&quot; &quot;C|T&quot; &quot;TAA|TAA&quot; &quot;G|G&quot;    &quot;G|GC&quot;    &quot;G|G&quot; &quot;G|G&quot; &quot;C|G&quot; &quot;CGAGCAT|CGAGCAT&quot; &quot;C|C&quot;       &quot;CT|CT&quot; &quot;CGTGA|CGTGA&quot;
##   HG00102 &quot;GTT|GT&quot; &quot;A|A&quot; &quot;G|G&quot; &quot;A|G&quot; &quot;A|A&quot; &quot;A|A&quot; &quot;C|C&quot; &quot;G|G&quot; &quot;T|T&quot; &quot;T|T&quot;     &quot;G|GTTA&quot; &quot;G|G&quot;     &quot;G|C&quot; &quot;G|G&quot; &quot;G|G&quot; &quot;CGAGCAT|CGAGCAT&quot; &quot;CATTATT|C&quot; &quot;CT|CT&quot; &quot;CGTGA|CGTGA&quot;
##          variant
## sample    1014      1056 
##   HG00096 &quot;C|C&quot;     &quot;A|A&quot;
##   HG00097 &quot;C|C&quot;     &quot;A|A&quot;
##   HG00099 &quot;C|C&quot;     &quot;A|A&quot;
##   HG00100 &quot;C|C&quot;     &quot;A|G&quot;
##   HG00101 &quot;CCATT|C&quot; &quot;A|A&quot;
##   HG00102 &quot;C|CCATT&quot; &quot;A|G&quot;</code></pre>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="gds-format.html#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="co"># count of reference alleles</span></span>
<span id="cb74-2"><a href="gds-format.html#cb74-2" aria-hidden="true" tabindex="-1"></a>dos <span class="ot">&lt;-</span> <span class="fu">refDosage</span>(gds)</span>
<span id="cb74-3"><a href="gds-format.html#cb74-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(dos)</span></code></pre></div>
<pre><code>##          variant
## sample    30 69 73 161 162 195 243 253 407 431 434 610 627 645 689 756 765 814 988 1014 1056
##   HG00096  1  2  2   1   2   2   2   2   2   1   1   0   2   2   1   0   0   2   2    2    2
##   HG00097  0  2  2   2   2   2   2   2   2   0   1   0   2   2   2   0   0   2   2    2    2
##   HG00099  1  2  2   2   2   2   2   2   2   2   1   0   2   2   2   0   0   2   2    2    2
##   HG00100  0  2  2   1   2   2   2   2   2   1   1   0   1   2   1   0   0   2   2    2    1
##   HG00101  0  2  2   1   2   2   2   2   1   0   2   1   2   2   1   0   0   2   2    1    2
##   HG00102  1  2  2   1   2   2   2   2   2   0   1   0   1   2   2   0   0   2   2    1    1</code></pre>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="gds-format.html#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="co"># count of *any* alternate alleles</span></span>
<span id="cb76-2"><a href="gds-format.html#cb76-2" aria-hidden="true" tabindex="-1"></a>dos <span class="ot">&lt;-</span> <span class="fu">altDosage</span>(gds)</span>
<span id="cb76-3"><a href="gds-format.html#cb76-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(dos)</span></code></pre></div>
<pre><code>##          variant
## sample    30 69 73 161 162 195 243 253 407 431 434 610 627 645 689 756 765 814 988 1014 1056
##   HG00096  1  0  0   1   0   0   0   0   0   1   1   2   0   0   1   2   2   0   0    0    0
##   HG00097  2  0  0   0   0   0   0   0   0   2   1   2   0   0   0   2   2   0   0    0    0
##   HG00099  1  0  0   0   0   0   0   0   0   0   1   2   0   0   0   2   2   0   0    0    0
##   HG00100  2  0  0   1   0   0   0   0   0   1   1   2   1   0   1   2   2   0   0    0    1
##   HG00101  2  0  0   1   0   0   0   0   1   2   0   1   0   0   1   2   2   0   0    1    0
##   HG00102  1  0  0   1   0   0   0   0   0   2   1   2   1   0   0   2   2   0   0    1    1</code></pre>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="gds-format.html#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="co"># count of the first alternate allele </span></span>
<span id="cb78-2"><a href="gds-format.html#cb78-2" aria-hidden="true" tabindex="-1"></a>dos <span class="ot">&lt;-</span> <span class="fu">alleleDosage</span>(gds, <span class="at">n=</span><span class="dv">1</span>)</span>
<span id="cb78-3"><a href="gds-format.html#cb78-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(dos)</span></code></pre></div>
<pre><code>##          variant
## sample    30 69 73 161 162 195 243 253 407 431 434 610 627 645 689 756 765 814 988 1014 1056
##   HG00096  1  0  0   1   0   0   0   0   0   1   1   0   0   0   0   0   0   0   0    0    0
##   HG00097  2  0  0   0   0   0   0   0   0   0   1   1   0   0   0   0   0   0   0    0    0
##   HG00099  1  0  0   0   0   0   0   0   0   0   1   1   0   0   0   0   0   0   0    0    0
##   HG00100  2  0  0   1   0   0   0   0   0   0   1   2   0   0   0   0   0   0   0    0    0
##   HG00101  2  0  0   1   0   0   0   0   0   2   0   0   0   0   0   0   0   0   0    1    0
##   HG00102  1  0  0   1   0   0   0   0   0   0   1   0   0   0   0   0   1   0   0    1    0</code></pre>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="gds-format.html#cb80-1" aria-hidden="true" tabindex="-1"></a><span class="co"># count of the third alternate allele</span></span>
<span id="cb80-2"><a href="gds-format.html#cb80-2" aria-hidden="true" tabindex="-1"></a>dos <span class="ot">&lt;-</span> <span class="fu">alleleDosage</span>(gds, <span class="at">n=</span><span class="dv">3</span>)</span>
<span id="cb80-3"><a href="gds-format.html#cb80-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(dos)</span></code></pre></div>
<pre><code>##          variant
## sample    30 69 73 161 162 195 243 253 407 431 434 610 627 645 689 756 765 814 988 1014 1056
##   HG00096  0  0  0   0   0   0   0   0   0   0   0   2   0   0   0   0   0   0   0    0    0
##   HG00097  0  0  0   0   0   0   0   0   0   0   0   1   0   0   0   0   0   0   0    0    0
##   HG00099  0  0  0   0   0   0   0   0   0   0   0   1   0   0   0   0   0   0   0    0    0
##   HG00100  0  0  0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0    0    0
##   HG00101  0  0  0   0   0   0   0   0   0   0   0   1   0   0   0   0   0   0   0    0    0
##   HG00102  0  0  0   0   0   0   0   0   0   0   0   2   0   0   0   0   0   0   0    0    0</code></pre>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="gds-format.html#cb82-1" aria-hidden="true" tabindex="-1"></a><span class="co"># count of *each* of the alternate alleles</span></span>
<span id="cb82-2"><a href="gds-format.html#cb82-2" aria-hidden="true" tabindex="-1"></a><span class="co"># returns multiple columns per variant</span></span>
<span id="cb82-3"><a href="gds-format.html#cb82-3" aria-hidden="true" tabindex="-1"></a>dos <span class="ot">&lt;-</span> <span class="fu">expandedAltDosage</span>(gds)</span>
<span id="cb82-4"><a href="gds-format.html#cb82-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(dos)</span></code></pre></div>
<pre><code>##          variant
## sample    30 30 69 69 73 73 161 161 162 162 195 195 243 243 253 253 407 407 431 431 434 434 610 610 610 627 627 645 645 689 689 756 756 765 765 814 814 988 988 1014
##   HG00096  1  0  0  0  0  0   1   0   0   0   0   0   0   0   0   0   0   0   1   0   1   0   0   0   2   0   0   0   0   0   1   0   2   0   2   0   0   0   0    0
##   HG00097  2  0  0  0  0  0   0   0   0   0   0   0   0   0   0   0   0   0   0   2   1   0   1   0   1   0   0   0   0   0   0   0   2   0   2   0   0   0   0    0
##   HG00099  1  0  0  0  0  0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   1   0   1   0   1   0   0   0   0   0   0   0   2   0   2   0   0   0   0    0
##   HG00100  2  0  0  0  0  0   1   0   0   0   0   0   0   0   0   0   0   0   0   1   1   0   2   0   0   0   1   0   0   0   1   0   2   0   2   0   0   0   0    0
##   HG00101  2  0  0  0  0  0   1   0   0   0   0   0   0   0   0   0   0   1   2   0   0   0   0   0   1   0   0   0   0   0   1   0   2   0   2   0   0   0   0    1
##   HG00102  1  0  0  0  0  0   1   0   0   0   0   0   0   0   0   0   0   0   0   2   1   0   0   0   2   0   1   0   0   0   0   0   2   1   1   0   0   0   0    1
##          variant
## sample    1014 1056 1056
##   HG00096    0    0    0
##   HG00097    0    0    0
##   HG00099    0    0    0
##   HG00100    0    0    1
##   HG00101    0    0    0
##   HG00102    0    0    1</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>Use the <code>hwe</code> function in SeqVarTools to run a Hardy-Weinberg Equilibrium test on each variant. Identify a variant with low p-value and inspect its genotypes. (Note that the HWE test is only valid for biallelic variants, and will return <code>NA</code> for multiallelic variants.)</li>
</ol>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="gds-format.html#cb84-1" aria-hidden="true" tabindex="-1"></a><span class="co"># reset the filter to all variants</span></span>
<span id="cb84-2"><a href="gds-format.html#cb84-2" aria-hidden="true" tabindex="-1"></a><span class="fu">seqResetFilter</span>(gds)</span></code></pre></div>
<pre><code>## # of selected samples: 1,126
## # of selected variants: 1,120</code></pre>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="gds-format.html#cb86-1" aria-hidden="true" tabindex="-1"></a><span class="co"># run HWE test</span></span>
<span id="cb86-2"><a href="gds-format.html#cb86-2" aria-hidden="true" tabindex="-1"></a>hwe.res <span class="ot">&lt;-</span> <span class="fu">hwe</span>(gds)</span>
<span id="cb86-3"><a href="gds-format.html#cb86-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb86-4"><a href="gds-format.html#cb86-4" aria-hidden="true" tabindex="-1"></a><span class="co"># identify variants with small p-values</span></span>
<span id="cb86-5"><a href="gds-format.html#cb86-5" aria-hidden="true" tabindex="-1"></a>lowp <span class="ot">&lt;-</span> <span class="sc">!</span><span class="fu">is.na</span>(hwe.res<span class="sc">$</span>p) <span class="sc">&amp;</span> hwe.res<span class="sc">$</span>p <span class="sc">&lt;</span> <span class="fl">1e-4</span></span>
<span id="cb86-6"><a href="gds-format.html#cb86-6" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(hwe.res[lowp,])</span></code></pre></div>
<pre><code>##     variant.id nAA nAa naa     afreq            p         f
## 75          75 702 336  88 0.7726465 1.070663e-06 0.1506466
## 92          92 632 381 113 0.7304618 3.558878e-06 0.1407120
## 98          98 672 335 119 0.7455595 2.369695e-12 0.2158342
## 105        105  93 272 761 0.2033748 7.851777e-16 0.2544970
## 114        114 299 482 345 0.4795737 1.745346e-06 0.1424409
## 150        150 471 447 208 0.6167851 8.020208e-08 0.1602251</code></pre>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="gds-format.html#cb88-1" aria-hidden="true" tabindex="-1"></a><span class="co"># look at the genotypes of the most significant variant</span></span>
<span id="cb88-2"><a href="gds-format.html#cb88-2" aria-hidden="true" tabindex="-1"></a>minp <span class="ot">&lt;-</span> <span class="fu">which.min</span>(hwe.res<span class="sc">$</span>p)</span>
<span id="cb88-3"><a href="gds-format.html#cb88-3" aria-hidden="true" tabindex="-1"></a>hwe.res[minp,]</span></code></pre></div>
<pre><code>##     variant.id nAA  nAa naa     afreq             p          f
## 608        608  76 1048   2 0.5328597 5.211823e-234 -0.8695311</code></pre>
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb90-1"><a href="gds-format.html#cb90-1" aria-hidden="true" tabindex="-1"></a><span class="fu">seqSetFilter</span>(gds, <span class="at">variant.id=</span>minp)</span></code></pre></div>
<pre><code>## # of selected variants: 1</code></pre>
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb92-1"><a href="gds-format.html#cb92-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">getGenotype</span>(gds))</span></code></pre></div>
<pre><code>## 
## 0|0 0|1 1|0 1|1 
##  76 559 489   2</code></pre>
<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb94-1"><a href="gds-format.html#cb94-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">altDosage</span>(gds))</span></code></pre></div>
<pre><code>## 
##    0    1    2 
##   76 1048    2</code></pre>
<p>Dont forget to close your connection to the GDS file when youre done!</p>
<div class="sourceCode" id="cb96"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb96-1"><a href="gds-format.html#cb96-1" aria-hidden="true" tabindex="-1"></a><span class="fu">seqClose</span>(gds)</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="phenotype-harmonization.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
