<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>7 Mixed models | SISG 2021 Module 16: Computational Pipeline for WGS Data</title>
  <meta name="description" content="Course materials for SISG 2021 Module 16" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="7 Mixed models | SISG 2021 Module 16: Computational Pipeline for WGS Data" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Course materials for SISG 2021 Module 16" />
  <meta name="github-repo" content="UW-GAC/SISG_2021" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="7 Mixed models | SISG 2021 Module 16: Computational Pipeline for WGS Data" />
  
  <meta name="twitter:description" content="Course materials for SISG 2021 Module 16" />
  



<meta name="date" content="2021-07-20" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="ancestry-and-relatedness-inference.html"/>
<link rel="next" href="variant-annotation.html"/>
<script src="libs/header-attrs-2.9/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#schedule"><i class="fa fa-check"></i><b>1.1</b> Schedule</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#r-packages-used"><i class="fa fa-check"></i><b>1.2</b> R packages used</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#resources"><i class="fa fa-check"></i><b>1.3</b> Resources</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="gds-format.html"><a href="gds-format.html"><i class="fa fa-check"></i><b>2</b> GDS format</a>
<ul>
<li class="chapter" data-level="2.1" data-path="gds-format.html"><a href="gds-format.html#convert-a-vcf-to-gds"><i class="fa fa-check"></i><b>2.1</b> Convert a VCF to GDS</a></li>
<li class="chapter" data-level="2.2" data-path="gds-format.html"><a href="gds-format.html#exploring-a-gds-file"><i class="fa fa-check"></i><b>2.2</b> Exploring a GDS file</a></li>
<li class="chapter" data-level="2.3" data-path="gds-format.html"><a href="gds-format.html#exercises"><i class="fa fa-check"></i><b>2.3</b> Exercises</a></li>
<li class="chapter" data-level="2.4" data-path="gds-format.html"><a href="gds-format.html#solutions"><i class="fa fa-check"></i><b>2.4</b> Solutions</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="phenotype-harmonization.html"><a href="phenotype-harmonization.html"><i class="fa fa-check"></i><b>3</b> Phenotype Harmonization</a>
<ul>
<li class="chapter" data-level="3.1" data-path="phenotype-harmonization.html"><a href="phenotype-harmonization.html#inspect-individual-study-data-in-r"><i class="fa fa-check"></i><b>3.1</b> Inspect individual study data in R</a></li>
<li class="chapter" data-level="3.2" data-path="phenotype-harmonization.html"><a href="phenotype-harmonization.html#compare-study-values"><i class="fa fa-check"></i><b>3.2</b> Compare study values</a></li>
<li class="chapter" data-level="3.3" data-path="phenotype-harmonization.html"><a href="phenotype-harmonization.html#using-regression-models-to-compare-studies"><i class="fa fa-check"></i><b>3.3</b> Using regression models to compare studies</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="phenotype-harmonization.html"><a href="phenotype-harmonization.html#create-an-annotated-data-frame"><i class="fa fa-check"></i><b>3.3.1</b> Create an Annotated Data Frame</a></li>
<li class="chapter" data-level="3.3.2" data-path="phenotype-harmonization.html"><a href="phenotype-harmonization.html#fit-a-regression-model-without-study"><i class="fa fa-check"></i><b>3.3.2</b> Fit a regression model without study</a></li>
<li class="chapter" data-level="3.3.3" data-path="phenotype-harmonization.html"><a href="phenotype-harmonization.html#fit-a-model-with-study"><i class="fa fa-check"></i><b>3.3.3</b> Fit a model with study</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="phenotype-harmonization.html"><a href="phenotype-harmonization.html#final-considerations"><i class="fa fa-check"></i><b>3.4</b> Final considerations</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="association-tests-part-i.html"><a href="association-tests-part-i.html"><i class="fa fa-check"></i><b>4</b> Association tests - Part I</a>
<ul>
<li class="chapter" data-level="4.1" data-path="association-tests-part-i.html"><a href="association-tests-part-i.html#null-model"><i class="fa fa-check"></i><b>4.1</b> Null model</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="association-tests-part-i.html"><a href="association-tests-part-i.html#prepare-the-data"><i class="fa fa-check"></i><b>4.1.1</b> Prepare the data</a></li>
<li class="chapter" data-level="4.1.2" data-path="association-tests-part-i.html"><a href="association-tests-part-i.html#fit-the-null-model"><i class="fa fa-check"></i><b>4.1.2</b> Fit the null model</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="association-tests-part-i.html"><a href="association-tests-part-i.html#exercise"><i class="fa fa-check"></i><b>4.2</b> Exercise</a></li>
<li class="chapter" data-level="4.3" data-path="association-tests-part-i.html"><a href="association-tests-part-i.html#solution"><i class="fa fa-check"></i><b>4.3</b> Solution</a></li>
<li class="chapter" data-level="4.4" data-path="association-tests-part-i.html"><a href="association-tests-part-i.html#single-variant-association-tests"><i class="fa fa-check"></i><b>4.4</b> Single-variant association tests</a></li>
<li class="chapter" data-level="4.5" data-path="association-tests-part-i.html"><a href="association-tests-part-i.html#exercise-1"><i class="fa fa-check"></i><b>4.5</b> Exercise</a></li>
<li class="chapter" data-level="4.6" data-path="association-tests-part-i.html"><a href="association-tests-part-i.html#solution-1"><i class="fa fa-check"></i><b>4.6</b> Solution</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="association-tests-part-ii.html"><a href="association-tests-part-ii.html"><i class="fa fa-check"></i><b>5</b> Association tests - Part II</a>
<ul>
<li class="chapter" data-level="5.1" data-path="association-tests-part-ii.html"><a href="association-tests-part-ii.html#sliding-window-tests"><i class="fa fa-check"></i><b>5.1</b> Sliding window tests</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="association-tests-part-ii.html"><a href="association-tests-part-ii.html#burden-test"><i class="fa fa-check"></i><b>5.1.1</b> Burden test</a></li>
<li class="chapter" data-level="5.1.2" data-path="association-tests-part-ii.html"><a href="association-tests-part-ii.html#skat-test"><i class="fa fa-check"></i><b>5.1.2</b> SKAT test</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="association-tests-part-ii.html"><a href="association-tests-part-ii.html#exercise-2"><i class="fa fa-check"></i><b>5.2</b> Exercise</a></li>
<li class="chapter" data-level="5.3" data-path="association-tests-part-ii.html"><a href="association-tests-part-ii.html#solution-2"><i class="fa fa-check"></i><b>5.3</b> Solution</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="ancestry-and-relatedness-inference.html"><a href="ancestry-and-relatedness-inference.html"><i class="fa fa-check"></i><b>6</b> Ancestry and Relatedness Inference</a>
<ul>
<li class="chapter" data-level="6.1" data-path="ancestry-and-relatedness-inference.html"><a href="ancestry-and-relatedness-inference.html#ld-pruning"><i class="fa fa-check"></i><b>6.1</b> LD-pruning</a></li>
<li class="chapter" data-level="6.2" data-path="ancestry-and-relatedness-inference.html"><a href="ancestry-and-relatedness-inference.html#computing-a-grm"><i class="fa fa-check"></i><b>6.2</b> Computing a GRM</a></li>
<li class="chapter" data-level="6.3" data-path="ancestry-and-relatedness-inference.html"><a href="ancestry-and-relatedness-inference.html#de-convoluting-ancestry-and-relatedness"><i class="fa fa-check"></i><b>6.3</b> De-convoluting ancestry and relatedness</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="ancestry-and-relatedness-inference.html"><a href="ancestry-and-relatedness-inference.html#king"><i class="fa fa-check"></i><b>6.3.1</b> KING</a></li>
<li class="chapter" data-level="6.3.2" data-path="ancestry-and-relatedness-inference.html"><a href="ancestry-and-relatedness-inference.html#pc-air"><i class="fa fa-check"></i><b>6.3.2</b> PC-AiR</a></li>
<li class="chapter" data-level="6.3.3" data-path="ancestry-and-relatedness-inference.html"><a href="ancestry-and-relatedness-inference.html#pc-relate"><i class="fa fa-check"></i><b>6.3.3</b> PC-Relate</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="ancestry-and-relatedness-inference.html"><a href="ancestry-and-relatedness-inference.html#exercises-1"><i class="fa fa-check"></i><b>6.4</b> Exercises</a></li>
<li class="chapter" data-level="6.5" data-path="ancestry-and-relatedness-inference.html"><a href="ancestry-and-relatedness-inference.html#solutions-1"><i class="fa fa-check"></i><b>6.5</b> Solutions</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="mixed-models.html"><a href="mixed-models.html"><i class="fa fa-check"></i><b>7</b> Mixed models</a>
<ul>
<li class="chapter" data-level="7.1" data-path="mixed-models.html"><a href="mixed-models.html#null-model-1"><i class="fa fa-check"></i><b>7.1</b> Null model</a></li>
<li class="chapter" data-level="7.2" data-path="mixed-models.html"><a href="mixed-models.html#single-variant-tests"><i class="fa fa-check"></i><b>7.2</b> Single-variant tests</a></li>
<li class="chapter" data-level="7.3" data-path="mixed-models.html"><a href="mixed-models.html#exercise-3"><i class="fa fa-check"></i><b>7.3</b> Exercise</a></li>
<li class="chapter" data-level="7.4" data-path="mixed-models.html"><a href="mixed-models.html#solution-3"><i class="fa fa-check"></i><b>7.4</b> Solution</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="variant-annotation.html"><a href="variant-annotation.html"><i class="fa fa-check"></i><b>8</b> Variant annotation</a>
<ul>
<li class="chapter" data-level="8.1" data-path="variant-annotation.html"><a href="variant-annotation.html#use-cases"><i class="fa fa-check"></i><b>8.1</b> Use cases</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="variant-annotation.html"><a href="variant-annotation.html#use-case-1"><i class="fa fa-check"></i><b>8.1.1</b> Use case 1</a></li>
<li class="chapter" data-level="8.1.2" data-path="variant-annotation.html"><a href="variant-annotation.html#use-case-2"><i class="fa fa-check"></i><b>8.1.2</b> Use case 2</a></li>
<li class="chapter" data-level="8.1.3" data-path="variant-annotation.html"><a href="variant-annotation.html#use-case-3"><i class="fa fa-check"></i><b>8.1.3</b> Use case 3</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="variant-annotation.html"><a href="variant-annotation.html#exercise-4"><i class="fa fa-check"></i><b>8.2</b> Exercise</a></li>
<li class="chapter" data-level="8.3" data-path="variant-annotation.html"><a href="variant-annotation.html#solution-4"><i class="fa fa-check"></i><b>8.3</b> Solution</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="annotation-informed-aggregate-association-tests.html"><a href="annotation-informed-aggregate-association-tests.html"><i class="fa fa-check"></i><b>9</b> Annotation informed aggregate association tests</a>
<ul>
<li class="chapter" data-level="9.1" data-path="annotation-informed-aggregate-association-tests.html"><a href="annotation-informed-aggregate-association-tests.html#aggregate-unit-for-association-testing-exercise"><i class="fa fa-check"></i><b>9.1</b> Aggregate unit for association testing exercise</a></li>
<li class="chapter" data-level="9.2" data-path="annotation-informed-aggregate-association-tests.html"><a href="annotation-informed-aggregate-association-tests.html#association-testing-with-aggregate-units"><i class="fa fa-check"></i><b>9.2</b> Association testing with aggregate units</a></li>
<li class="chapter" data-level="9.3" data-path="annotation-informed-aggregate-association-tests.html"><a href="annotation-informed-aggregate-association-tests.html#exercise-5"><i class="fa fa-check"></i><b>9.3</b> Exercise</a></li>
<li class="chapter" data-level="9.4" data-path="annotation-informed-aggregate-association-tests.html"><a href="annotation-informed-aggregate-association-tests.html#solution-5"><i class="fa fa-check"></i><b>9.4</b> Solution</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="running-a-gwas-workflow-on-biodata-catalyst.html"><a href="running-a-gwas-workflow-on-biodata-catalyst.html"><i class="fa fa-check"></i><b>10</b> Running a GWAS workflow on BioData Catalyst</a>
<ul>
<li class="chapter" data-level="10.1" data-path="running-a-gwas-workflow-on-biodata-catalyst.html"><a href="running-a-gwas-workflow-on-biodata-catalyst.html#preparing-the-genotype-data"><i class="fa fa-check"></i><b>10.1</b> Preparing the genotype data</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="running-a-gwas-workflow-on-biodata-catalyst.html"><a href="running-a-gwas-workflow-on-biodata-catalyst.html#bcftools-merge-and-filter"><i class="fa fa-check"></i><b>10.1.1</b> Bcftools Merge and Filter</a></li>
<li class="chapter" data-level="10.1.2" data-path="running-a-gwas-workflow-on-biodata-catalyst.html"><a href="running-a-gwas-workflow-on-biodata-catalyst.html#vcf-to-gds-converter"><i class="fa fa-check"></i><b>10.1.2</b> VCF to GDS Converter</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="running-a-gwas-workflow-on-biodata-catalyst.html"><a href="running-a-gwas-workflow-on-biodata-catalyst.html#ancestry-and-relatedness"><i class="fa fa-check"></i><b>10.2</b> Ancestry and Relatedness</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="running-a-gwas-workflow-on-biodata-catalyst.html"><a href="running-a-gwas-workflow-on-biodata-catalyst.html#king-robust"><i class="fa fa-check"></i><b>10.2.1</b> KING robust</a></li>
<li class="chapter" data-level="10.2.2" data-path="running-a-gwas-workflow-on-biodata-catalyst.html"><a href="running-a-gwas-workflow-on-biodata-catalyst.html#pc-air-1"><i class="fa fa-check"></i><b>10.2.2</b> PC-AiR</a></li>
<li class="chapter" data-level="10.2.3" data-path="running-a-gwas-workflow-on-biodata-catalyst.html"><a href="running-a-gwas-workflow-on-biodata-catalyst.html#pc-relate-1"><i class="fa fa-check"></i><b>10.2.3</b> PC-Relate</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="running-a-gwas-workflow-on-biodata-catalyst.html"><a href="running-a-gwas-workflow-on-biodata-catalyst.html#association-testing"><i class="fa fa-check"></i><b>10.3</b> Association Testing</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="running-a-gwas-workflow-on-biodata-catalyst.html"><a href="running-a-gwas-workflow-on-biodata-catalyst.html#null-model-2"><i class="fa fa-check"></i><b>10.3.1</b> Null Model</a></li>
<li class="chapter" data-level="10.3.2" data-path="running-a-gwas-workflow-on-biodata-catalyst.html"><a href="running-a-gwas-workflow-on-biodata-catalyst.html#single-variant-association-test"><i class="fa fa-check"></i><b>10.3.2</b> Single variant association test</a></li>
<li class="chapter" data-level="10.3.3" data-path="running-a-gwas-workflow-on-biodata-catalyst.html"><a href="running-a-gwas-workflow-on-biodata-catalyst.html#aggregate-variant-test"><i class="fa fa-check"></i><b>10.3.3</b> Aggregate variant test</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="running-a-gwas-workflow-on-biodata-catalyst.html"><a href="running-a-gwas-workflow-on-biodata-catalyst.html#analysis-follow-up"><i class="fa fa-check"></i><b>10.4</b> Analysis follow-up</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">SISG 2021 Module 16: Computational Pipeline for WGS Data</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="mixed-models" class="section level1" number="7">
<h1><span class="header-section-number">7</span> Mixed models</h1>
<p>These exercises extend what was previously introduced in the association tests from regression models to mixed models that account for genetic relatedness among samples.</p>
<div id="null-model-1" class="section level2" number="7.1">
<h2><span class="header-section-number">7.1</span> Null model</h2>
<p>Recall that the first step in our association testing procedure is to fit the null model. In addition to the <code>AnnotatedDataFrame</code> with phenotype data that we used previously, we will also use the ancestry PCs and pairwise kinship estimates we created in the previous session. We will use the first 4 PCs to adjust for ancestry.</p>
<div class="sourceCode" id="cb304"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb304-1"><a href="mixed-models.html#cb304-1" aria-hidden="true" tabindex="-1"></a><span class="co"># sample annotation</span></span>
<span id="cb304-2"><a href="mixed-models.html#cb304-2" aria-hidden="true" tabindex="-1"></a>repo_path <span class="ot">&lt;-</span> <span class="st">&quot;https://github.com/UW-GAC/SISG_2021/raw/master&quot;</span></span>
<span id="cb304-3"><a href="mixed-models.html#cb304-3" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">dir.exists</span>(<span class="st">&quot;data&quot;</span>)) <span class="fu">dir.create</span>(<span class="st">&quot;data&quot;</span>)</span>
<span id="cb304-4"><a href="mixed-models.html#cb304-4" aria-hidden="true" tabindex="-1"></a>sampfile <span class="ot">&lt;-</span> <span class="st">&quot;data/sample_phenotype_annotation.RData&quot;</span></span>
<span id="cb304-5"><a href="mixed-models.html#cb304-5" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(sampfile)) <span class="fu">download.file</span>(<span class="fu">file.path</span>(repo_path, sampfile), sampfile)</span>
<span id="cb304-6"><a href="mixed-models.html#cb304-6" aria-hidden="true" tabindex="-1"></a>annot <span class="ot">&lt;-</span> <span class="fu">get</span>(<span class="fu">load</span>(sampfile))</span>
<span id="cb304-7"><a href="mixed-models.html#cb304-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Biobase)</span>
<span id="cb304-8"><a href="mixed-models.html#cb304-8" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">pData</span>(annot))</span></code></pre></div>
<pre><code>##   sample.id subject.id Population          Population.Description sex status age  height   study
## 1   HG00096    HG00096        GBR British in England and Scotland   M      0  47 165.300 study_1
## 2   HG00097    HG00097        GBR British in England and Scotland   F      1  47 144.780 study_3
## 3   HG00099    HG00099        GBR British in England and Scotland   F      0  40 185.500 study_2
## 4   HG00100    HG00100        GBR British in England and Scotland   F      1  45 150.622 study_3
## 5   HG00101    HG00101        GBR British in England and Scotland   M      0  40 177.800 study_3
## 6   HG00102    HG00102        GBR British in England and Scotland   F      0  49 169.100 study_1</code></pre>
<div class="sourceCode" id="cb306"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb306-1"><a href="mixed-models.html#cb306-1" aria-hidden="true" tabindex="-1"></a><span class="co"># load the ancestry PCs</span></span>
<span id="cb306-2"><a href="mixed-models.html#cb306-2" aria-hidden="true" tabindex="-1"></a>pcfile <span class="ot">&lt;-</span> <span class="st">&quot;data/pcs.RData&quot;</span></span>
<span id="cb306-3"><a href="mixed-models.html#cb306-3" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(pcfile)) <span class="fu">download.file</span>(<span class="fu">file.path</span>(repo_path, pcfile), pcfile)</span>
<span id="cb306-4"><a href="mixed-models.html#cb306-4" aria-hidden="true" tabindex="-1"></a>pcs <span class="ot">&lt;-</span> <span class="fu">get</span>(<span class="fu">load</span>(pcfile))</span>
<span id="cb306-5"><a href="mixed-models.html#cb306-5" aria-hidden="true" tabindex="-1"></a>pcs <span class="ot">&lt;-</span> pcs[,<span class="fu">c</span>(<span class="st">&quot;sample.id&quot;</span>, <span class="st">&quot;PC1&quot;</span>, <span class="st">&quot;PC2&quot;</span>, <span class="st">&quot;PC3&quot;</span>, <span class="st">&quot;PC4&quot;</span>)]</span>
<span id="cb306-6"><a href="mixed-models.html#cb306-6" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(pcs)</span></code></pre></div>
<pre><code>##         sample.id         PC1         PC2         PC3          PC4
## HG00096   HG00096 -0.02189121 -0.03637024 -0.01328390  0.005533490
## HG00097   HG00097 -0.02002924 -0.03285711 -0.01207786  0.005390183
## HG00099   HG00099 -0.02088688 -0.03330825 -0.01510526  0.004499776
## HG00100   HG00100 -0.02031270 -0.03975678 -0.01908062  0.004291604
## HG00101   HG00101 -0.02038482 -0.03437442 -0.01328870  0.002310498
## HG00102   HG00102 -0.02144725 -0.03970277 -0.01391896 -0.001245155</code></pre>
<div class="sourceCode" id="cb308"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb308-1"><a href="mixed-models.html#cb308-1" aria-hidden="true" tabindex="-1"></a><span class="co"># merge PCs with the sample annotation</span></span>
<span id="cb308-2"><a href="mixed-models.html#cb308-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb308-3"><a href="mixed-models.html#cb308-3" aria-hidden="true" tabindex="-1"></a>dat <span class="ot">&lt;-</span> <span class="fu">left_join</span>(<span class="fu">pData</span>(annot), pcs, <span class="at">by=</span><span class="st">&quot;sample.id&quot;</span>)</span>
<span id="cb308-4"><a href="mixed-models.html#cb308-4" aria-hidden="true" tabindex="-1"></a><span class="co"># update the AnnotatedDataFrame</span></span>
<span id="cb308-5"><a href="mixed-models.html#cb308-5" aria-hidden="true" tabindex="-1"></a><span class="fu">pData</span>(annot) <span class="ot">&lt;-</span> dat</span>
<span id="cb308-6"><a href="mixed-models.html#cb308-6" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(annot, <span class="at">file=</span><span class="st">&quot;data/sample_phenotype_pcs.RData&quot;</span>)</span></code></pre></div>
<p>We can create a kinship matrix from the output of <code>pcrelate</code>. We multiply the kinship values by 2 to get values on the same scale as the standard GRM. This matrix is represented in R as a symmetric matrix object from the Matrix package.</p>
<div class="sourceCode" id="cb309"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb309-1"><a href="mixed-models.html#cb309-1" aria-hidden="true" tabindex="-1"></a>kinfile <span class="ot">&lt;-</span> <span class="st">&quot;data/pcrelate_kinship.RData&quot;</span></span>
<span id="cb309-2"><a href="mixed-models.html#cb309-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(kinfile)) <span class="fu">download.file</span>(<span class="fu">file.path</span>(repo_path, kinfile), kinfile)</span>
<span id="cb309-3"><a href="mixed-models.html#cb309-3" aria-hidden="true" tabindex="-1"></a>pcrel <span class="ot">&lt;-</span> <span class="fu">get</span>(<span class="fu">load</span>(kinfile))</span>
<span id="cb309-4"><a href="mixed-models.html#cb309-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GENESIS)</span>
<span id="cb309-5"><a href="mixed-models.html#cb309-5" aria-hidden="true" tabindex="-1"></a>kinship <span class="ot">&lt;-</span> <span class="fu">pcrelateToMatrix</span>(pcrel, <span class="at">scaleKin=</span><span class="dv">2</span>, <span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb309-6"><a href="mixed-models.html#cb309-6" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(kinship)</span></code></pre></div>
<pre><code>## [1] 1126 1126</code></pre>
<div class="sourceCode" id="cb311"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb311-1"><a href="mixed-models.html#cb311-1" aria-hidden="true" tabindex="-1"></a>kinship[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>## 5 x 5 Matrix of class &quot;dsyMatrix&quot;
##              HG00096      HG00097      HG00099      HG00100      HG00101
## HG00096 0.9906231740  0.036922756  0.010409075 0.0005780419  0.016522827
## HG00097 0.0369227560  1.001560563 -0.000653982 0.0035158297  0.007678616
## HG00099 0.0104090751 -0.000653982  0.937748649 0.0556213201 -0.003235074
## HG00100 0.0005780419  0.003515830  0.055621320 1.0111688406  0.017959707
## HG00101 0.0165228269  0.007678616 -0.003235074 0.0179597069  0.965028548</code></pre>
<p>When running a mixed model analysis, we still fit the null model using the <code>fitNullModel</code> function in GENESIS. Now, we include the kinship matrix in the model with the <code>cov.mat</code> argument, which is used to specify the random effect(s) in the model with covariance structure(s) proportional to the supplied matrix(s). The inclusion of these random effects is what makes this a mixed model, rather than a simple regression model. We also add the ancestry PCs to the list of covariates and allow for heterogeneous residual variance by <code>study</code> with the <code>group.var</code> argument, as before.</p>
<div class="sourceCode" id="cb313"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb313-1"><a href="mixed-models.html#cb313-1" aria-hidden="true" tabindex="-1"></a>nullmod <span class="ot">&lt;-</span> <span class="fu">fitNullModel</span>(annot, </span>
<span id="cb313-2"><a href="mixed-models.html#cb313-2" aria-hidden="true" tabindex="-1"></a>                        <span class="at">outcome=</span><span class="st">&quot;height&quot;</span>, </span>
<span id="cb313-3"><a href="mixed-models.html#cb313-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">covars=</span><span class="fu">c</span>(<span class="st">&quot;sex&quot;</span>, <span class="st">&quot;age&quot;</span>, <span class="st">&quot;study&quot;</span>, <span class="fu">paste0</span>(<span class="st">&quot;PC&quot;</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>)), </span>
<span id="cb313-4"><a href="mixed-models.html#cb313-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">cov.mat=</span>kinship, </span>
<span id="cb313-5"><a href="mixed-models.html#cb313-5" aria-hidden="true" tabindex="-1"></a>                        <span class="at">group.var=</span><span class="st">&quot;study&quot;</span>, </span>
<span id="cb313-6"><a href="mixed-models.html#cb313-6" aria-hidden="true" tabindex="-1"></a>                        <span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb313-7"><a href="mixed-models.html#cb313-7" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(nullmod, <span class="at">file=</span><span class="st">&quot;data/null_mixed_model.RData&quot;</span>)</span></code></pre></div>
<p>We can investigate the output from <code>fitNullModel</code>.</p>
<div class="sourceCode" id="cb314"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb314-1"><a href="mixed-models.html#cb314-1" aria-hidden="true" tabindex="-1"></a><span class="co"># description of the model we fit</span></span>
<span id="cb314-2"><a href="mixed-models.html#cb314-2" aria-hidden="true" tabindex="-1"></a>nullmod<span class="sc">$</span>model</span></code></pre></div>
<pre><code>## $hetResid
## [1] TRUE
## 
## $family
## 
## Family: gaussian 
## Link function: identity 
## 
## 
## $outcome
## [1] &quot;height&quot;
## 
## $covars
## [1] &quot;sex&quot;   &quot;age&quot;   &quot;study&quot; &quot;PC1&quot;   &quot;PC2&quot;   &quot;PC3&quot;   &quot;PC4&quot;  
## 
## $formula
## [1] &quot;height ~ sex + age + study + PC1 + PC2 + PC3 + PC4 + (1|A) + var(study)&quot;</code></pre>
<div class="sourceCode" id="cb316"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb316-1"><a href="mixed-models.html#cb316-1" aria-hidden="true" tabindex="-1"></a><span class="co"># fixed effect regression estimates</span></span>
<span id="cb316-2"><a href="mixed-models.html#cb316-2" aria-hidden="true" tabindex="-1"></a>nullmod<span class="sc">$</span>fixef</span></code></pre></div>
<pre><code>##                       Est          SE         Stat         pval
## (Intercept)  163.74431361  3.15130038 2.699932e+03 0.000000e+00
## sexM           6.27072759  0.67911122 8.526171e+01 2.613754e-20
## age            0.07420444  0.06839188 1.177201e+00 2.779265e-01
## studystudy_2  10.54243352  0.81122848 1.688867e+02 1.295192e-38
## studystudy_3  -8.95546858  0.83932675 1.138451e+02 1.408933e-26
## PC1            0.05117574 11.25310400 2.068158e-05 9.963715e-01
## PC2          -61.24485338 11.22450534 2.977178e+01 4.860157e-08
## PC3           -8.81351399 11.09566802 6.309447e-01 4.270090e-01
## PC4           -5.61519972 11.19652394 2.515151e-01 6.160103e-01</code></pre>
<div class="sourceCode" id="cb318"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb318-1"><a href="mixed-models.html#cb318-1" aria-hidden="true" tabindex="-1"></a><span class="co"># variance component estimates by group.var</span></span>
<span id="cb318-2"><a href="mixed-models.html#cb318-2" aria-hidden="true" tabindex="-1"></a>nullmod<span class="sc">$</span>varComp</span></code></pre></div>
<pre><code>##       V_A V_study_1 V_study_3 V_study_2 
##   0.00000  93.61147 168.76021 152.37404</code></pre>
<div class="sourceCode" id="cb320"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb320-1"><a href="mixed-models.html#cb320-1" aria-hidden="true" tabindex="-1"></a><span class="co"># model fit: fitted values, residuals</span></span>
<span id="cb320-2"><a href="mixed-models.html#cb320-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(nullmod<span class="sc">$</span>fit)</span></code></pre></div>
<pre><code>##         outcome workingY fitted.values resid.marginal resid.conditional linear.predictor     resid.PY resid.cholesky sample.id
## HG00096 165.300  165.300      175.8150    -10.5150254       -10.5150254         175.8150 -0.112326247    -1.08679039   HG00096
## HG00097 144.780  144.780      160.3639    -15.5839388       -15.5839388         160.3639 -0.092343679    -1.19961587   HG00097
## HG00099 185.500  185.500      179.4017      6.0983220         6.0983220         179.4017  0.040022053     0.49403175   HG00099
## HG00100 150.622  150.622      160.7060    -10.0839725       -10.0839725         160.7060 -0.059753258    -0.77624108   HG00100
## HG00101 177.800  177.800      166.2361     11.5638906        11.5638906         166.2361  0.068522613     0.89016178   HG00101
## HG00102 169.100  169.100      169.9405     -0.8404905        -0.8404905         169.9405 -0.008978499    -0.08686969   HG00102</code></pre>
<div class="sourceCode" id="cb322"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb322-1"><a href="mixed-models.html#cb322-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb322-2"><a href="mixed-models.html#cb322-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(nullmod<span class="sc">$</span>fit, <span class="fu">aes</span>(<span class="at">x =</span> fitted.values, <span class="at">y =</span> resid.marginal)) <span class="sc">+</span> </span>
<span id="cb322-3"><a href="mixed-models.html#cb322-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span> </span>
<span id="cb322-4"><a href="mixed-models.html#cb322-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>) <span class="sc">+</span> </span>
<span id="cb322-5"><a href="mixed-models.html#cb322-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">&#39;lm&#39;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-23-1.png" width="672" /></p>
</div>
<div id="single-variant-tests" class="section level2" number="7.2">
<h2><span class="header-section-number">7.2</span> Single-variant tests</h2>
<p>Now we can run a single-variant test, accounting for genetic ancestry and genetic relatedness among the subjects. We use the same <code>assocTestSingle</code> function as before; the only difference is that we pass in our new null model.</p>
<div class="sourceCode" id="cb323"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb323-1"><a href="mixed-models.html#cb323-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SeqVarTools)</span>
<span id="cb323-2"><a href="mixed-models.html#cb323-2" aria-hidden="true" tabindex="-1"></a>gdsfile <span class="ot">&lt;-</span> <span class="st">&quot;data/1KG_phase3_subset_chr1.gds&quot;</span></span>
<span id="cb323-3"><a href="mixed-models.html#cb323-3" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(gdsfile)) <span class="fu">download.file</span>(<span class="fu">file.path</span>(repo_path, gdsfile), gdsfile)</span>
<span id="cb323-4"><a href="mixed-models.html#cb323-4" aria-hidden="true" tabindex="-1"></a>gdsfmt<span class="sc">::</span><span class="fu">showfile.gds</span>(<span class="at">closeall=</span><span class="cn">TRUE</span>) <span class="co"># make sure file is not already open</span></span>
<span id="cb323-5"><a href="mixed-models.html#cb323-5" aria-hidden="true" tabindex="-1"></a>gds <span class="ot">&lt;-</span> <span class="fu">seqOpen</span>(gdsfile)</span>
<span id="cb323-6"><a href="mixed-models.html#cb323-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb323-7"><a href="mixed-models.html#cb323-7" aria-hidden="true" tabindex="-1"></a><span class="co"># make the seqVarData object</span></span>
<span id="cb323-8"><a href="mixed-models.html#cb323-8" aria-hidden="true" tabindex="-1"></a>seqData <span class="ot">&lt;-</span> <span class="fu">SeqVarData</span>(gds, <span class="at">sampleData=</span>annot)</span>
<span id="cb323-9"><a href="mixed-models.html#cb323-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb323-10"><a href="mixed-models.html#cb323-10" aria-hidden="true" tabindex="-1"></a><span class="co"># make the iterator object</span></span>
<span id="cb323-11"><a href="mixed-models.html#cb323-11" aria-hidden="true" tabindex="-1"></a>iterator <span class="ot">&lt;-</span> <span class="fu">SeqVarBlockIterator</span>(seqData, <span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb323-12"><a href="mixed-models.html#cb323-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb323-13"><a href="mixed-models.html#cb323-13" aria-hidden="true" tabindex="-1"></a><span class="co"># run the single-variant association test</span></span>
<span id="cb323-14"><a href="mixed-models.html#cb323-14" aria-hidden="true" tabindex="-1"></a>assoc <span class="ot">&lt;-</span> <span class="fu">assocTestSingle</span>(iterator, nullmod)</span></code></pre></div>
<pre><code>## # of selected samples: 1,126</code></pre>
<div class="sourceCode" id="cb325"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb325-1"><a href="mixed-models.html#cb325-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(assoc)</span></code></pre></div>
<pre><code>## [1] 1129   14</code></pre>
<div class="sourceCode" id="cb327"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb327-1"><a href="mixed-models.html#cb327-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(assoc)</span></code></pre></div>
<pre><code>##   variant.id chr     pos allele.index n.obs         freq MAC       Score  Score.SE Score.Stat  Score.pval        Est   Est.SE          PVE
## 1          1   1  970546            1  1126 0.0039964476   9 -0.17384025 0.2592275 -0.6706088 0.502469798  -2.586950 3.857615 0.0004026103
## 2          2   1  985900            1  1126 0.0492895204 111 -1.55598468 0.8183596 -1.9013460 0.057256708  -2.323363 1.221957 0.0032364488
## 3          3   1 1025045            1  1126 0.0004440497   1 -0.29433530 0.1029724 -2.8583911 0.004257953 -27.758817 9.711343 0.0073145856
## 4          4   1 1265550            1  1126 0.0008880995   2 -0.09940909 0.1078116 -0.9220633 0.356495581  -8.552545 9.275442 0.0007611458
## 5          5   1 1472676            1  1126 0.0071047957  16  0.14878210 0.3468438  0.4289599 0.667952399   1.236752 2.883142 0.0001647327
## 6          6   1 1735725            1  1126 0.0022202487   5 -0.13567249 0.1982518 -0.6843443 0.493757749  -3.451895 5.044091 0.0004192719</code></pre>
<p>We make the usual QQ plot, filtering to variants with minor allele count (MAC) &gt;= 5.</p>
<div class="sourceCode" id="cb329"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb329-1"><a href="mixed-models.html#cb329-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb329-2"><a href="mixed-models.html#cb329-2" aria-hidden="true" tabindex="-1"></a>qqPlot <span class="ot">&lt;-</span> <span class="cf">function</span>(pval) {</span>
<span id="cb329-3"><a href="mixed-models.html#cb329-3" aria-hidden="true" tabindex="-1"></a>    pval <span class="ot">&lt;-</span> pval[<span class="sc">!</span><span class="fu">is.na</span>(pval)]</span>
<span id="cb329-4"><a href="mixed-models.html#cb329-4" aria-hidden="true" tabindex="-1"></a>    n <span class="ot">&lt;-</span> <span class="fu">length</span>(pval)</span>
<span id="cb329-5"><a href="mixed-models.html#cb329-5" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">:</span>n</span>
<span id="cb329-6"><a href="mixed-models.html#cb329-6" aria-hidden="true" tabindex="-1"></a>    dat <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">obs=</span><span class="fu">sort</span>(pval),</span>
<span id="cb329-7"><a href="mixed-models.html#cb329-7" aria-hidden="true" tabindex="-1"></a>                      <span class="at">exp=</span>x<span class="sc">/</span>n,</span>
<span id="cb329-8"><a href="mixed-models.html#cb329-8" aria-hidden="true" tabindex="-1"></a>                      <span class="at">upper=</span><span class="fu">qbeta</span>(<span class="fl">0.025</span>, x, <span class="fu">rev</span>(x)),</span>
<span id="cb329-9"><a href="mixed-models.html#cb329-9" aria-hidden="true" tabindex="-1"></a>                      <span class="at">lower=</span><span class="fu">qbeta</span>(<span class="fl">0.975</span>, x, <span class="fu">rev</span>(x)))</span>
<span id="cb329-10"><a href="mixed-models.html#cb329-10" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb329-11"><a href="mixed-models.html#cb329-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(dat, <span class="fu">aes</span>(<span class="sc">-</span><span class="fu">log10</span>(exp), <span class="sc">-</span><span class="fu">log10</span>(obs))) <span class="sc">+</span></span>
<span id="cb329-12"><a href="mixed-models.html#cb329-12" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="sc">-</span><span class="fu">log10</span>(exp), <span class="sc">-</span><span class="fu">log10</span>(upper)), <span class="at">color=</span><span class="st">&quot;gray&quot;</span>) <span class="sc">+</span></span>
<span id="cb329-13"><a href="mixed-models.html#cb329-13" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="sc">-</span><span class="fu">log10</span>(exp), <span class="sc">-</span><span class="fu">log10</span>(lower)), <span class="at">color=</span><span class="st">&quot;gray&quot;</span>) <span class="sc">+</span></span>
<span id="cb329-14"><a href="mixed-models.html#cb329-14" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb329-15"><a href="mixed-models.html#cb329-15" aria-hidden="true" tabindex="-1"></a>        <span class="fu">geom_abline</span>(<span class="at">intercept=</span><span class="dv">0</span>, <span class="at">slope=</span><span class="dv">1</span>, <span class="at">color=</span><span class="st">&quot;red&quot;</span>) <span class="sc">+</span></span>
<span id="cb329-16"><a href="mixed-models.html#cb329-16" aria-hidden="true" tabindex="-1"></a>        <span class="fu">xlab</span>(<span class="fu">expression</span>(<span class="fu">paste</span>(<span class="sc">-</span>log[<span class="dv">10</span>], <span class="st">&quot;(expected P)&quot;</span>))) <span class="sc">+</span></span>
<span id="cb329-17"><a href="mixed-models.html#cb329-17" aria-hidden="true" tabindex="-1"></a>        <span class="fu">ylab</span>(<span class="fu">expression</span>(<span class="fu">paste</span>(<span class="sc">-</span>log[<span class="dv">10</span>], <span class="st">&quot;(observed P)&quot;</span>))) <span class="sc">+</span></span>
<span id="cb329-18"><a href="mixed-models.html#cb329-18" aria-hidden="true" tabindex="-1"></a>        <span class="fu">theme_bw</span>()</span>
<span id="cb329-19"><a href="mixed-models.html#cb329-19" aria-hidden="true" tabindex="-1"></a>}    </span>
<span id="cb329-20"><a href="mixed-models.html#cb329-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb329-21"><a href="mixed-models.html#cb329-21" aria-hidden="true" tabindex="-1"></a><span class="fu">qqPlot</span>(assoc<span class="sc">$</span>Score.pval[assoc<span class="sc">$</span>MAC <span class="sc">&gt;=</span> <span class="dv">5</span>])</span></code></pre></div>
<p><img src="_main_files/figure-html/assoc_single_qq_mm-1.png" width="672" /></p>
<p>Notice that we observe much less inflation than before, when we did not adjust for ancestry and relatedness.</p>
</div>
<div id="exercise-3" class="section level2" number="7.3">
<h2><span class="header-section-number">7.3</span> Exercise</h2>
<ol style="list-style-type: decimal">
<li>Perform a single-variant association test for <code>status</code>. Adjust for sex, age, study, ancestry, and kinship in the model. Don’t forget to consider the <code>family</code> parameter. Make a QQ plot of the p-values for all variants with MAC &gt;= 5.</li>
</ol>
</div>
<div id="solution-3" class="section level2" number="7.4">
<h2><span class="header-section-number">7.4</span> Solution</h2>
<ol style="list-style-type: decimal">
<li>Perform a single-variant association test for <code>status</code>. Adjust for sex, age, study, ancestry, and kinship in the model. Don’t forget to consider the <code>family</code> parameter. Make a QQ plot of the p-values for all variants with MAC &gt;= 5.</li>
</ol>
<p>The first step is to fit the null model. We include the first 4 PCs as covariates in our model to adjust for ancestry, and we include a random effect proportional to the kinship matrix to adjust for genetic relatedness. Recall that with a binary outcome, we set <code>family = binomial</code> – because we have a random effect, this will fit an approximate logistic mixed model using the <a href="https://www.cell.com/ajhg/fulltext/S0002-9297(16)00063-X">GMMAT method</a>.</p>
<div class="sourceCode" id="cb330"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb330-1"><a href="mixed-models.html#cb330-1" aria-hidden="true" tabindex="-1"></a>nullmod.status <span class="ot">&lt;-</span> <span class="fu">fitNullModel</span>(annot, </span>
<span id="cb330-2"><a href="mixed-models.html#cb330-2" aria-hidden="true" tabindex="-1"></a>                               <span class="at">outcome=</span><span class="st">&quot;status&quot;</span>, </span>
<span id="cb330-3"><a href="mixed-models.html#cb330-3" aria-hidden="true" tabindex="-1"></a>                               <span class="at">covars=</span><span class="fu">c</span>(<span class="st">&quot;sex&quot;</span>, <span class="st">&quot;age&quot;</span>, <span class="st">&quot;study&quot;</span>, <span class="fu">paste0</span>(<span class="st">&quot;PC&quot;</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>)), </span>
<span id="cb330-4"><a href="mixed-models.html#cb330-4" aria-hidden="true" tabindex="-1"></a>                               <span class="at">cov.mat=</span>kinship, </span>
<span id="cb330-5"><a href="mixed-models.html#cb330-5" aria-hidden="true" tabindex="-1"></a>                               <span class="at">family =</span> binomial,</span>
<span id="cb330-6"><a href="mixed-models.html#cb330-6" aria-hidden="true" tabindex="-1"></a>                               <span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb330-7"><a href="mixed-models.html#cb330-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb330-8"><a href="mixed-models.html#cb330-8" aria-hidden="true" tabindex="-1"></a><span class="co"># description of the model we fit</span></span>
<span id="cb330-9"><a href="mixed-models.html#cb330-9" aria-hidden="true" tabindex="-1"></a>nullmod.status<span class="sc">$</span>model</span></code></pre></div>
<pre><code>## $hetResid
## [1] FALSE
## 
## $family
## 
## Family: binomial 
## Link function: logit 
## 
## 
## $outcome
## [1] &quot;status&quot;
## 
## $covars
## [1] &quot;sex&quot;   &quot;age&quot;   &quot;study&quot; &quot;PC1&quot;   &quot;PC2&quot;   &quot;PC3&quot;   &quot;PC4&quot;  
## 
## $formula
## [1] &quot;status ~ sex + age + study + PC1 + PC2 + PC3 + PC4 + (1|A)&quot;</code></pre>
<div class="sourceCode" id="cb332"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb332-1"><a href="mixed-models.html#cb332-1" aria-hidden="true" tabindex="-1"></a><span class="co"># fixed effect regression estimates</span></span>
<span id="cb332-2"><a href="mixed-models.html#cb332-2" aria-hidden="true" tabindex="-1"></a>nullmod.status<span class="sc">$</span>fixef</span></code></pre></div>
<pre><code>##                       Est         SE         Stat        pval
## (Intercept)  -3.182182913 0.96998026 1.076278e+01 0.001035617
## sexM          0.124463598 0.20598679 3.650951e-01 0.545690084
## age           0.019985093 0.02075004 9.276289e-01 0.335479807
## studystudy_2  0.003063956 0.24852285 1.519961e-04 0.990163394
## studystudy_3 -0.145204949 0.25448616 3.255626e-01 0.568283747
## PC1          -3.074068322 3.46894981 7.852917e-01 0.375527625
## PC2           5.076675586 3.21042684 2.500537e+00 0.113807493
## PC3          -0.236916365 3.45512128 4.701793e-03 0.945332199
## PC4           1.946715491 3.41350477 3.252400e-01 0.568475488</code></pre>
<div class="sourceCode" id="cb334"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb334-1"><a href="mixed-models.html#cb334-1" aria-hidden="true" tabindex="-1"></a><span class="co"># variance component estimates by group.var</span></span>
<span id="cb334-2"><a href="mixed-models.html#cb334-2" aria-hidden="true" tabindex="-1"></a>nullmod.status<span class="sc">$</span>varComp</span></code></pre></div>
<pre><code>##       V_A 
## 0.2923299</code></pre>
<p>Now that we have the null model, we perform the single-variant association tests and make the QQ plot the same way as before.</p>
<div class="sourceCode" id="cb336"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb336-1"><a href="mixed-models.html#cb336-1" aria-hidden="true" tabindex="-1"></a><span class="fu">resetIterator</span>(iterator, <span class="at">verbose=</span><span class="cn">FALSE</span>)</span>
<span id="cb336-2"><a href="mixed-models.html#cb336-2" aria-hidden="true" tabindex="-1"></a><span class="co"># run the single-variant association test</span></span>
<span id="cb336-3"><a href="mixed-models.html#cb336-3" aria-hidden="true" tabindex="-1"></a>assoc.status <span class="ot">&lt;-</span> <span class="fu">assocTestSingle</span>(iterator, nullmod.status, <span class="at">test=</span><span class="st">&quot;Score&quot;</span>)</span></code></pre></div>
<pre><code>## # of selected samples: 1,126</code></pre>
<div class="sourceCode" id="cb338"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb338-1"><a href="mixed-models.html#cb338-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(assoc.status)</span></code></pre></div>
<pre><code>## [1] 1129   14</code></pre>
<div class="sourceCode" id="cb340"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb340-1"><a href="mixed-models.html#cb340-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(assoc.status)</span></code></pre></div>
<pre><code>##   variant.id chr     pos allele.index n.obs         freq MAC      Score  Score.SE Score.Stat Score.pval        Est    Est.SE          PVE
## 1          1   1  970546            1  1126 0.0039964476   9  0.1975748 0.8349756  0.2366235 0.81294892  0.2833897 1.1976398 6.097484e-05
## 2          2   1  985900            1  1126 0.0492895204 111 -2.5741449 2.6188824 -0.9829173 0.32564813 -0.3753194 0.3818423 1.052129e-03
## 3          3   1 1025045            1  1126 0.0004440497   1 -0.0721219 0.2555958 -0.2821717 0.77781189 -1.1039760 3.9124270 8.670854e-05
## 4          4   1 1265550            1  1126 0.0008880995   2  0.7862901 0.4237118  1.8557191 0.06349361  4.3796725 2.3600946 3.750244e-03
## 5          5   1 1472676            1  1126 0.0071047957  16  0.6300877 1.0837470  0.5813974 0.56097266  0.5364696 0.9227246 3.681131e-04
## 6          6   1 1735725            1  1126 0.0022202487   5 -0.4029747 0.5941659 -0.6782191 0.49763280 -1.1414641 1.6830315 5.009278e-04</code></pre>
<div class="sourceCode" id="cb342"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb342-1"><a href="mixed-models.html#cb342-1" aria-hidden="true" tabindex="-1"></a><span class="fu">qqPlot</span>(assoc.status<span class="sc">$</span>Score.pval[assoc.status<span class="sc">$</span>MAC <span class="sc">&gt;=</span> <span class="dv">5</span>])</span></code></pre></div>
<p><img src="_main_files/figure-html/exercise_mm_assoc-1.png" width="672" /></p>
<div class="sourceCode" id="cb343"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb343-1"><a href="mixed-models.html#cb343-1" aria-hidden="true" tabindex="-1"></a><span class="fu">seqClose</span>(seqData)</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="ancestry-and-relatedness-inference.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="variant-annotation.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
